{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Category', 'Subcategory', 'Enrichment', 'P-value', 'P-adjusted',\n",
      "       'Q-value', 'Observed', 'miRNAs/precursors'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "miEAA = pd.read_csv('./s3_s4_miEAA.csv')\n",
    "print(miEAA.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Interactions (NPInter)'\n",
      " 'Annotations derived over miRTarBase (Gene Ontology)'\n",
      " 'Target genes (miRTarBase)' 'Diseases (MNDR)' 'Published Diseases'\n",
      " 'Pathways (KEGG)' 'KEGG (miRPathDB)' 'Tissue specific (isomiRdb)'\n",
      " 'Expressed in tissue (Tissue Atlas)']\n"
     ]
    }
   ],
   "source": [
    "# Print unique categories\n",
    "print(miEAA['Category'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Score', 'SuperPath Name', '# SuperPath Total Genes',\n",
       "       '# SuperPath Matched Genes', 'Matched Genes (Symbols)', 'Evidence URL'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc = pd.read_csv('./s3_s4_gc.csv')\n",
    "gc.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Category</th>\n",
       "      <th>Subcategory</th>\n",
       "      <th>Enrichment</th>\n",
       "      <th>P-value</th>\n",
       "      <th>P-adjusted</th>\n",
       "      <th>Q-value</th>\n",
       "      <th>Observed</th>\n",
       "      <th>miRNAs/precursors</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Interactions (NPInter)</td>\n",
       "      <td>MAT2A</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.327302</td>\n",
       "      <td>3</td>\n",
       "      <td>hsa-miR-1205; hsa-miR-29b-1-5p; hsa-miR-1250-5p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>potassium ion export across plasma membrane GO...</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.001003</td>\n",
       "      <td>0.001003</td>\n",
       "      <td>0.510639</td>\n",
       "      <td>4</td>\n",
       "      <td>hsa-miR-1205; hsa-miR-29b-1-5p; hsa-miR-4516; ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>regulation of small GTPase mediated signal tra...</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.001207</td>\n",
       "      <td>0.001207</td>\n",
       "      <td>0.510639</td>\n",
       "      <td>8</td>\n",
       "      <td>hsa-miR-1205; hsa-miR-29b-1-5p; hsa-miR-6769b-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>central nervous system development GO0007417</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.001536</td>\n",
       "      <td>0.001536</td>\n",
       "      <td>0.510639</td>\n",
       "      <td>9</td>\n",
       "      <td>hsa-miR-1205; hsa-miR-1250-5p; hsa-miR-4516; h...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>negative regulation of CD8-positive, alpha-bet...</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.001538</td>\n",
       "      <td>0.001538</td>\n",
       "      <td>0.510639</td>\n",
       "      <td>3</td>\n",
       "      <td>hsa-miR-1205; hsa-miR-29b-1-5p; hsa-miR-4516</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Category  \\\n",
       "0                             Interactions (NPInter)   \n",
       "1  Annotations derived over miRTarBase (Gene Onto...   \n",
       "2  Annotations derived over miRTarBase (Gene Onto...   \n",
       "3  Annotations derived over miRTarBase (Gene Onto...   \n",
       "4  Annotations derived over miRTarBase (Gene Onto...   \n",
       "\n",
       "                                         Subcategory Enrichment   P-value  \\\n",
       "0                                              MAT2A   enriched  0.000385   \n",
       "1  potassium ion export across plasma membrane GO...   enriched  0.001003   \n",
       "2  regulation of small GTPase mediated signal tra...   enriched  0.001207   \n",
       "3       central nervous system development GO0007417   enriched  0.001536   \n",
       "4  negative regulation of CD8-positive, alpha-bet...   enriched  0.001538   \n",
       "\n",
       "   P-adjusted   Q-value  Observed  \\\n",
       "0    0.000385  0.327302         3   \n",
       "1    0.001003  0.510639         4   \n",
       "2    0.001207  0.510639         8   \n",
       "3    0.001536  0.510639         9   \n",
       "4    0.001538  0.510639         3   \n",
       "\n",
       "                                   miRNAs/precursors  \n",
       "0    hsa-miR-1205; hsa-miR-29b-1-5p; hsa-miR-1250-5p  \n",
       "1  hsa-miR-1205; hsa-miR-29b-1-5p; hsa-miR-4516; ...  \n",
       "2  hsa-miR-1205; hsa-miR-29b-1-5p; hsa-miR-6769b-...  \n",
       "3  hsa-miR-1205; hsa-miR-1250-5p; hsa-miR-4516; h...  \n",
       "4       hsa-miR-1205; hsa-miR-29b-1-5p; hsa-miR-4516  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miEAA.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Score</th>\n",
       "      <th>SuperPath Name</th>\n",
       "      <th># SuperPath Total Genes</th>\n",
       "      <th># SuperPath Matched Genes</th>\n",
       "      <th>Matched Genes (Symbols)</th>\n",
       "      <th>Evidence URL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Signal Transduction</td>\n",
       "      <td>2597</td>\n",
       "      <td>453</td>\n",
       "      <td>MAP2K1, ARL4C, NHS, CARM1, ADM, ADM2, NCKAP1, ...</td>\n",
       "      <td>https://pathcards.genecards.org/card/signal_tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Infectious Disease</td>\n",
       "      <td>1818</td>\n",
       "      <td>339</td>\n",
       "      <td>CANX, CAPN1, RPS2, MAP2K1, MAP2K3, RPL12, MUC4...</td>\n",
       "      <td>https://pathcards.genecards.org/card/infectiou...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Gene Expression (Transcription)</td>\n",
       "      <td>1543</td>\n",
       "      <td>320</td>\n",
       "      <td>LRPPRC, MAP2K6, ZNF317, CARM1, ZNF154, HAND2, ...</td>\n",
       "      <td>https://pathcards.genecards.org/card/gene_expr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>53.0</td>\n",
       "      <td>Regulation of Activated PAK-2p34 By Proteasome...</td>\n",
       "      <td>943</td>\n",
       "      <td>193</td>\n",
       "      <td>CANX, EIF2S1, EIF2S3, MAP2K4, ABCA12, FBXL7, T...</td>\n",
       "      <td>https://pathcards.genecards.org/card/regulatio...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>53.0</td>\n",
       "      <td>Cellular Responses to Stimuli</td>\n",
       "      <td>819</td>\n",
       "      <td>175</td>\n",
       "      <td>EIF2S1, RPS2, MAP2K3, RPL12, EIF2S3, CAPZA1, C...</td>\n",
       "      <td>https://pathcards.genecards.org/card/cellular_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Score                                     SuperPath Name  \\\n",
       "0  166.1                                Signal Transduction   \n",
       "1  166.1                                 Infectious Disease   \n",
       "2  166.1                    Gene Expression (Transcription)   \n",
       "3   53.0  Regulation of Activated PAK-2p34 By Proteasome...   \n",
       "4   53.0                      Cellular Responses to Stimuli   \n",
       "\n",
       "   # SuperPath Total Genes  # SuperPath Matched Genes  \\\n",
       "0                     2597                        453   \n",
       "1                     1818                        339   \n",
       "2                     1543                        320   \n",
       "3                      943                        193   \n",
       "4                      819                        175   \n",
       "\n",
       "                             Matched Genes (Symbols)  \\\n",
       "0  MAP2K1, ARL4C, NHS, CARM1, ADM, ADM2, NCKAP1, ...   \n",
       "1  CANX, CAPN1, RPS2, MAP2K1, MAP2K3, RPL12, MUC4...   \n",
       "2  LRPPRC, MAP2K6, ZNF317, CARM1, ZNF154, HAND2, ...   \n",
       "3  CANX, EIF2S1, EIF2S3, MAP2K4, ABCA12, FBXL7, T...   \n",
       "4  EIF2S1, RPS2, MAP2K3, RPL12, EIF2S3, CAPZA1, C...   \n",
       "\n",
       "                                        Evidence URL  \n",
       "0  https://pathcards.genecards.org/card/signal_tr...  \n",
       "1  https://pathcards.genecards.org/card/infectiou...  \n",
       "2  https://pathcards.genecards.org/card/gene_expr...  \n",
       "3  https://pathcards.genecards.org/card/regulatio...  \n",
       "4  https://pathcards.genecards.org/card/cellular_...  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Interactions (NPInter)'\n",
      " 'Annotations derived over miRTarBase (Gene Ontology)'\n",
      " 'Target genes (miRTarBase)' 'Diseases (MNDR)' 'Published Diseases'\n",
      " 'Pathways (KEGG)' 'KEGG (miRPathDB)' 'Tissue specific (isomiRdb)'\n",
      " 'Expressed in tissue (Tissue Atlas)']\n",
      "Category\n",
      "Annotations derived over miRTarBase (Gene Ontology)    354\n",
      "Interactions (NPInter)                                 122\n",
      "Target genes (miRTarBase)                               47\n",
      "Expressed in tissue (Tissue Atlas)                      15\n",
      "Diseases (MNDR)                                         12\n",
      "Pathways (KEGG)                                         11\n",
      "Published Diseases                                       5\n",
      "KEGG (miRPathDB)                                         1\n",
      "Tissue specific (isomiRdb)                               1\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# print the categories from Reactome\n",
    "print(miEAA['Category'].unique())\n",
    "print(miEAA['Category'].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Interactions (NPInter)\n",
      "['MAT2A' 'C18orf8' 'CHD4' 'GPX3' 'IP6K3' 'SERPING1' 'USP9Y' 'ENPP2'\n",
      " 'KIAA1109' 'PRNP' 'CRIM1' 'CTSB' 'PNMA1' 'RBM3' 'CIRBP' 'DISC1' 'EDIL3'\n",
      " 'ERBB2' 'FMR1' 'G3BP1' 'GNAI1' 'HES1' 'HPRT1' 'MBIP' 'NPTN' 'PHYHIPL'\n",
      " 'PTEN' 'REEP3' 'REEP5' 'SUZ12' 'UBE2B' 'USO1' 'VEZT' 'ALKBH5' 'ACPL2'\n",
      " 'ACTA1' 'ACVR2A' 'ADAMTS9' 'ANKRD26' 'ARF6' 'ARGLU1' 'ARRDC3' 'ATP13A3'\n",
      " 'ATP1A1' 'ATXN1' 'BDP1' 'BOD1' 'BTAF1' 'C14orf37' 'CAB39' 'CAPN2' 'CBLN2'\n",
      " 'CDK2AP1' 'CEBPD' 'CENPQ' 'CFLAR' 'COL4A2' 'COPS8' 'DNAJB9' 'DOCK5'\n",
      " 'EIF1' 'FAM13C' 'FBXL3' 'FBXW11' 'FGB' 'FOS' 'FOXC1' 'GATA6' 'GINS1'\n",
      " 'GJA1' 'GOLGA4' 'GPR37' 'GREM1' 'IBTK' 'IDH3G' 'JPH1' 'KCNMB4' 'KIF5C'\n",
      " 'LHFPL2' 'LIMCH1' 'LRP12' 'MBNL3' 'MINK1' 'MOG' 'MTUS1' 'NEFL' 'NEK9'\n",
      " 'NIPSNAP3A' 'OGG1' 'PCSK6' 'PIP4K2A' 'PISD' 'PMP22' 'PPP1R15B' 'PPP1R1C'\n",
      " 'PRDM2' 'PTPLAD1' 'RAB12' 'RALGPS2' 'RANBP9' 'RBM12B' 'RBM23' 'RBM24'\n",
      " 'ROBO1' 'RRBP1' 'SLC3A1' 'SNRNP70' 'SORL1' 'SPAST' 'SUSD1' 'TMEM165'\n",
      " 'TMEM184B' 'TNRC18' 'UBQLN1' 'UXS1' 'WHSC1' 'YTHDF2' 'YWHAH' 'ZCCHC3'\n",
      " 'ZEB2' 'ZNF512' 'ZZZ3']\n",
      "Annotations derived over miRTarBase (Gene Ontology)\n",
      "['potassium ion export across plasma membrane GO0097623'\n",
      " 'regulation of small GTPase mediated signal transduction GO0051056'\n",
      " 'central nervous system development GO0007417'\n",
      " 'negative regulation of CD8-positive, alpha-beta T cell differentiation GO0043377'\n",
      " 'axonogenesis GO0007409' 'kinetochore GO0000776'\n",
      " 'membrane repolarization during cardiac muscle cell action potential GO0086013'\n",
      " 'microtubule nucleation GO0007020'\n",
      " 'cellular response to interleukin-1 GO0071347'\n",
      " 'N-acetylneuraminate catabolic process GO0019262'\n",
      " 'antigen processing and presentation of endogenous peptide antigen via MHC class I GO0019885'\n",
      " 'cellular response to hepatocyte growth factor stimulus GO0035729'\n",
      " 'cellular response to nicotine GO0071316'\n",
      " 'hepatocyte growth factor receptor signaling pathway GO0048012'\n",
      " 'mitochondrial fission GO0000266'\n",
      " 'negative regulation of forebrain neuron differentiation GO2000978'\n",
      " 'regulation of neuron apoptotic process GO0043523'\n",
      " 'structural constituent of skin epidermis GO0030280'\n",
      " 'tRNA-splicing ligase complex GO0072669'\n",
      " 'tRNA splicing, via endonucleolytic cleavage and ligation GO0006388'\n",
      " 'transmembrane receptor protein serinethreonine kinase activity GO0004675'\n",
      " 'tricuspid valve morphogenesis GO0003186'\n",
      " 'cellular response to growth factor stimulus GO0071363'\n",
      " 'Cul4-RING E3 ubiquitin ligase complex GO0080008' 'U5 snRNP GO0005682'\n",
      " 'nuclear matrix GO0016363' 'tau protein binding GO0048156'\n",
      " 'vesicle fusion with Golgi apparatus GO0048280'\n",
      " 'potassium ion transmembrane transport GO0071805'\n",
      " 'endoplasmic reticulum tubular network GO0071782'\n",
      " 'peroxisomal matrix GO0005782' 'CCR chemokine receptor binding GO0048020'\n",
      " 'cellular response to organic cyclic compound GO0071407'\n",
      " 'eosinophil chemotaxis GO0048245'\n",
      " 'membrane repolarization during ventricular cardiac muscle cell action potential GO0098915'\n",
      " 'monocyte chemotaxis GO0002548'\n",
      " 'voltage-gated potassium channel activity involved in cardiac muscle cell action potential repolarization GO0086008'\n",
      " 'protein autophosphorylation GO0046777'\n",
      " 'cellular response to lipopolysaccharide GO0071222'\n",
      " 'NLS-bearing protein import into nucleus GO0006607'\n",
      " 'apical junction complex GO0043296'\n",
      " 'membrane protein intracellular domain proteolysis GO0031293'\n",
      " 'eukaryotic translation initiation factor 4F complex GO0016281'\n",
      " 'monoatomic ion homeostasis GO0050801'\n",
      " 'positive regulation of protein import into nucleus GO0042307'\n",
      " 'positive regulation of sodium ion transmembrane transporter activity GO2000651'\n",
      " 'positive regulation of interleukin-1 beta production GO0032731'\n",
      " 'stimulatory C-type lectin receptor signaling pathway GO0002223'\n",
      " 'p53 binding GO0002039' 'dynein complex binding GO0070840'\n",
      " 'positive regulation of Rho protein signal transduction GO0035025'\n",
      " 'Fc-epsilon receptor signaling pathway GO0038095'\n",
      " 'cortical actin cytoskeleton GO0030864'\n",
      " 'hematopoietic progenitor cell differentiation GO0002244'\n",
      " 'spliceosomal complex assembly GO0000245'\n",
      " 'regulation of translation GO0006417' 'cell differentiation GO0030154'\n",
      " 'U2-type precatalytic spliceosome GO0071005'\n",
      " 'dystroglycan binding GO0002162' 'endothelial cell activation GO0042118'\n",
      " 'establishment of epithelial cell polarity GO0090162'\n",
      " 'learning GO0007612'\n",
      " 'positive regulation of innate immune response GO0045089'\n",
      " 'positive regulation of type I interferon-mediated signaling pathway GO0060340'\n",
      " 'RSC-type complex GO0016586' 'apical plasma membrane GO0016324'\n",
      " 'gamma-secretase complex GO0070765'\n",
      " 'barbed-end actin filament capping GO0051016'\n",
      " 'killing of cells of another organism GO0031640'\n",
      " 'negative regulation of BMP signaling pathway GO0030514'\n",
      " 'positive regulation of natural killer cell mediated cytotoxicity GO0045954'\n",
      " 'precatalytic spliceosome GO0071011'\n",
      " 'triglyceride catabolic process GO0019433'\n",
      " 'androgen receptor signaling pathway GO0030521'\n",
      " 'lymphocyte chemotaxis GO0048247'\n",
      " 'negative regulation of cell-matrix adhesion GO0001953'\n",
      " 'ovarian follicle development GO0001541'\n",
      " 'positive regulation of cytoplasmic translation GO2000767'\n",
      " 'positive regulation of interleukin-1 production GO0032732'\n",
      " 'prostaglandin metabolic process GO0006693'\n",
      " 'SCF ubiquitin ligase complex GO0019005'\n",
      " 'antimicrobial humoral immune response mediated by antimicrobial peptide GO0061844'\n",
      " 'condensed chromosome GO0000793' 'angiostatin binding GO0043532'\n",
      " 'chloride channel inhibitor activity GO0019869'\n",
      " 'mitochondrial proton-transporting ATP synthase, catalytic core GO0005754'\n",
      " 'positive regulation of smoothened signaling pathway GO0045880'\n",
      " 'proteoglycan binding GO0043394'\n",
      " 'proton-transporting ATP synthase complex, catalytic core F1 GO0045261'\n",
      " 'proton-transporting ATP synthase complex GO0045259'\n",
      " '7-methylguanosine cap hypermethylation GO0036261'\n",
      " 'CD8-positive, alpha-beta T cell activation GO0036037'\n",
      " 'N-acetylgalactosaminyl-proteoglycan 3-beta-glucuronosyltransferase activity GO0050510'\n",
      " 'T cell receptor binding GO0042608' 'U1 snRNP GO0005685'\n",
      " 'cell fate specification GO0001708'\n",
      " 'cellular response to dexamethasone stimulus GO0071549'\n",
      " 'collagen type V trimer GO0005588' 'collagen type XI trimer GO0005592'\n",
      " 'commitment complex GO0000243'\n",
      " 'eukaryotic translation initiation factor 3 complex GO0005852'\n",
      " 'eye morphogenesis GO0048592'\n",
      " 'formation of cytoplasmic translation initiation complex GO0001732'\n",
      " 'glucose-1,6-bisphosphate synthase activity GO0047933'\n",
      " 'glucuronosyl-N-acetylgalactosaminyl-proteoglycan 4-beta-N-acetylgalactosaminyltransferase activity GO0047238'\n",
      " 'histone H4K20 methyltransferase activity GO0042799'\n",
      " 'integrin biosynthetic process GO0045112'\n",
      " 'inward rectifier potassium channel complex GO1902937'\n",
      " 'membrane depolarization during action potential GO0086010'\n",
      " 'membrane repolarization during action potential GO0086011'\n",
      " 'monoatomic anion transmembrane transporter activity GO0008509'\n",
      " 'negative regulation of endodermal cell differentiation GO1903225'\n",
      " 'negative regulation of potassium ion export across plasma membrane GO1903765'\n",
      " 'negative regulation of transcription by RNA polymerase III GO0016480'\n",
      " 'negative regulation of tyrosine phosphorylation of STAT protein GO0042532'\n",
      " 'neuron remodeling GO0016322' 'pICln-Sm protein complex GO0034715'\n",
      " 'positive regulation of CD8-positive, alpha-beta T cell activation GO2001187'\n",
      " 'positive regulation of CD8-positive, alpha-beta T cell proliferation GO2000566'\n",
      " 'positive regulation of enamel mineralization GO0070175'\n",
      " 'positive regulation of interleukin-17 production GO0032740'\n",
      " 'positive regulation of potassium ion transmembrane transport GO1901381'\n",
      " 'protein complex oligomerization GO0051259'\n",
      " 'regulation of bone resorption GO0045124'\n",
      " 'regulation of potassium ion transmembrane transport GO1901379'\n",
      " 'regulation of protein localization to plasma membrane GO1903076'\n",
      " 'regulation of resting membrane potential GO0060075'\n",
      " 'spliceosomal tri-snRNP complex GO0097526'\n",
      " 'transport vesicle membrane GO0030658'\n",
      " 'ventricular cardiac muscle cell action potential GO0086005'\n",
      " 'membrane repolarization GO0086009'\n",
      " 'nuclear androgen receptor binding GO0050681' 'amelogenesis GO0097186'\n",
      " 'base-excision repair, gap-filling GO0006287' 'ciliary base GO0097546'\n",
      " 'dorsalventral pattern formation GO0009953'\n",
      " 'establishment of mitotic spindle localization GO0040001'\n",
      " 'locomotory behavior GO0007626'\n",
      " 'negative regulation of amyloid-beta clearance GO1900222'\n",
      " 'positive regulation of viral life cycle GO1903902'\n",
      " 'regulation of mitochondrial translation GO0070129'\n",
      " 'regulation of membrane potential GO0042391'\n",
      " 'positive regulation of peptidyl-tyrosine phosphorylation GO0050731'\n",
      " 'activation of NF-kappaB-inducing kinase activity GO0007250'\n",
      " 'positive regulation of interleukin-10 production GO0032733'\n",
      " 'positive regulation of phagocytosis GO0050766'\n",
      " 'protein targeting GO0006605'\n",
      " 'endoplasmic reticulum organization GO0007029'\n",
      " 'protein N-linked glycosylation GO0006487'\n",
      " 'embryonic organ development GO0048568'\n",
      " 'extracellular matrix assembly GO0085029'\n",
      " 'transmembrane transporter activity GO0022857'\n",
      " 'execution phase of apoptosis GO0097194'\n",
      " 'glycerophospholipid catabolic process GO0046475'\n",
      " 'interleukin-17A-mediated signaling pathway GO0038173'\n",
      " 'mRNA regulatory element binding translation repressor activity GO0000900'\n",
      " 'peptidyl-tyrosine dephosphorylation GO0035335'\n",
      " 'positive regulation of execution phase of apoptosis GO1900119'\n",
      " 'positive regulation of phosphorylation GO0042327'\n",
      " 'post-Golgi vesicle-mediated transport GO0006892'\n",
      " 'protein-lysine N-methyltransferase activity GO0016279'\n",
      " 'ubiquitin-ubiquitin ligase activity GO0034450'\n",
      " 'voltage-gated potassium channel activity GO0005249'\n",
      " 'neuron projection GO0043005' 'receptor complex GO0043235'\n",
      " 'maturation of SSU-rRNA from tricistronic rRNA transcript SSU-rRNA, 5.8S rRNA, LSU-rRNA GO0000462'\n",
      " 'regulation of G protein-coupled receptor signaling pathway GO0008277'\n",
      " 'cellular response to reactive oxygen species GO0034614'\n",
      " 'nuclear localization sequence binding GO0008139'\n",
      " '3-5 DNA helicase activity GO0043138'\n",
      " 'GDP-mannose 4,6-dehydratase activity GO0008446'\n",
      " 'GDP-mannose metabolic process GO0019673' 'NADP+ binding GO0070401'\n",
      " 'axonemal outer doublet GO0097545'\n",
      " 'centriolar subdistal appendage GO0120103'\n",
      " 'cholesterol transport GO0030301'\n",
      " 'de novo GDP-L-fucose biosynthetic process GO0042351'\n",
      " 'four-way junction helicase activity GO0009378'\n",
      " 'inner dynein arm assembly GO0036159'\n",
      " 'marginal zone B cell differentiation GO0002315'\n",
      " 'microtubule-based process GO0007017'\n",
      " 'neurotransmitter biosynthetic process GO0042136'\n",
      " 'nitric oxide mediated signal transduction GO0007263'\n",
      " 'platelet dense tubular network GO0031094'\n",
      " 'regulation of endocytic recycling GO2001135'\n",
      " 'regulation of heme biosynthetic process GO0070453'\n",
      " 'regulation of protein metabolic process GO0051246'\n",
      " 'insulin receptor signaling pathway GO0008286'\n",
      " 'transcription factor TFTC complex GO0033276'\n",
      " 'AMP-activated protein kinase activity GO0004679'\n",
      " 'DNA-dependent protein kinase complex GO0070418'\n",
      " 'DNA end binding GO0045027' 'Ku70Ku80 complex GO0043564'\n",
      " 'NLS-dependent protein nuclear import complex GO0042564'\n",
      " 'actin filament reorganization GO0090527'\n",
      " 'cell junction assembly GO0034329' 'cyclin D2-CDK4 complex GO0097129'\n",
      " 'ether lipid metabolic process GO0046485'\n",
      " 'heterochromatin organization GO0070828'\n",
      " 'monoatomic anion transport GO0006820'\n",
      " 'nonhomologous end joining complex GO0070419'\n",
      " 'nuclear import signal receptor activity GO0061608'\n",
      " 'nucleobase-containing compound metabolic process GO0006139'\n",
      " 'nucleotide-activated protein kinase complex GO0031588'\n",
      " 'peptidyl-lysine dimethylation GO0018027'\n",
      " 'peptidyl-lysine monomethylation GO0018026'\n",
      " 'positive regulation of interferon-beta production GO0032728'\n",
      " 'positive regulation of natural killer cell mediated cytotoxicity directed against tumor cell target GO0002860'\n",
      " 'recombinational repair GO0000725'\n",
      " 'regulation of heart rate by hormone GO0003064'\n",
      " 'regulation of membrane repolarization GO0060306'\n",
      " 'response to heat GO0009408'\n",
      " 'susceptibility to T cell mediated cytotoxicity GO0060370'\n",
      " 'susceptibility to natural killer cell mediated cytotoxicity GO0042271'\n",
      " 'MHC class I protein binding GO0042288'\n",
      " 'regulation of glycolytic process GO0006110'\n",
      " 'SCF-dependent proteasomal ubiquitin-dependent protein catabolic process GO0031146'\n",
      " 'delayed rectifier potassium channel activity GO0005251'\n",
      " 'positive regulation of chemotaxis GO0050921'\n",
      " 'potassium ion homeostasis GO0055075'\n",
      " 'antigen processing and presentation GO0019882'\n",
      " 'central nervous system neuron development GO0021954'\n",
      " 'positive regulation of T cell proliferation GO0042102'\n",
      " 'trans-Golgi network GO0005802'\n",
      " 'cyclin-dependent protein serinethreonine kinase activator activity GO0061575'\n",
      " 'phosphorylation GO0016310'\n",
      " 'positive regulation of inflammatory response GO0050729'\n",
      " 'regulation of transcription involved in G1S transition of mitotic cell cycle GO0000083'\n",
      " 'establishment or maintenance of cell polarity GO0007163'\n",
      " 'negative regulation of I-kappaB kinaseNF-kappaB signaling GO0043124'\n",
      " 'calcium-dependent protein binding GO0048306'\n",
      " 'engulfment of apoptotic cell GO0043652'\n",
      " 'enzyme activator activity GO0008047'\n",
      " 'non-membrane spanning protein tyrosine kinase activity GO0004715'\n",
      " 'negative regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway GO1902236'\n",
      " 'negative regulation of protein localization to plasma membrane GO1903077'\n",
      " 'post-translational protein modification GO0043687'\n",
      " 'regulation of dosage compensation by inactivation of X chromosome GO1900095'\n",
      " 'regulation of focal adhesion assembly GO0051893'\n",
      " 'cytokine-mediated signaling pathway GO0019221'\n",
      " 'growth factor activity GO0008083'\n",
      " 'K48-linked deubiquitinase activity GO1990380'\n",
      " 'L-glutamine transmembrane transporter activity GO0015186'\n",
      " 'MIS12MIND type complex GO0000444' 'Set1CCOMPASS complex GO0048188'\n",
      " 'acyl-CoA oxidase activity GO0003997' 'autolysosome GO0044754'\n",
      " 'brain morphogenesis GO0048854'\n",
      " 'embryonic cranial skeleton morphogenesis GO0048701'\n",
      " 'fatty acid beta-oxidation using acyl-CoA oxidase GO0033540'\n",
      " 'fatty acid catabolic process GO0009062'\n",
      " 'glucuronosyltransferase activity GO0015020'\n",
      " 'glutamine transport GO0006868'\n",
      " 'hydrogen peroxide biosynthetic process GO0050665'\n",
      " 'lysophospholipase activity GO0004622'\n",
      " 'mRNA splice site selection GO0006376'\n",
      " 'megakaryocyte development GO0035855'\n",
      " 'negative regulation of hematopoietic progenitor cell differentiation GO1901533'\n",
      " 'negative regulation of hippo signaling GO0035331'\n",
      " 'neurotransmitter transport GO0006836'\n",
      " 'palmitoyl-CoA oxidase activity GO0016401'\n",
      " 'positive regulation of DNA damage response, signal transduction by p53 class mediator resulting in transcription of p21 class mediator GO1902164'\n",
      " 'positive regulation of protein acetylation GO1901985'\n",
      " 'positive regulation of tyrosine phosphorylation of STAT protein GO0042531'\n",
      " 'regulation of p38MAPK cascade GO1900744'\n",
      " 'regulation of vesicle size GO0097494'\n",
      " 'renal water homeostasis GO0003091' 'secretory granule GO0030141'\n",
      " 'very long-chain fatty acid beta-oxidation GO0140493'\n",
      " 'actin filament GO0005884' 'regulation of gene expression GO0010468'\n",
      " 'mitochondrial proton-transporting ATP synthase complex GO0005753'\n",
      " 'proton motive force-driven ATP synthesis GO0015986'\n",
      " 'synaptic vesicle membrane GO0030672' 'excitatory synapse GO0060076'\n",
      " 'extrinsic apoptotic signaling pathway via death domain receptors GO0008625'\n",
      " 'fructose binding GO0070061'\n",
      " 'growth hormone receptor signaling pathway via JAK-STAT GO0060397'\n",
      " 'negative regulation of vascular endothelial growth factor receptor signaling pathway GO0030948'\n",
      " 'plasma membrane signaling receptor complex GO0098802'\n",
      " 'positive regulation of endopeptidase activity GO0010950'\n",
      " 'positive regulation of nitric-oxide synthase activity GO0051000'\n",
      " 'receptor signaling pathway via JAK-STAT GO0007259'\n",
      " 'trans-Golgi network transport vesicle GO0030140' 'chemotaxis GO0006935'\n",
      " 'mitotic sister chromatid segregation GO0000070'\n",
      " 'negative regulation of phosphoprotein phosphatase activity GO0032515'\n",
      " 'vasculogenesis GO0001570'\n",
      " 'A-type transient outward potassium channel activity GO0005250'\n",
      " 'BRE binding GO0042835' 'CRD-mediated mRNA stability complex GO0070937'\n",
      " 'CRD-mediated mRNA stabilization GO0070934'\n",
      " 'GTP metabolic process GO0046039' 'Golgi disassembly GO0090166'\n",
      " 'L-methionine salvage GO0071267' 'PAR polarity complex GO0120157'\n",
      " 'S-adenosylmethionine metabolic process GO0046500'\n",
      " 'S-methylmethionine-homocysteine S-methyltransferase activity GO0061627'\n",
      " 'S-methylmethionine metabolic process GO0033477'\n",
      " 'activation of cysteine-type endopeptidase activity GO0097202'\n",
      " 'activation of cysteine-type endopeptidase activity involved in apoptotic process by cytochrome c GO0008635'\n",
      " 'activation of phospholipase D activity GO0031584'\n",
      " 'amino-acid betaine metabolic process GO0006577' 'apoptosome GO0043293'\n",
      " 'arginine binding GO0034618' 'asymmetric cell division GO0008356'\n",
      " 'axonemal dynein complex assembly GO0070286' 'basal dendrite GO0097441'\n",
      " 'betaine-homocysteine S-methyltransferase activity GO0047150'\n",
      " 'bone remodeling GO0046849' 'cell motility GO0048870'\n",
      " 'central nervous system myelination GO0022010'\n",
      " 'chromatin binding GO0003682' 'dihydropyrimidinase activity GO0004157'\n",
      " 'dynein axonemal particle GO0120293'\n",
      " 'embryonic brain development GO1990403'\n",
      " 'entry of viral genome into host nucleus through nuclear pore complex via importin GO0075506'\n",
      " 'fatty acid elongase complex GO0009923' 'forebrain development GO0030900'\n",
      " 'gastrulation GO0007369'\n",
      " 'hydrolase activity, acting on carbon-nitrogen but not peptide bonds, in cyclic amides GO0016812'\n",
      " 'hydrolase activity GO0016787' 'lipid droplet formation GO0140042'\n",
      " 'mRNA 3-splice site recognition GO0000389'\n",
      " 'meiotic spindle assembly GO0090306' 'molecular function GO0003674'\n",
      " 'myelination in peripheral nervous system GO0022011'\n",
      " 'negative regulation of JUN kinase activity GO0043508'\n",
      " 'negative regulation of nuclear-transcribed mRNA catabolic process, deadenylation-dependent decay GO1900152'\n",
      " 'neutrophil migration GO1990266' 'nucleolus GO0005730'\n",
      " 'photoreceptor disc membrane GO0097381'\n",
      " 'positive regulation of RNA polymerase II regulatory region sequence-specific DNA binding GO1905636'\n",
      " 'positive regulation of cytokine production involved in immune response GO0002720'\n",
      " 'positive regulation of endothelial cell differentiation GO0045603'\n",
      " 'positive regulation of gene expression GO0010628'\n",
      " 'positive regulation of immune response GO0050778'\n",
      " 'positive regulation of protein glycosylation GO0060050'\n",
      " 'positive regulation of superoxide anion generation GO0032930'\n",
      " 'positive regulation of type II interferon-mediated signaling pathway GO0060335'\n",
      " 'protein localization to site of double-strand break GO1990166'\n",
      " 'protein serinethreonine kinase activity GO0004674'\n",
      " 'pyrimidine nucleobase catabolic process GO0006208'\n",
      " 'regulation of cytokine production involved in inflammatory response GO1900015'\n",
      " 'regulation of hippo signaling GO0035330'\n",
      " 'regulation of rhodopsin mediated signaling pathway GO0022400'\n",
      " 'regulatory T cell differentiation GO0045066'\n",
      " 'response to UV-A GO0070141' 'response to UV-C GO0010225'\n",
      " 'rhodopsin kinase activity GO0050254' 'sarcomere GO0030017'\n",
      " 'stem cell factor receptor binding GO0005173'\n",
      " 'tubular endosome GO0097422' 'vesicle cargo loading GO0035459']\n",
      "Target genes (miRTarBase)\n",
      "['B2M' 'CCDC71L' 'WDR13' 'YIPF4' 'EIF4H' 'LRRC58' 'DCAF7' 'ATP5A1' 'CHSY1'\n",
      " 'COL5A1' 'EIF3H' 'KCNH2' 'KCNK5' 'NR3C1' 'PGM2L1' 'SLC4A2' 'TNFAIP1'\n",
      " 'AHCY' 'BTG2' 'CCL16' 'CCND1' 'CFAP73' 'DNAJC28' 'DNTTIP2' 'EPS15L1'\n",
      " 'GMDS' 'PTPN4' 'SRRD' 'ZCCHC2' 'RUFY2' 'CCDC80' 'PVR' 'ACOX1' 'TAOK1'\n",
      " 'BHMT2' 'CCND2' 'DAZAP2' 'FAM241A' 'GOLGA2' 'GPR155' 'KPNA6' 'PER1'\n",
      " 'PNRC1' 'SLC35F5' 'SLC35F6' 'ZNF12' 'ZNF317']\n",
      "Diseases (MNDR)\n",
      "['Embryonal cancer' 'Glioma' 'Arthritis' 'Urinary bladder cancer'\n",
      " 'Stroke lacunar' 'Cholangiocarcinoma' 'Neurodegenerative diseases'\n",
      " 'Vascular diseases' 'Bladder urothelial carcinoma' 'Breast carcinoma'\n",
      " 'Head and neck cancer' 'Pancreatic ductal adenocarcinoma']\n",
      "Published Diseases\n",
      "['pancreatic cancer ducatal deregulated'\n",
      " 'pancreatic cancer ducatal upregulated' 'pancreatitis deregulated'\n",
      " 'peridontitis downregulated' 'pancreatitis downregulated']\n",
      "Pathways (KEGG)\n",
      "['Non-small cell lung cancer' 'Pancreatic cancer'\n",
      " 'Amino sugar and nucleotide sugar metabolism' 'Cushing syndrome'\n",
      " 'Starch and sucrose metabolism' 'Fructose and mannose metabolism'\n",
      " 'Longevity regulating pathway - multiple species' 'Breast cancer'\n",
      " 'Gastric cancer' 'Fatty acid degradation'\n",
      " 'Carbohydrate digestion and absorption']\n",
      "KEGG (miRPathDB)\n",
      "['Epstein-Barr virus infection']\n",
      "Tissue specific (isomiRdb)\n",
      "['saliva specific']\n",
      "Expressed in tissue (Tissue Atlas)\n",
      "['adipocyte' 'artery' 'colon' 'dura mater' 'epididymis' 'kidney' 'liver'\n",
      " 'lung' 'lymph node' 'nerve' 'pancreas' 'small intestine' 'spleen'\n",
      " 'stomach' 'tunica albuginea']\n"
     ]
    }
   ],
   "source": [
    "# find the overlap of the overlap of supepath name and subcategory\n",
    "for i in miEAA['Category'].unique():\n",
    "    print(i)\n",
    "    print(miEAA[miEAA['Category'] == i]['Subcategory'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Signal Transduction', 'Infectious Disease',\n",
       "       'Gene Expression (Transcription)',\n",
       "       'Regulation of Activated PAK-2p34 By Proteasome Mediated Degradation',\n",
       "       'Cellular Responses to Stimuli',\n",
       "       'RNA Polymerase I Promoter Opening', 'Pleural Mesothelioma',\n",
       "       'Metabolism of Proteins', 'Cellular Senescence',\n",
       "       'Signaling By Receptor Tyrosine Kinases', 'Prolactin Signaling',\n",
       "       'Apoptotic Pathways in Synovial Fibroblasts', 'ERK Signaling',\n",
       "       'Signaling By Rho GTPases', 'Cell Cycle, Mitotic',\n",
       "       'MAPK Family Signaling Cascades', 'Innate Immune System',\n",
       "       'GPCR Pathway', 'Toll-like Receptor Signaling Pathway',\n",
       "       'Sertoli-Sertoli Cell Junction Dynamics',\n",
       "       'Regulation of TP53 Activity',\n",
       "       'Chromatin Regulation / Acetylation', 'Cytoskeletal Signaling',\n",
       "       'Integrin-mediated Cell Adhesion', 'ESR-mediated Signaling',\n",
       "       'SARS-CoV-2 Infection',\n",
       "       'Validated Targets of C-MYC Transcriptional Activation',\n",
       "       'Chromatin Organization', 'Endometrial Cancer',\n",
       "       'NFAT and Cardiac Hypertrophy', 'P70S6K Signaling',\n",
       "       'TGF-Beta Pathway', 'VEGF Pathway (Qiagen)', 'MAPK-Erk Pathway',\n",
       "       'Homology Directed Repair', 'MAP Kinase Signaling',\n",
       "       'Focal Adhesion',\n",
       "       'Class I MHC Mediated Antigen Processing and Presentation',\n",
       "       'Akt Signaling', 'Nervous System Development',\n",
       "       'Assembly of The Pre-replicative Complex',\n",
       "       'Retinoblastoma Gene in Cancer', 'CKAP4 Signaling Pathway Map',\n",
       "       'Breast Cancer Pathway', 'Regulation of Telomerase',\n",
       "       'Vesicle-mediated Transport', 'MAPK Signaling Pathway',\n",
       "       'Cytokine Signaling in Immune System',\n",
       "       'PIP3 Activates AKT Signaling',\n",
       "       'Response to Elevated Platelet Cytosolic Ca2+',\n",
       "       'Cell Cycle Regulation of G1/S Transition (part 1)',\n",
       "       'Development VEGF Signaling Via VEGFR2 - Generic Cascades',\n",
       "       'TP53 Regulates Metabolic Genes', 'NGF Pathway',\n",
       "       'PI3K-Akt Signaling Pathway',\n",
       "       'MiRNA Regulation of DNA Damage Response',\n",
       "       'Angiopoietin-like Protein 8 Regulatory Pathway', 'DNA Damage',\n",
       "       'Wnt / Hedgehog / Notch',\n",
       "       'Cell Cycle Role of 14-3-3 Proteins in Cell Cycle Regulation',\n",
       "       'Signaling By VEGF', 'Cell Cycle Checkpoints',\n",
       "       'Development EGFR Signaling Pathway', 'CNTF Signaling',\n",
       "       'MTOR Signaling Pathway', 'Senescence and Autophagy in Cancer',\n",
       "       'Development Dopamine D2 Receptor Transactivation of EGFR',\n",
       "       'HIF1Alpha Pathway', 'Integrin Pathway',\n",
       "       'Nanog in Mammalian ESC Pluripotency',\n",
       "       'Processing of Capped Intron-Containing Pre-mRNA',\n",
       "       'Translation Insulin Regulation of Translation',\n",
       "       'Development Ligand-independent Activation of ESR1 and ESR2',\n",
       "       'Colorectal Cancer Metastasis', 'Autophagy Pathway',\n",
       "       'Pre-NOTCH Expression and Processing', 'Cell Cycle',\n",
       "       'Serotonin HTR1 Group and FOS Pathway', 'GDNF Signaling',\n",
       "       'Integrated Breast Cancer Pathway', 'Deubiquitination',\n",
       "       'Development IGF-1 Receptor Signaling',\n",
       "       'DYRK1A Involvement Regarding Cell Proliferation in Brain Development',\n",
       "       'Semaphorin Interactions', 'Head and Neck Squamous Cell Carcinoma',\n",
       "       'Translational Control', 'MAPK Signaling: Oxidative Stress',\n",
       "       'CCR3 Pathway in Eosinophils',\n",
       "       'DREAM Repression and Dynorphin Expression',\n",
       "       'Small Cell Lung Cancer', 'TGF-beta Signaling Pathways',\n",
       "       'Signaling By WNT', 'Chks in Checkpoint Regulation',\n",
       "       'MAPK Signaling: Mitogens',\n",
       "       'Transport of The SLBP Independent Mature MRNA',\n",
       "       'NcRNAs Involved in Wnt Signaling in Hepatocellular Carcinoma',\n",
       "       'ADORA2B Mediated Anti-inflammatory Cytokines Production',\n",
       "       'Microtubule Cytoskeleton Regulation', 'VEGF Signaling Pathway',\n",
       "       'Nuclear Events (kinase and Transcription Factor Activation)',\n",
       "       'PDGFR-beta Signaling Pathway', 'Signaling By NTRKs',\n",
       "       'Development FGFR Signaling Pathway', 'Gene Silencing By RNA',\n",
       "       'Sudden Infant Death Syndrome (SIDS) Susceptibility Pathways',\n",
       "       'MAP Kinase Activation', 'AMP-activated Protein Kinase Signaling',\n",
       "       'HCMV Infection', 'Gastrin Signaling Pathway',\n",
       "       'TCR Signaling (Qiagen)', 'P53 Transcriptional Gene Network',\n",
       "       'Epithelial to Mesenchymal Transition in Colorectal Cancer',\n",
       "       'Signaling Events Mediated By VEGFR1 and VEGFR2',\n",
       "       'Glucose / Energy Metabolism', 'Adipogenesis',\n",
       "       'CCR5 Pathway in Macrophages',\n",
       "       'MyD88 Dependent Cascade Initiated on Endosome',\n",
       "       'Nuclear Receptors Meta-pathway',\n",
       "       'Interleukin-4 and Interleukin-13 Signaling',\n",
       "       'IL-9 Signaling Pathways',\n",
       "       'Corticotropin-releasing Hormone Signaling Pathway',\n",
       "       'Circadian Clock', 'Hepatocyte Growth Factor Receptor Signaling',\n",
       "       'Smith-Magenis and Potocki-Lupski Syndrome Copy Number Variation',\n",
       "       'TNFR1 Pathway',\n",
       "       'Transcription Androgen Receptor Nuclear Signaling',\n",
       "       'HIF-1-alpha Transcription Factor Network',\n",
       "       'Intrinsic Pathway For Apoptosis', 'VEGFA-VEGFR2 Signaling',\n",
       "       'MNAR-PELP1 and Estrogen Receptor Interaction',\n",
       "       'Cyclins and Cell Cycle Regulation',\n",
       "       'Mammary Gland Development Pathway - Pregnancy and Lactation (Stage 3 of 4)',\n",
       "       'Activation of CAMP-Dependent PKA',\n",
       "       'Death Receptor Signaling Pathway', 'Packaging Of Telomere Ends',\n",
       "       'Beta-Adrenergic Signaling', 'Metabolism',\n",
       "       'Inhibition of Signaling By Overexpressed EGFR',\n",
       "       'Clear Cell Renal Cell Carcinoma Pathways',\n",
       "       'Oncogene Induced Senescence', 'Signaling By ALK in Cancer',\n",
       "       'DNA Double Strand Break Response',\n",
       "       'Thyroid Stimulating Hormone (TSH) Signaling Pathway',\n",
       "       'Development Endothelin-1/EDNRA Transactivation of EGFR',\n",
       "       'RUNX1 Interacts With Co-factors Whose Precise Effect on RUNX1 Targets Is Not Known',\n",
       "       'Constitutive Signaling By AKT1 E17K in Cancer',\n",
       "       'AP-1 Transcription Factor Network', 'Apoptosis and Autophagy',\n",
       "       'IGF1 Pathway', 'PAK Pathway',\n",
       "       'Regulation of RUNX1 Expression and Activity',\n",
       "       'Non-integrin Membrane-ECM Interactions',\n",
       "       'Mitotic G1 Phase and G1/S Transition',\n",
       "       'Signaling By TGFB Family Members',\n",
       "       'Signaling Events Mediated By HDAC Class I',\n",
       "       'Activation of The Pre-replicative Complex',\n",
       "       'Apoptosis and Survival BAD Phosphorylation', 'Bladder Cancer',\n",
       "       'Development_TGF-beta Receptor Signaling',\n",
       "       'Unfolded Protein Response',\n",
       "       'Respiratory Syncytial Virus Infection Pathway', 'FMLP Pathway',\n",
       "       'NF-kappaB Pathway', 'FOXM1 Transcription Factor Network',\n",
       "       'Programmed Cell Death', 'IL2 Signaling Events Mediated By STAT5',\n",
       "       'Apoptosis-related Network Due to Altered Notch3 in Ovarian Cancer',\n",
       "       'PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling',\n",
       "       'Role of Hypoxia, Angiogenesis, and FGF Pathway in OA Chondrocyte Hypertrophy',\n",
       "       'B Cell Receptor Signaling Pathway',\n",
       "       'Photodynamic Therapy-induced HIF-1 Survival Signaling',\n",
       "       'G12-G13 in Cellular Signaling',\n",
       "       'Immune Response IL-2 Activation and Signaling Pathway',\n",
       "       'G-AlphaQ Signaling',\n",
       "       'NRP1-triggered Signaling Pathways in Pancreatic Cancer',\n",
       "       'Angiotensin II Receptor Type 1 Pathway',\n",
       "       'Lissencephaly Gene (LIS1) in Neuronal Migration and Development',\n",
       "       'Exercise-induced Circadian Regulation',\n",
       "       'Immune Response_IL-6 Signaling Pathway',\n",
       "       'Notch Signaling Pathways', 'Jak-Stat Signaling Pathway',\n",
       "       'Plasma Membrane Estrogen Receptor Signaling',\n",
       "       'VEGFR3 Signaling in Lymphatic Endothelium',\n",
       "       'Physico-chemical Features and Toxicity-associated Pathways',\n",
       "       'Regulation of Actin Cytoskeleton', 'Signaling By PTK6',\n",
       "       'IL-1 Family Signaling Pathways',\n",
       "       'PPARA Activates Gene Expression',\n",
       "       'EML4 and NUDC in Mitotic Spindle Formation',\n",
       "       'DNA Damage Response (only ATM Dependent)',\n",
       "       'Metabolic States and Circadian Oscillators',\n",
       "       'Genotoxicity Pathway',\n",
       "       'Disorders of Folate Metabolism and Transport',\n",
       "       'P38 MAPK Signaling Pathway (WikiPathways)',\n",
       "       'Type 2 Papillary Renal Cell Carcinoma',\n",
       "       'E2F Transcription Factor Network', '4-1BB Pathway',\n",
       "       'Fc-GammaR Pathway', 'P53 Pathway', 'P75(NTR)-mediated Signaling',\n",
       "       'Modulators of TCR Signaling and T Cell Activation',\n",
       "       'Transcriptional Regulation By E2F6', 'RHOBTB GTPase Cycle',\n",
       "       'PDGFR-alpha Signaling Pathway', 'SARS-CoV-1-host Interactions',\n",
       "       'Overlap Between Signal Transduction Pathways Contributing to LMNA Laminopathies',\n",
       "       'ATM Pathway', 'Adhesion',\n",
       "       'G-protein Signaling G-Protein Alpha-i Signaling Cascades',\n",
       "       'P53 Signaling', 'Guidance Cues and Growth Cone Motility',\n",
       "       'Regulation of Actin Dynamics For Phagocytic Cup Formation',\n",
       "       'C-MYB Transcription Factor Network', 'Integrins in Angiogenesis',\n",
       "       'Ubiquitination Cascade'], dtype=object)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc['SuperPath Name'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Category</th>\n",
       "      <th>Subcategory</th>\n",
       "      <th>Enrichment</th>\n",
       "      <th>P-value</th>\n",
       "      <th>P-adjusted</th>\n",
       "      <th>Q-value</th>\n",
       "      <th>Observed</th>\n",
       "      <th>miRNAs/precursors</th>\n",
       "      <th>Score</th>\n",
       "      <th>SuperPath Name</th>\n",
       "      <th># SuperPath Total Genes</th>\n",
       "      <th># SuperPath Matched Genes</th>\n",
       "      <th>Matched Genes (Symbols)</th>\n",
       "      <th>Evidence URL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Category, Subcategory, Enrichment, P-value, P-adjusted, Q-value, Observed, miRNAs/precursors, Score, SuperPath Name, # SuperPath Total Genes, # SuperPath Matched Genes, Matched Genes (Symbols), Evidence URL]\n",
       "Index: []"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlap = miEAA.merge(gc, left_on='Subcategory', right_on='SuperPath Name')\n",
    "overlap.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "myenv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
