{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Category', 'Subcategory', 'Enrichment', 'P-value', 'P-adjusted',\n",
      "       'Q-value', 'Observed', 'miRNAs/precursors'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "miEAA = pd.read_csv('./s1_s2_mieAA.csv')\n",
    "print(miEAA.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Annotations derived over miRTarBase (Gene Ontology)' 'Pathways (KEGG)'\n",
      " 'Target genes (miRTarBase)' 'Expressed in tissue (Tissue Atlas)'\n",
      " 'GO Molecular function (miRPathDB)']\n"
     ]
    }
   ],
   "source": [
    "# Print unique categories\n",
    "print(miEAA['Category'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Score', 'SuperPath Name', '# SuperPath Total Genes',\n",
       "       '# SuperPath Matched Genes', 'Matched Genes (Symbols)', 'Evidence URL'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc = pd.read_csv('./s1_s2_gc.csv')\n",
    "gc.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Category</th>\n",
       "      <th>Subcategory</th>\n",
       "      <th>Enrichment</th>\n",
       "      <th>P-value</th>\n",
       "      <th>P-adjusted</th>\n",
       "      <th>Q-value</th>\n",
       "      <th>Observed</th>\n",
       "      <th>miRNAs/precursors</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>transcription corepressor binding GO0001222</td>\n",
       "      <td>depleted</td>\n",
       "      <td>0.000333</td>\n",
       "      <td>0.000333</td>\n",
       "      <td>0.488152</td>\n",
       "      <td>5</td>\n",
       "      <td>hsa-miR-4472; hsa-miR-139-3p; hsa-miR-4647; hs...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>alpha-tubulin binding GO0043014</td>\n",
       "      <td>depleted</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>0.488152</td>\n",
       "      <td>6</td>\n",
       "      <td>hsa-miR-6769b-5p; hsa-miR-4472; hsa-miR-139-3p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>beta-tubulin binding GO0048487</td>\n",
       "      <td>depleted</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>0.488152</td>\n",
       "      <td>6</td>\n",
       "      <td>hsa-miR-6769b-5p; hsa-miR-4472; hsa-miR-139-3p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>positive regulation of GTPase activity GO0043547</td>\n",
       "      <td>depleted</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>0.488152</td>\n",
       "      <td>6</td>\n",
       "      <td>hsa-miR-6895-3p; hsa-miR-4472; hsa-miR-139-3p;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Pathways (KEGG)</td>\n",
       "      <td>Glioma</td>\n",
       "      <td>depleted</td>\n",
       "      <td>0.006527</td>\n",
       "      <td>0.006527</td>\n",
       "      <td>0.515387</td>\n",
       "      <td>8</td>\n",
       "      <td>hsa-miR-548h-5p; hsa-miR-937-5p; hsa-miR-4439;...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Category  \\\n",
       "0  Annotations derived over miRTarBase (Gene Onto...   \n",
       "1  Annotations derived over miRTarBase (Gene Onto...   \n",
       "2  Annotations derived over miRTarBase (Gene Onto...   \n",
       "3  Annotations derived over miRTarBase (Gene Onto...   \n",
       "4                                    Pathways (KEGG)   \n",
       "\n",
       "                                        Subcategory Enrichment   P-value  \\\n",
       "0       transcription corepressor binding GO0001222   depleted  0.000333   \n",
       "1                   alpha-tubulin binding GO0043014   depleted  0.002997   \n",
       "2                    beta-tubulin binding GO0048487   depleted  0.002997   \n",
       "3  positive regulation of GTPase activity GO0043547   depleted  0.002997   \n",
       "4                                            Glioma   depleted  0.006527   \n",
       "\n",
       "   P-adjusted   Q-value  Observed  \\\n",
       "0    0.000333  0.488152         5   \n",
       "1    0.002997  0.488152         6   \n",
       "2    0.002997  0.488152         6   \n",
       "3    0.002997  0.488152         6   \n",
       "4    0.006527  0.515387         8   \n",
       "\n",
       "                                   miRNAs/precursors  \n",
       "0  hsa-miR-4472; hsa-miR-139-3p; hsa-miR-4647; hs...  \n",
       "1  hsa-miR-6769b-5p; hsa-miR-4472; hsa-miR-139-3p...  \n",
       "2  hsa-miR-6769b-5p; hsa-miR-4472; hsa-miR-139-3p...  \n",
       "3  hsa-miR-6895-3p; hsa-miR-4472; hsa-miR-139-3p;...  \n",
       "4  hsa-miR-548h-5p; hsa-miR-937-5p; hsa-miR-4439;...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miEAA.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Score</th>\n",
       "      <th>SuperPath Name</th>\n",
       "      <th># SuperPath Total Genes</th>\n",
       "      <th># SuperPath Matched Genes</th>\n",
       "      <th>Matched Genes (Symbols)</th>\n",
       "      <th>Evidence URL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>49.83</td>\n",
       "      <td>Gene Expression (Transcription)</td>\n",
       "      <td>1543</td>\n",
       "      <td>161</td>\n",
       "      <td>LRPPRC, ZNF154, HAND2, POLR1B, DR1, ZNF699, ZN...</td>\n",
       "      <td>https://pathcards.genecards.org/card/gene_expr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>33.16</td>\n",
       "      <td>Infectious Disease</td>\n",
       "      <td>1818</td>\n",
       "      <td>163</td>\n",
       "      <td>SLC5A5, MUC17, MAP2K1, MAP2K3, MAP2K4, SLC12A6...</td>\n",
       "      <td>https://pathcards.genecards.org/card/infectiou...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>32.47</td>\n",
       "      <td>Regulation of Activated PAK-2p34 By Proteasome...</td>\n",
       "      <td>943</td>\n",
       "      <td>100</td>\n",
       "      <td>MAP2K4, HAND2, MAP2K7, INO80B, BRD2, TAB1, B2M...</td>\n",
       "      <td>https://pathcards.genecards.org/card/regulatio...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>31.17</td>\n",
       "      <td>Signal Transduction</td>\n",
       "      <td>2597</td>\n",
       "      <td>212</td>\n",
       "      <td>MAP2K1, ARL4C, NHS, ADM, NCKAP1, STAT5A, AXL, ...</td>\n",
       "      <td>https://pathcards.genecards.org/card/signal_tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>30.74</td>\n",
       "      <td>Cellular Responses to Stimuli</td>\n",
       "      <td>819</td>\n",
       "      <td>89</td>\n",
       "      <td>MAP2K3, CAPZA1, CAPZA2, RAI1, SZT2, LRPPRC, MA...</td>\n",
       "      <td>https://pathcards.genecards.org/card/cellular_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Score                                     SuperPath Name  \\\n",
       "0  49.83                    Gene Expression (Transcription)   \n",
       "1  33.16                                 Infectious Disease   \n",
       "2  32.47  Regulation of Activated PAK-2p34 By Proteasome...   \n",
       "3  31.17                                Signal Transduction   \n",
       "4  30.74                      Cellular Responses to Stimuli   \n",
       "\n",
       "   # SuperPath Total Genes  # SuperPath Matched Genes  \\\n",
       "0                     1543                        161   \n",
       "1                     1818                        163   \n",
       "2                      943                        100   \n",
       "3                     2597                        212   \n",
       "4                      819                         89   \n",
       "\n",
       "                             Matched Genes (Symbols)  \\\n",
       "0  LRPPRC, ZNF154, HAND2, POLR1B, DR1, ZNF699, ZN...   \n",
       "1  SLC5A5, MUC17, MAP2K1, MAP2K3, MAP2K4, SLC12A6...   \n",
       "2  MAP2K4, HAND2, MAP2K7, INO80B, BRD2, TAB1, B2M...   \n",
       "3  MAP2K1, ARL4C, NHS, ADM, NCKAP1, STAT5A, AXL, ...   \n",
       "4  MAP2K3, CAPZA1, CAPZA2, RAI1, SZT2, LRPPRC, MA...   \n",
       "\n",
       "                                        Evidence URL  \n",
       "0  https://pathcards.genecards.org/card/gene_expr...  \n",
       "1  https://pathcards.genecards.org/card/infectiou...  \n",
       "2  https://pathcards.genecards.org/card/regulatio...  \n",
       "3  https://pathcards.genecards.org/card/signal_tr...  \n",
       "4  https://pathcards.genecards.org/card/cellular_...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Annotations derived over miRTarBase (Gene Ontology)' 'Pathways (KEGG)'\n",
      " 'Target genes (miRTarBase)' 'Expressed in tissue (Tissue Atlas)'\n",
      " 'GO Molecular function (miRPathDB)']\n",
      "Category\n",
      "Annotations derived over miRTarBase (Gene Ontology)    141\n",
      "Expressed in tissue (Tissue Atlas)                      31\n",
      "Pathways (KEGG)                                         22\n",
      "Target genes (miRTarBase)                                7\n",
      "GO Molecular function (miRPathDB)                        1\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# print the categories from Reactome\n",
    "print(miEAA['Category'].unique())\n",
    "print(miEAA['Category'].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['transcription corepressor binding GO0001222',\n",
       "       'alpha-tubulin binding GO0043014',\n",
       "       'beta-tubulin binding GO0048487',\n",
       "       'positive regulation of GTPase activity GO0043547', 'Glioma',\n",
       "       'Parkinson disease',\n",
       "       'extracellular matrix structural constituent GO0005201',\n",
       "       'G-protein beta-subunit binding GO0031681',\n",
       "       'L-amino acid transmembrane transporter activity GO0015179',\n",
       "       'L-leucine transmembrane transporter activity GO0015190',\n",
       "       'branching morphogenesis of an epithelial tube GO0048754',\n",
       "       'cytoplasmic sequestering of transcription factor GO0042994',\n",
       "       'embryonic placenta development GO0001892',\n",
       "       'epidermal growth factor receptor signaling pathway GO0007173',\n",
       "       'isoleucine transport GO0015818', 'leucine transport GO0015820',\n",
       "       'methionine transport GO0015821',\n",
       "       'neutral L-amino acid transmembrane transporter activity GO0015175',\n",
       "       'phenylalanine transport GO0015823', 'response to UV-C GO0010225',\n",
       "       'G protein-coupled receptor internalization GO0002031',\n",
       "       'Grb2-EGFR complex GO0070436',\n",
       "       'I-kappaB phosphorylation GO0007252',\n",
       "       'L-histidine transport GO1902024',\n",
       "       'L-leucine import across plasma membrane GO1903801',\n",
       "       'L-tryptophan transmembrane transport GO1904556',\n",
       "       'L-tryptophan transmembrane transporter activity GO0015196',\n",
       "       'adrenomedullin receptor binding GO0031700',\n",
       "       'adrenomedullin receptor signaling pathway GO1990410',\n",
       "       'amino acid transport complex GO1990184',\n",
       "       'amylin receptor signaling pathway GO0097647',\n",
       "       'aromatic amino acid transmembrane transporter activity GO0015173',\n",
       "       'cellular response to exogenous dsRNA GO0071360',\n",
       "       'external side of apical plasma membrane GO0098591',\n",
       "       'insulin receptor substrate binding GO0043560',\n",
       "       'leucine import across plasma membrane GO0098713',\n",
       "       'negative regulation of vasoconstriction GO0045906',\n",
       "       'neural crest cell migration GO0001755',\n",
       "       'phosphatase activity GO0016791',\n",
       "       'positive regulation of cytokine production involved in immune response GO0002720',\n",
       "       'positive regulation of heart rate GO0010460',\n",
       "       'positive regulation of interleukin-4 production GO0032753',\n",
       "       'positive regulation of progesterone biosynthetic process GO2000184',\n",
       "       'positive regulation of tumor necrosis factor production GO0032760',\n",
       "       'positive regulation of vasoconstriction GO0045907',\n",
       "       'regulation of systemic arterial blood pressure GO0003073',\n",
       "       'regulation of urine volume GO0035809',\n",
       "       'thyroid hormone transmembrane transporter activity GO0015349',\n",
       "       'thyroid hormone transport GO0070327',\n",
       "       'tryptophan transport GO0015827', 'tyrosine transport GO0015828',\n",
       "       'valine transport GO0015829', 'xenobiotic transport GO0042908',\n",
       "       'Mineral absorption', 'Proximal tubule bicarbonate reclamation',\n",
       "       'ADM', 'C12orf49', 'GRB2', 'SLC7A5',\n",
       "       'phosphatidylinositol biosynthetic process GO0006661',\n",
       "       'response to lipopolysaccharide GO0032496',\n",
       "       'Inositol phosphate metabolism',\n",
       "       'protein-folding chaperone binding GO0051087',\n",
       "       'protein sumoylation GO0016925', 'Phagosome',\n",
       "       'negative regulation of protein kinase activity GO0006469',\n",
       "       'Hepatocellular carcinoma', 'adipocyte', 'arachnoid mater',\n",
       "       'artery', 'bladder', 'bone', 'brain', 'colon', 'dura mater',\n",
       "       'epididymis', 'esophagus', 'fascia', 'gallbladder', 'liver',\n",
       "       'lung', 'lymph node', 'muscle', 'myocardium', 'nerve', 'pancreas',\n",
       "       'pleura', 'prostate', 'skin', 'spinal cord', 'spleen', 'testis',\n",
       "       'thyroid', 'tunica albuginea', 'vein',\n",
       "       'intrinsic apoptotic signaling pathway in response to endoplasmic reticulum stress GO0070059',\n",
       "       'MHC class I peptide loading complex GO0042824',\n",
       "       'U4U6 x U5 tri-snRNP complex GO0046540', 'U4 snRNP GO0005687',\n",
       "       'negative regulation of amyloid-beta formation GO1902430',\n",
       "       'negative regulation of trophoblast cell migration GO1901164',\n",
       "       'nuclear androgen receptor binding GO0050681',\n",
       "       'signal complex assembly GO0007172',\n",
       "       'spermatid nucleus differentiation GO0007289',\n",
       "       'structural constituent of muscle GO0008307',\n",
       "       'Mucin type O-glycan biosynthesis',\n",
       "       'organic cyclic compound binding',\n",
       "       'Central carbon metabolism in cancer',\n",
       "       'Signaling pathways regulating pluripotency of stem cells',\n",
       "       'kidney', 'small intestine', 'stomach', 'ESCEZ complex GO0035098',\n",
       "       'L-glutamine transmembrane transporter activity GO0015186',\n",
       "       'acetyl-CoA carboxylase activity GO0003989',\n",
       "       'acetyl-CoA metabolic process GO0006084',\n",
       "       'adipose tissue development GO0060612',\n",
       "       'branching involved in ureteric bud morphogenesis GO0001658',\n",
       "       'chromosome organization GO0051276',\n",
       "       'fatty-acyl-CoA biosynthetic process GO0046949',\n",
       "       'glutamine transport GO0006868',\n",
       "       'membrane repolarization GO0086009',\n",
       "       'mitochondrial translational elongation GO0070125',\n",
       "       'nuclear steroid receptor activity GO0003707',\n",
       "       'positive regulation of cell-matrix adhesion GO0001954',\n",
       "       'positive regulation of chemokine C-X-C motif ligand 1 production GO2000340',\n",
       "       'pre-miRNA binding GO0070883',\n",
       "       'protein localization to basolateral plasma membrane GO1903361',\n",
       "       'regulation of cellular response to stress GO0080135',\n",
       "       'regulation of defense response to virus by host GO0050691',\n",
       "       'translation factor activity, RNA binding GO0008135',\n",
       "       'Fatty acid biosynthesis', 'ACACA', 'NUCB1', 'ZNF347',\n",
       "       'basal plasma membrane GO0009925',\n",
       "       'cyclin-dependent protein kinase holoenzyme complex GO0000307',\n",
       "       'fat cell differentiation GO0045444', 'filopodium GO0030175',\n",
       "       'peptide antigen binding GO0042605',\n",
       "       'phagocytic vesicle membrane GO0030670',\n",
       "       'phosphotyrosine residue binding GO0001784',\n",
       "       'response to virus GO0009615',\n",
       "       'unfolded protein binding GO0051082',\n",
       "       'Retrograde endocannabinoid signaling', 'p53 binding GO0002039',\n",
       "       'site of double-strand break GO0035861',\n",
       "       'small GTPase binding GO0031267', 'JAK-STAT signaling pathway',\n",
       "       'Neuroactive ligand-receptor interaction',\n",
       "       'Phospholipase D signaling pathway',\n",
       "       'DNA-binding transcription activator activity GO0001216',\n",
       "       'RNA polymerase II transcription regulatory region sequence-specific DNA binding GO0000977',\n",
       "       'cilium GO0005929',\n",
       "       'positive regulation of peptidyl-tyrosine phosphorylation GO0050731',\n",
       "       'cytoplasmic vesicle GO0031410', 'endocytic recycling GO0032456',\n",
       "       'growth cone GO0030426',\n",
       "       'multicellular organism development GO0007275',\n",
       "       'peptidyl-threonine phosphorylation GO0018107',\n",
       "       'substrate adhesion-dependent cell spreading GO0034446',\n",
       "       'Adrenergic signaling in cardiomyocytes', 'ATP binding GO0005524',\n",
       "       'DNA damage response GO0006974', 'bone development GO0060348',\n",
       "       'calcium ion import across plasma membrane GO0098703',\n",
       "       'cell surface GO0009986',\n",
       "       'intracellular signal transduction GO0035556',\n",
       "       'nuclear speck GO0016607',\n",
       "       'peptidyl-prolyl cis-trans isomerase activity GO0003755',\n",
       "       'positive regulation of cell migration GO0030335',\n",
       "       'positive regulation of gene expression GO0010628',\n",
       "       'protein stabilization GO0050821',\n",
       "       'regulation of cytokinesis GO0032465',\n",
       "       'regulation of filopodium assembly GO0051489', 'Breast cancer',\n",
       "       'Human T-cell leukemia virus 1 infection',\n",
       "       'PI3K-Akt signaling pathway', 'Shigellosis',\n",
       "       'MLL34 complex GO0044666', 'cell projection GO0042995',\n",
       "       'chromosome, telomeric region GO0000781',\n",
       "       'phosphatidylinositol-4,5-bisphosphate binding GO0005546',\n",
       "       'Chemokine signaling pathway',\n",
       "       'amino acid transmembrane transporter activity GO0015171',\n",
       "       'cytosolic large ribosomal subunit GO0022625',\n",
       "       'four-way junction DNA binding GO0000400',\n",
       "       'in utero embryonic development GO0001701',\n",
       "       'microvillus membrane GO0031528',\n",
       "       'mitotic spindle assembly GO0090307',\n",
       "       'negative regulation of vascular endothelial growth factor receptor signaling pathway GO0030948',\n",
       "       'neutral amino acid transport GO0015804',\n",
       "       'phosphatidylinositol dephosphorylation GO0046856',\n",
       "       'plasma membrane repair GO0001778',\n",
       "       'positive regulation of neutrophil chemotaxis GO0090023',\n",
       "       'positive regulation of stress fiber assembly GO0051496',\n",
       "       'regulation of cell growth GO0001558',\n",
       "       'regulation of endocytosis GO0030100',\n",
       "       'signal transduction in response to DNA damage GO0042770',\n",
       "       'spindle assembly GO0051225', 'vasculogenesis GO0001570',\n",
       "       'Phosphatidylinositol signaling system'], dtype=object)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# find the overlap of the overlap of supepath name and subcategory\n",
    "miEAA['Subcategory'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Gene Expression (Transcription)', 'Infectious Disease',\n",
       "       'Regulation of Activated PAK-2p34 By Proteasome Mediated Degradation',\n",
       "       'Signal Transduction', 'Cellular Responses to Stimuli',\n",
       "       'MAP Kinase Signaling', 'Metabolism of Proteins', 'DNA Damage',\n",
       "       'Pleural Mesothelioma', 'IL-17 Family Signaling Pathways',\n",
       "       'RNA Polymerase I Promoter Opening', 'Toll Comparative Pathway',\n",
       "       'Cellular Senescence', 'MAPK Signaling: Oxidative Stress',\n",
       "       'Signaling By Rho GTPases',\n",
       "       'Signaling By Receptor Tyrosine Kinases', 'TCR Signaling (Qiagen)',\n",
       "       'Regulation of TP53 Activity', 'Signaling By TGFB Family Members',\n",
       "       'Sertoli-Sertoli Cell Junction Dynamics',\n",
       "       'Tacrolimus/Cyclosporine Pathway, Pharmacodynamics',\n",
       "       'IL-1 Family Signaling Pathways', 'GPCR Pathway',\n",
       "       'MAPK Family Signaling Cascades',\n",
       "       'Assembly of The Pre-replicative Complex',\n",
       "       'Regulation of Actin Cytoskeleton', 'Cell Cycle Checkpoints',\n",
       "       'TGF-Beta Pathway',\n",
       "       'Class I MHC Mediated Antigen Processing and Presentation',\n",
       "       'Chromatin Organization', 'Toll-Like Receptor Signaling Pathways',\n",
       "       'Cytoskeletal Signaling', 'Homology Directed Repair',\n",
       "       'PDGFR-beta Signaling Pathway', 'Innate Immune System',\n",
       "       'Cell Cycle, Mitotic', 'Cell Cycle', 'CNTF Signaling',\n",
       "       'Chromatin Regulation / Acetylation',\n",
       "       'Apoptotic Pathways in Synovial Fibroblasts',\n",
       "       'Cell Cycle Role of 14-3-3 Proteins in Cell Cycle Regulation',\n",
       "       'Toll-like Receptor Signaling Pathway',\n",
       "       'Transcription_P53 Signaling Pathway',\n",
       "       'Transcriptional Regulation By The AP-2 (TFAP2) Family of Transcription Factors',\n",
       "       'Ovarian Tumor Domain Proteases',\n",
       "       'Immune Response IL-2 Activation and Signaling Pathway',\n",
       "       'Ubiquitination Cascade', '5q35 Copy Number Variation',\n",
       "       'TGF-beta Signaling Pathways', 'PKMTs Methylate Histone Lysines',\n",
       "       'Development_Glucocorticoid Receptor Signaling', 'Akt Signaling',\n",
       "       'AP-1 Transcription Factor Network',\n",
       "       'Transcriptional Activity of SMAD2/SMAD3:SMAD4 Heterotrimer',\n",
       "       'Validated Targets of C-MYC Transcriptional Activation',\n",
       "       'Signaling Events Mediated By Hepatocyte Growth Factor Receptor (c-Met)',\n",
       "       'Activation of The Pre-replicative Complex',\n",
       "       'Processing of Capped Intron-Containing Pre-mRNA',\n",
       "       'SARS-CoV-2 Infection', 'ERK Signaling', 'MAPK Signaling Pathway',\n",
       "       'Tyrosine Kinases / Adaptors', 'Deubiquitination',\n",
       "       'Downstream Signaling in Na√Øve CD8+ T Cells',\n",
       "       'Disorders of Folate Metabolism and Transport',\n",
       "       'Signaling Events Mediated By HDAC Class I', 'Endometrial Cancer',\n",
       "       'MTOR Signaling Pathway', 'PtdIns(4,5)P2 in Cytokinesis Pathway',\n",
       "       'EPH-Ephrin Signaling', 'Glucose / Energy Metabolism',\n",
       "       'Prolactin Signaling', 'P53 Pathway', 'ESR-mediated Signaling',\n",
       "       'Immune Response_IL-6 Signaling Pathway',\n",
       "       'Cell Cycle Regulation of G1/S Transition (part 1)',\n",
       "       'Signaling By ALK in Cancer',\n",
       "       'Corticotropin-releasing Hormone Signaling Pathway',\n",
       "       'CD28 Signaling in T-Helper Cell',\n",
       "       'Calcineurin-regulated NFAT-dependent Transcription in Lymphocytes',\n",
       "       'Packaging Of Telomere Ends', 'Integrated Breast Cancer Pathway',\n",
       "       'VEGF Signaling Pathway',\n",
       "       'Transport of Inorganic Cations/anions and Amino Acids/oligopeptides',\n",
       "       'MAP Kinase Activation', 'Development_TGF-beta Receptor Signaling',\n",
       "       '3q29 Copy Number Variation Syndrome', 'Chromosome Maintenance',\n",
       "       'Nervous System Development'], dtype=object)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc['SuperPath Name'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Category</th>\n",
       "      <th>Subcategory</th>\n",
       "      <th>Enrichment</th>\n",
       "      <th>P-value</th>\n",
       "      <th>P-adjusted</th>\n",
       "      <th>Q-value</th>\n",
       "      <th>Observed</th>\n",
       "      <th>miRNAs/precursors</th>\n",
       "      <th>Score</th>\n",
       "      <th>SuperPath Name</th>\n",
       "      <th># SuperPath Total Genes</th>\n",
       "      <th># SuperPath Matched Genes</th>\n",
       "      <th>Matched Genes (Symbols)</th>\n",
       "      <th>Evidence URL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Category, Subcategory, Enrichment, P-value, P-adjusted, Q-value, Observed, miRNAs/precursors, Score, SuperPath Name, # SuperPath Total Genes, # SuperPath Matched Genes, Matched Genes (Symbols), Evidence URL]\n",
       "Index: []"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlap = miEAA.merge(gc, left_on='Subcategory', right_on='SuperPath Name')\n",
    "overlap.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "myenv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
