{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: './ctl_s1_miEAA.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[2], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m miEAA \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m./ctl_s1_miEAA.csv\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28mprint\u001b[39m(miEAA\u001b[38;5;241m.\u001b[39mcolumns)\n",
      "File \u001b[0;32m~/miniconda3/envs/myenv/lib/python3.10/site-packages/pandas/io/parsers/readers.py:1026\u001b[0m, in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[1;32m   1013\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[1;32m   1014\u001b[0m     dialect,\n\u001b[1;32m   1015\u001b[0m     delimiter,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   1022\u001b[0m     dtype_backend\u001b[38;5;241m=\u001b[39mdtype_backend,\n\u001b[1;32m   1023\u001b[0m )\n\u001b[1;32m   1024\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[0;32m-> 1026\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/myenv/lib/python3.10/site-packages/pandas/io/parsers/readers.py:620\u001b[0m, in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    617\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[1;32m    619\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[0;32m--> 620\u001b[0m parser \u001b[38;5;241m=\u001b[39m \u001b[43mTextFileReader\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    622\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[1;32m    623\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[0;32m~/miniconda3/envs/myenv/lib/python3.10/site-packages/pandas/io/parsers/readers.py:1620\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m   1617\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m   1619\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m-> 1620\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/myenv/lib/python3.10/site-packages/pandas/io/parsers/readers.py:1880\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[0;34m(self, f, engine)\u001b[0m\n\u001b[1;32m   1878\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode:\n\u001b[1;32m   1879\u001b[0m         mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m-> 1880\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1881\u001b[0m \u001b[43m    \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1882\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1883\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1884\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcompression\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1885\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmemory_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmemory_map\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1886\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_text\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_text\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1887\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding_errors\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstrict\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1888\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstorage_options\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1889\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1890\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m   1891\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[0;32m~/miniconda3/envs/myenv/lib/python3.10/site-packages/pandas/io/common.py:873\u001b[0m, in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    868\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[1;32m    869\u001b[0m     \u001b[38;5;66;03m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[1;32m    870\u001b[0m     \u001b[38;5;66;03m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[1;32m    871\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mencoding \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mmode:\n\u001b[1;32m    872\u001b[0m         \u001b[38;5;66;03m# Encoding\u001b[39;00m\n\u001b[0;32m--> 873\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[1;32m    874\u001b[0m \u001b[43m            \u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    875\u001b[0m \u001b[43m            \u001b[49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    876\u001b[0m \u001b[43m            \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    877\u001b[0m \u001b[43m            \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    878\u001b[0m \u001b[43m            \u001b[49m\u001b[43mnewline\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    879\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    880\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    881\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[1;32m    882\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: './ctl_s1_miEAA.csv'"
     ]
    }
   ],
   "source": [
    "miEAA = pd.read_csv('./ctl_s1_miEAA.csv')\n",
    "print(miEAA.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Annotations derived over miRTarBase (Gene Ontology)'\n",
      " 'Target genes (miRTarBase)' 'Pathways (KEGG)' 'Immune cells'\n",
      " 'Gene Ontology (miRWalk)' 'WikiPathways (miRPathDB)' 'Diseases (MNDR)'\n",
      " 'Published Diseases' 'Drugs (SM2miR)' 'Pathways (miRWalk)'\n",
      " 'Interactions (NPInter)' 'Reactome (miRPathDB)'\n",
      " 'Tissue specific (isomiRdb)' 'GO Biological process (miRPathDB)'\n",
      " 'KEGG (miRPathDB)' 'GO Molecular function (miRPathDB)'\n",
      " 'Localization (RNALocate)' 'Chromosomal location (miRBase)']\n"
     ]
    }
   ],
   "source": [
    "# Print unique categories\n",
    "print(miEAA['Category'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Score', 'SuperPath Name', '# SuperPath Total Genes',\n",
       "       '# SuperPath Matched Genes', 'Matched Genes (Symbols)', 'Evidence URL'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc = pd.read_csv('./ctl_s1_gc.csv')\n",
    "gc.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Category</th>\n",
       "      <th>Subcategory</th>\n",
       "      <th>Enrichment</th>\n",
       "      <th>P-value</th>\n",
       "      <th>P-adjusted</th>\n",
       "      <th>Q-value</th>\n",
       "      <th>Observed</th>\n",
       "      <th>miRNAs/precursors</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>cytokine activity GO0005125</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0.361506</td>\n",
       "      <td>13</td>\n",
       "      <td>hsa-miR-125a-3p; hsa-miR-6802-5p; hsa-miR-1307...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>monocyte chemotaxis GO0002548</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0.361506</td>\n",
       "      <td>3</td>\n",
       "      <td>hsa-miR-125a-3p; hsa-miR-1307-3p; hsa-miR-6729-5p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>positive regulation of cytoskeleton organizati...</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0.361506</td>\n",
       "      <td>3</td>\n",
       "      <td>hsa-miR-6861-5p; hsa-miR-125a-3p; hsa-miR-6729-5p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>triglyceride metabolic process GO0006641</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0.361506</td>\n",
       "      <td>3</td>\n",
       "      <td>hsa-miR-125a-3p; hsa-miR-1307-3p; hsa-miR-6729-5p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>regulation of DNA-binding transcription factor...</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.000824</td>\n",
       "      <td>0.000824</td>\n",
       "      <td>0.361506</td>\n",
       "      <td>8</td>\n",
       "      <td>hsa-miR-125a-3p; hsa-miR-1307-3p; hsa-miR-6729...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Category  \\\n",
       "0  Annotations derived over miRTarBase (Gene Onto...   \n",
       "1  Annotations derived over miRTarBase (Gene Onto...   \n",
       "2  Annotations derived over miRTarBase (Gene Onto...   \n",
       "3  Annotations derived over miRTarBase (Gene Onto...   \n",
       "4  Annotations derived over miRTarBase (Gene Onto...   \n",
       "\n",
       "                                         Subcategory Enrichment   P-value  \\\n",
       "0                        cytokine activity GO0005125   enriched  0.000064   \n",
       "1                      monocyte chemotaxis GO0002548   enriched  0.000705   \n",
       "2  positive regulation of cytoskeleton organizati...   enriched  0.000705   \n",
       "3           triglyceride metabolic process GO0006641   enriched  0.000705   \n",
       "4  regulation of DNA-binding transcription factor...   enriched  0.000824   \n",
       "\n",
       "   P-adjusted   Q-value  Observed  \\\n",
       "0    0.000064  0.361506        13   \n",
       "1    0.000705  0.361506         3   \n",
       "2    0.000705  0.361506         3   \n",
       "3    0.000705  0.361506         3   \n",
       "4    0.000824  0.361506         8   \n",
       "\n",
       "                                   miRNAs/precursors  \n",
       "0  hsa-miR-125a-3p; hsa-miR-6802-5p; hsa-miR-1307...  \n",
       "1  hsa-miR-125a-3p; hsa-miR-1307-3p; hsa-miR-6729-5p  \n",
       "2  hsa-miR-6861-5p; hsa-miR-125a-3p; hsa-miR-6729-5p  \n",
       "3  hsa-miR-125a-3p; hsa-miR-1307-3p; hsa-miR-6729-5p  \n",
       "4  hsa-miR-125a-3p; hsa-miR-1307-3p; hsa-miR-6729...  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miEAA.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Score</th>\n",
       "      <th>SuperPath Name</th>\n",
       "      <th># SuperPath Total Genes</th>\n",
       "      <th># SuperPath Matched Genes</th>\n",
       "      <th>Matched Genes (Symbols)</th>\n",
       "      <th>Evidence URL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Signal Transduction</td>\n",
       "      <td>2597</td>\n",
       "      <td>559</td>\n",
       "      <td>MUC20, ARL2, MAP2K2, DGKH, ARL4C, PTK6, NHS, C...</td>\n",
       "      <td>https://pathcards.genecards.org/card/signal_tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Metabolism of Proteins</td>\n",
       "      <td>2077</td>\n",
       "      <td>494</td>\n",
       "      <td>CANX, EIF2B5, MUC19, MUC20, RPS19, ARL2, GAN, ...</td>\n",
       "      <td>https://pathcards.genecards.org/card/metabolis...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Infectious Disease</td>\n",
       "      <td>1818</td>\n",
       "      <td>455</td>\n",
       "      <td>CANX, MUC19, MUC20, RPS19, MAP2K2, MAP2K3, CAP...</td>\n",
       "      <td>https://pathcards.genecards.org/card/infectiou...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Innate Immune System</td>\n",
       "      <td>2054</td>\n",
       "      <td>441</td>\n",
       "      <td>CANX, MUC19, MUC20, MAP2K3, GAN, EIF2S3, SLC11...</td>\n",
       "      <td>https://pathcards.genecards.org/card/innate_im...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Gene Expression (Transcription)</td>\n",
       "      <td>1543</td>\n",
       "      <td>428</td>\n",
       "      <td>NPM1, CARM1, NUDT21, THRA, ZNF154, ZNF155, AXI...</td>\n",
       "      <td>https://pathcards.genecards.org/card/gene_expr...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Score                   SuperPath Name  # SuperPath Total Genes  \\\n",
       "0  166.1              Signal Transduction                     2597   \n",
       "1  166.1           Metabolism of Proteins                     2077   \n",
       "2  166.1               Infectious Disease                     1818   \n",
       "3  166.1             Innate Immune System                     2054   \n",
       "4  166.1  Gene Expression (Transcription)                     1543   \n",
       "\n",
       "   # SuperPath Matched Genes  \\\n",
       "0                        559   \n",
       "1                        494   \n",
       "2                        455   \n",
       "3                        441   \n",
       "4                        428   \n",
       "\n",
       "                             Matched Genes (Symbols)  \\\n",
       "0  MUC20, ARL2, MAP2K2, DGKH, ARL4C, PTK6, NHS, C...   \n",
       "1  CANX, EIF2B5, MUC19, MUC20, RPS19, ARL2, GAN, ...   \n",
       "2  CANX, MUC19, MUC20, RPS19, MAP2K2, MAP2K3, CAP...   \n",
       "3  CANX, MUC19, MUC20, MAP2K3, GAN, EIF2S3, SLC11...   \n",
       "4  NPM1, CARM1, NUDT21, THRA, ZNF154, ZNF155, AXI...   \n",
       "\n",
       "                                        Evidence URL  \n",
       "0  https://pathcards.genecards.org/card/signal_tr...  \n",
       "1  https://pathcards.genecards.org/card/metabolis...  \n",
       "2  https://pathcards.genecards.org/card/infectiou...  \n",
       "3  https://pathcards.genecards.org/card/innate_im...  \n",
       "4  https://pathcards.genecards.org/card/gene_expr...  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Annotations derived over miRTarBase (Gene Ontology)'\n",
      " 'Target genes (miRTarBase)' 'Pathways (KEGG)' 'Immune cells'\n",
      " 'Gene Ontology (miRWalk)' 'WikiPathways (miRPathDB)' 'Diseases (MNDR)'\n",
      " 'Published Diseases' 'Drugs (SM2miR)' 'Pathways (miRWalk)'\n",
      " 'Interactions (NPInter)' 'Reactome (miRPathDB)'\n",
      " 'Tissue specific (isomiRdb)' 'GO Biological process (miRPathDB)'\n",
      " 'KEGG (miRPathDB)' 'GO Molecular function (miRPathDB)'\n",
      " 'Localization (RNALocate)' 'Chromosomal location (miRBase)']\n",
      "Category\n",
      "Annotations derived over miRTarBase (Gene Ontology)    619\n",
      "Target genes (miRTarBase)                              151\n",
      "Diseases (MNDR)                                         41\n",
      "Gene Ontology (miRWalk)                                 32\n",
      "Pathways (KEGG)                                         21\n",
      "Interactions (NPInter)                                  16\n",
      "WikiPathways (miRPathDB)                                10\n",
      "KEGG (miRPathDB)                                         6\n",
      "Reactome (miRPathDB)                                     5\n",
      "Published Diseases                                       4\n",
      "Immune cells                                             3\n",
      "Pathways (miRWalk)                                       2\n",
      "Drugs (SM2miR)                                           2\n",
      "Tissue specific (isomiRdb)                               1\n",
      "GO Biological process (miRPathDB)                        1\n",
      "GO Molecular function (miRPathDB)                        1\n",
      "Localization (RNALocate)                                 1\n",
      "Chromosomal location (miRBase)                           1\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# print the categories from Reactome\n",
    "print(miEAA['Category'].unique())\n",
    "print(miEAA['Category'].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['cytokine activity GO0005125', 'monocyte chemotaxis GO0002548',\n",
       "       'positive regulation of cytoskeleton organization GO0051495',\n",
       "       'triglyceride metabolic process GO0006641',\n",
       "       'regulation of DNA-binding transcription factor activity GO0051090',\n",
       "       'inositol hexakisphosphate 5-kinase activity GO0000832',\n",
       "       'mannosyl-oligosaccharide 1,2-alpha-mannosidase activity GO0004571',\n",
       "       'photoreceptor cell morphogenesis GO0008594',\n",
       "       'white fat cell differentiation GO0050872',\n",
       "       'RNA polymerase II transcription regulator complex GO0090575',\n",
       "       'tRNA pseudouridine synthesis GO0031119', 'ZKSCAN1',\n",
       "       'G2M transition of mitotic cell cycle GO0000086',\n",
       "       'positive regulation of G2M transition of mitotic cell cycle GO0010971',\n",
       "       'RNA polymerase III general transcription initiation factor binding GO0001025',\n",
       "       'high-density lipoprotein particle remodeling GO0034375',\n",
       "       'negative regulation of NLRP3 inflammasome complex assembly GO1900226',\n",
       "       'Fatty acid elongation', 'DDX19B', 'ZBTB43',\n",
       "       'cellular response to hydrogen peroxide GO0070301',\n",
       "       'integrin-mediated signaling pathway GO0007229',\n",
       "       'DNA replication GO0006260',\n",
       "       'cytoplasmic side of lysosomal membrane GO0098574',\n",
       "       'CD15 expressed', 'cytosolic small ribosomal subunit GO0022627',\n",
       "       'kinetochore GO0000776', 'mannosyltransferase activity GO0000030',\n",
       "       'protein methyltransferase activity GO0008276',\n",
       "       'ribosome binding GO0043022',\n",
       "       'cyclin-dependent protein kinase activity GO0097472',\n",
       "       'histone kinase activity GO0035173',\n",
       "       'GO0003700 sequence specific dna binding transcription factor activity',\n",
       "       'GO0005654 nucleoplasm', 'G1 to S cell cycle control',\n",
       "       'positive regulation of NLRP3 inflammasome complex assembly GO1900227',\n",
       "       'endocrine pancreas development GO0031018',\n",
       "       'GO0005524 atp binding',\n",
       "       'positive regulation of cytokine production GO0001819',\n",
       "       'Cholangiocarcinoma', 'TRAMP complex GO0031499',\n",
       "       'ncRNA processing GO0034470', 'snRNA catabolic process GO0016076',\n",
       "       'ZCCHC8', 'dilated cardiomyopathy deregulated',\n",
       "       'dilated cardiomyopathy upregulated', 'Trastuzumab',\n",
       "       'Integrated Cancer Pathway',\n",
       "       'cellular response to type II interferon GO0071346',\n",
       "       'chemoattractant activity GO0042056',\n",
       "       'Golgi ribbon formation GO0090161',\n",
       "       'N-acetylneuraminate metabolic process GO0006054',\n",
       "       'N-acylmannosamine kinase activity GO0009384',\n",
       "       'Shu complex GO0097196', 'T cell homeostasis GO0043029',\n",
       "       'UDP-N-acetylglucosamine 2-epimerase activity GO0008761',\n",
       "       'antigen processing and presentation of peptide antigen via MHC class I GO0002474',\n",
       "       'chemokine receptor activity GO0004950',\n",
       "       'cyclin D2-CDK4 complex GO0097129',\n",
       "       'cyclin D3-CDK4 complex GO0097130', 'cytolytic granule GO0044194',\n",
       "       'hepatic immune response GO0002384',\n",
       "       'immune system process GO0002376',\n",
       "       'interleukin-6 receptor complex GO0005896',\n",
       "       'long-chain fatty-acyl-CoA biosynthetic process GO0035338',\n",
       "       'mesenchymal cell development GO0014031',\n",
       "       'muscle system process GO0003012',\n",
       "       'positive regulation of activation of Janus kinase activity GO0010536',\n",
       "       'positive regulation of anterograde synaptic vesicle transport GO1903744',\n",
       "       'regulation of MyD88-dependent toll-like receptor signaling pathway GO0034124',\n",
       "       'regulation of cellular respiration GO0043457',\n",
       "       'sialylation GO0097503',\n",
       "       'type II interferon-mediated signaling pathway GO0060333',\n",
       "       'ARHGEF39', 'BORCS5', 'CDK4', 'CENPBD1', 'GNE', 'IRF1', 'MELK',\n",
       "       'MRPL36', 'SWSAP1', 'TRIM72', 'WDR31', 'ZNF430', 'ZNF439',\n",
       "       'GO0000166 nucleotide binding', 'GO0000922 spindle pole',\n",
       "       'GO0005085 guanyl nucleotide exchange factor activity',\n",
       "       'GO0016032 viral reproduction',\n",
       "       'GO0030529 ribonucleoprotein complex',\n",
       "       'GO0042802 identical protein binding',\n",
       "       'GO0043123 positive regulation of i kappab kinase nf kappab cascade',\n",
       "       'GO0060170 cilium membrane', 'hsa05110 Vibrio cholerae infection',\n",
       "       'Pancreatic adenocarcinoma', 'LIN28', 'Rbm20', 'TRIP6', 'Measles',\n",
       "       'mitotic spindle assembly checkpoint signaling GO0007094',\n",
       "       'L-cysteine desulfurase complex GO1990221',\n",
       "       'ferroxidase activity GO0004322',\n",
       "       'iron chaperone activity GO0034986',\n",
       "       'iron incorporation into metallo-sulfur cluster GO0018283',\n",
       "       'positive regulation of lyase activity GO0051349',\n",
       "       'positive regulation of succinate dehydrogenase activity GO1904231',\n",
       "       'protein autoprocessing GO0016540',\n",
       "       'regulation of ferrochelatase activity GO0010722',\n",
       "       'response to iron ion GO0010039', 'FXN', 'Cell Cycle',\n",
       "       'amyloid-beta formation GO0034205',\n",
       "       'amyloid precursor protein metabolic process GO0042982',\n",
       "       'gamma-secretase complex GO0070765',\n",
       "       'vascular endothelial growth factor production GO0010573',\n",
       "       'POLR2E', 'helicase activity GO0004386',\n",
       "       'positive regulation of osteoblast differentiation GO0045669',\n",
       "       'Cajal body GO0015030',\n",
       "       'ubiquitin-dependent protein catabolic process via the C-end degron rule pathway GO0140627',\n",
       "       'GAIT complex GO0097452', 'MHC class I protein binding GO0042288',\n",
       "       'WD40-repeat domain binding GO0071987',\n",
       "       'apoptotic DNA fragmentation GO0006309',\n",
       "       'dendritic cell chemotaxis GO0002407',\n",
       "       'positive regulation of chemokine C-X-C motif ligand 2 production GO2000343',\n",
       "       'positive regulation of monocyte chemotaxis GO0090026', 'SUSD6',\n",
       "       'Glioma', 'transcription by RNA polymerase III GO0006383',\n",
       "       'positive regulation of vascular endothelial growth factor production GO0010575',\n",
       "       'negative regulation of protein phosphorylation GO0001933',\n",
       "       'Systemic lupus erythematosus', 'BORC complex GO0099078',\n",
       "       'K48-linked deubiquitinase activity GO1990380',\n",
       "       'anterograde neuronal dense core vesicle transport GO1990048',\n",
       "       'organelle transport along microtubule GO0072384',\n",
       "       'regulation of endosome size GO0051036',\n",
       "       'regulation of lysosome size GO0062196',\n",
       "       'retrograde neuronal dense core vesicle transport GO1990049',\n",
       "       'synaptic vesicle transport GO0048489', 'Basal-like breast cancer',\n",
       "       'Breast cancer luminal', 'microtubule organizing center GO0005815',\n",
       "       'brush border membrane GO0031526',\n",
       "       'cellular response to interferon-beta GO0035458',\n",
       "       'complement activation GO0006956',\n",
       "       'negative regulation of glucose import GO0046325',\n",
       "       'reverse cholesterol transport GO0043691', 'PEA15',\n",
       "       'C5L2 anaphylatoxin chemotactic receptor binding GO0031715',\n",
       "       'GMP biosynthetic process GO0006177',\n",
       "       'axonemal microtubule GO0005879',\n",
       "       'branchiomotor neuron axon guidance GO0021785',\n",
       "       'chitobiosyldiphosphodolichol beta-mannosyltransferase activity GO0004578',\n",
       "       'chylomicron GO0042627',\n",
       "       'complement-mediated synapse pruning GO0150062',\n",
       "       'de novo AMP biosynthetic process GO0044208',\n",
       "       'de novo IMP biosynthetic process GO0006189',\n",
       "       'de novo XMP biosynthetic process GO0097294',\n",
       "       'detection of chemical stimulus involved in sensory perception of bitter taste GO0001580',\n",
       "       'facial nerve structural organization GO0021612',\n",
       "       'intraciliary anterograde transport GO0035720',\n",
       "       'large ribosomal subunit GO0015934',\n",
       "       'negative regulation of formation of translation preinitiation complex GO1901194',\n",
       "       'neuron projection guidance GO0097485',\n",
       "       'nuclease activity GO0004518',\n",
       "       'palmitoyltransferase activity GO0016409',\n",
       "       'positive regulation of G protein-coupled receptor signaling pathway GO0045745',\n",
       "       'positive regulation of apoptotic cell clearance GO2000427',\n",
       "       'positive regulation of glucose transmembrane transport GO0010828',\n",
       "       'protein palmitoylation GO0018345',\n",
       "       'pyramidal neuron development GO0021860',\n",
       "       'transforming growth factor beta ligand-receptor complex GO0070021',\n",
       "       'tricuspid valve morphogenesis GO0003186',\n",
       "       'trigeminal nerve structural organization GO0021637',\n",
       "       'vertebrate eye-specific patterning GO0150064',\n",
       "       'very-low-density lipoprotein particle assembly GO0034379', 'ALG1',\n",
       "       'C3', 'MRPS10', 'PLXNA3', 'PXMP4', 'RPL13A', 'TRUB2', 'ZNF101',\n",
       "       'ZNF43', 'Mitotic G1-G1S phases',\n",
       "       'roof of mouth development GO0060021',\n",
       "       'endothelial cell migration GO0043542',\n",
       "       'retrograde transport, endosome to Golgi GO0042147',\n",
       "       'type I interferon-mediated signaling pathway GO0060337',\n",
       "       'response to organic substance GO0010033',\n",
       "       'glucuronosyltransferase activity GO0015020',\n",
       "       'monoatomic ion homeostasis GO0050801',\n",
       "       'cyclin-dependent protein serinethreonine kinase activity GO0004693',\n",
       "       'proteasome-mediated ubiquitin-dependent protein catabolic process GO0043161',\n",
       "       'NFAT protein binding GO0051525',\n",
       "       'aminopeptidase activity GO0004177',\n",
       "       'histone H2A monoubiquitination GO0035518',\n",
       "       'positive regulation of DNA ligation GO0051106',\n",
       "       'positive regulation of dendritic cell cytokine production GO0002732',\n",
       "       'positive regulation of neuroinflammatory response GO0150078',\n",
       "       'positive regulation of reactive oxygen species biosynthetic process GO1903428',\n",
       "       'protein localization to kinetochore GO0034501',\n",
       "       'regulation of innate immune response GO0045088',\n",
       "       'telomerase activity GO0003720',\n",
       "       'mitotic metaphase plate congression GO0007080',\n",
       "       'intracellular protein transport GO0006886',\n",
       "       'regulation of ATP-dependent activity GO0043462',\n",
       "       'defense response to Gram-negative bacterium GO0050829',\n",
       "       'late endosome GO0005770', 'PML body organization GO0030578',\n",
       "       'SUMO binding GO0032183',\n",
       "       'Wnt signaling pathway, calcium modulating pathway GO0007223',\n",
       "       'activin receptor activity, type II GO0016362',\n",
       "       'bub1-bub3 complex GO1990298',\n",
       "       'cellular response to nitric oxide GO0071732',\n",
       "       'centriole replication GO0007099',\n",
       "       'lactoylglutathione lyase activity GO0004462',\n",
       "       'mitotic checkpoint complex GO0033597',\n",
       "       'neuronal ribonucleoprotein granule GO0071598',\n",
       "       'positive regulation of B cell activation GO0050871',\n",
       "       'positive regulation of activin receptor signaling pathway GO0032927',\n",
       "       'positive regulation of heterochromatin formation GO0031453',\n",
       "       'protein K33-linked ubiquitination GO1990390',\n",
       "       'protein localization to nucleoplasm GO1990173',\n",
       "       'serine phosphorylation of STAT protein GO0042501',\n",
       "       'tRNA modification GO0006400',\n",
       "       'transmembrane receptor protein serinethreonine kinase signaling pathway GO0007178',\n",
       "       'ACVR2B', 'AREL1', 'BUB3', 'C11orf1', 'CDK2', 'EIF4EBP2', 'GLO1',\n",
       "       'LEFTY1', 'NLK', 'PUS3', 'SERBP1', 'TBRG1',\n",
       "       'Regulation of signaling by NODAL', 'Signaling by NODAL',\n",
       "       'Mesodermal Commitment Pathway', 'aorta morphogenesis GO0035909',\n",
       "       'chondroitin sulfate biosynthetic process GO0030206',\n",
       "       'inorganic cation transmembrane transporter activity GO0022890',\n",
       "       'secondary palate development GO0062009',\n",
       "       'chemical synaptic transmission GO0007268',\n",
       "       'Estrogen-receptor negative breast cancer',\n",
       "       'Estrogen-receptor positive breast cancer', 'Lung cancer',\n",
       "       'Progesterone-receptor positive breast cancer',\n",
       "       'viral release from host cell GO0019076', 'ZSWIM1',\n",
       "       'negative regulation of DNA binding GO0043392',\n",
       "       'Glycerolipid metabolism',\n",
       "       'Cul4-RING E3 ubiquitin ligase complex GO0080008',\n",
       "       'heme biosynthetic process GO0006783', 'Osteosarcoma',\n",
       "       'Prostate cancer', 'Lymphoma', 'Melanoma',\n",
       "       'Rho GDP-dissociation inhibitor activity GO0005094',\n",
       "       'translation repressor activity GO0030371', 'ARHGDIA',\n",
       "       '10-formyltetrahydrofolate biosynthetic process GO0009257',\n",
       "       'Derlin-1 retrotranslocation complex GO0036513',\n",
       "       'embryonic neurocranium morphogenesis GO0048702',\n",
       "       'embryonic viscerocranium morphogenesis GO0048703',\n",
       "       'forebrain development GO0030900',\n",
       "       'formate-tetrahydrofolate ligase activity GO0004329',\n",
       "       'morphogenesis of an epithelium GO0002009',\n",
       "       'negative regulation of lipid catabolic process GO0050995',\n",
       "       'Breast invasive carcinoma', 'Carcinoma ductal breast',\n",
       "       'Gastric adenocarcinoma', 'Multiple myeloma',\n",
       "       'Oral squamous cell carcinoma', 'Ovarian clear cell carcinoma',\n",
       "       'Pancreatic ductal adenocarcinoma', 'Prostate adenocarcinoma',\n",
       "       'Stomach adenocarcinoma', 'semen specific',\n",
       "       'positive regulation of focal adhesion assembly GO0051894',\n",
       "       'translation initiation factor activity GO0003743',\n",
       "       'T cell receptor signaling pathway GO0050852',\n",
       "       'BMP signaling pathway GO0030509',\n",
       "       'aortic valve morphogenesis GO0003180',\n",
       "       'regulation of Rho protein signal transduction GO0035023',\n",
       "       '2-acylglycerol O-acyltransferase activity GO0003846',\n",
       "       '4-alpha-hydroxytetrahydrobiopterin dehydratase activity GO0008124',\n",
       "       'ESCEZ complex GO0035098', 'GMP 5-nucleotidase activity GO0050484',\n",
       "       'GMP metabolic process GO0046037',\n",
       "       'IMP 5-nucleotidase activity GO0050483',\n",
       "       'IMP catabolic process GO0006204',\n",
       "       'IMP metabolic process GO0046040',\n",
       "       'RNA lariat debranching enzyme activator activity GO0061632',\n",
       "       'TRC complex GO0072380', 'Toll-like receptor 2 binding GO0035663',\n",
       "       'acetylcholine binding GO0042166',\n",
       "       'actin-mediated cell contraction GO0070252',\n",
       "       'adenosine metabolic process GO0046085',\n",
       "       'allantoin metabolic process GO0000255',\n",
       "       'alpha-mannosidase activity GO0004559',\n",
       "       'caveola assembly GO0070836',\n",
       "       'cell projection organization GO0030030',\n",
       "       'cobalamin binding GO0031419',\n",
       "       'cobalamin metabolic process GO0009235',\n",
       "       'corrinoid adenosyltransferase activity GO0008817',\n",
       "       'dGMP metabolic process GO0046054',\n",
       "       'diacylglycerol O-acyltransferase activity GO0004144',\n",
       "       'embryonic process involved in female pregnancy GO0060136',\n",
       "       'fatty acid beta-oxidation using acyl-CoA dehydrogenase GO0033539',\n",
       "       'filamin binding GO0031005', 'glycolipid binding GO0051861',\n",
       "       'glycolipid transfer activity GO0017089',\n",
       "       'growth hormone receptor signaling pathway GO0060396',\n",
       "       'heart formation GO0060914', 'heme export GO0097037',\n",
       "       'heme transport GO0015886',\n",
       "       'isovaleryl-CoA dehydrogenase activity GO0008470',\n",
       "       'leucine catabolic process GO0006552', 'lipase activity GO0016298',\n",
       "       'lipid catabolic process GO0016042',\n",
       "       'membrane raft assembly GO0001765',\n",
       "       'mitochondrial RNA 5-end processing GO0000964',\n",
       "       'mitochondrion morphogenesis GO0070584',\n",
       "       'muscle alpha-actinin binding GO0051371',\n",
       "       'muscle structure development GO0061061',\n",
       "       'natural killer cell proliferation GO0001787',\n",
       "       'negative regulation of hydrogen peroxide biosynthetic process GO0010730',\n",
       "       'negative regulation of lymphangiogenesis GO1901491',\n",
       "       'negative regulation of very-low-density lipoprotein particle clearance GO0010916',\n",
       "       'neutrophil migration GO1990266',\n",
       "       'nucleoside phosphotransferase activity GO0050146',\n",
       "       'phospholipase A1 activity GO0008970',\n",
       "       'photoreceptor outer segment membrane GO0042622',\n",
       "       'plasma membrane raft assembly GO0044854',\n",
       "       'positive regulation of cardiac epithelial to mesenchymal transition GO0062043',\n",
       "       'positive regulation of cholesterol transport GO0032376',\n",
       "       'positive regulation of cytokine production involved in inflammatory response GO1900017',\n",
       "       'positive regulation of high-density lipoprotein particle clearance GO0010983',\n",
       "       'protein localization involved in establishment of planar polarity GO0090251',\n",
       "       'regulation of cellular senescence GO2000772',\n",
       "       'regulation of endothelial cell migration GO0010594',\n",
       "       'regulation of glomerular filtration GO0003093',\n",
       "       'regulation of integrin activation GO0033623',\n",
       "       'regulation of kinase activity GO0043549',\n",
       "       'regulation of vasculogenesis GO2001212',\n",
       "       'ribosome assembly GO0042255',\n",
       "       'tRNA guanine-N1--methyltransferase activity GO0009019',\n",
       "       'tetraspanin-enriched microdomain GO0097197',\n",
       "       'thyroid hormone metabolic process GO0042403',\n",
       "       'toll-like receptor 4 signaling pathway GO0034142',\n",
       "       'toll-like receptor 7 signaling pathway GO0034154',\n",
       "       'toll-like receptor TLR1TLR2 signaling pathway GO0038123',\n",
       "       'transferase activity, transferring alkyl or aryl other than methyl groups GO0016765',\n",
       "       'unconventional myosin complex GO0016461',\n",
       "       'uropod membrane GO0031259',\n",
       "       'very long-chain fatty acid biosynthetic process GO0042761',\n",
       "       'visual behavior GO0007632', 'ABHD18', 'ARPC2', 'CHCHD5',\n",
       "       'CWF19L1', 'CYTIP', 'EIF2A', 'EMP2', 'ENAH', 'FAXC', 'FBLIM1',\n",
       "       'FBXL2', 'FLVCR1', 'GATAD1', 'GLTP', 'HAUS2', 'HSPE1-MOB4', 'IVD',\n",
       "       'KLB', 'LIPG', 'MMAB', 'MOB4', 'NPHP1', 'NT5C2', 'OMD', 'OSTF1',\n",
       "       'PBOV1', 'PCBD2', 'PDE6A', 'PDLIM3', 'POLR3A', 'SCIMP', 'SGTB',\n",
       "       'SS18L1', 'TMEM132B', 'TMEM154', 'TMIGD2', 'TTC31', 'WDR5B',\n",
       "       'ZNF576', 'RP11-108M9.3', 'RP11-160E2.6',\n",
       "       'platelet activation GO0030168', 'protein transport GO0015031',\n",
       "       'MAP kinase activity GO0004707',\n",
       "       'phosphatidylinositol 3-kinase signaling GO0014065',\n",
       "       'positive regulation of protein localization to centrosome GO1904781',\n",
       "       'syntaxin-1 binding GO0017075',\n",
       "       'regulation of ubiquitin-protein transferase activity GO0051438',\n",
       "       'stress fiber assembly GO0043149',\n",
       "       'endoplasmic reticulum to Golgi vesicle-mediated transport GO0006888',\n",
       "       '2Fe-2S cluster assembly GO0044571',\n",
       "       '4Fe-4S cluster assembly GO0044572',\n",
       "       'ATPase-coupled inorganic anion transmembrane transporter activity GO0043225',\n",
       "       'export across plasma membrane GO0140115',\n",
       "       'iron-sulfur cluster assembly complex GO1990229',\n",
       "       'mitochondrial iron-sulfur cluster assembly complex GO0099128',\n",
       "       'positive regulation of aconitate hydratase activity GO1904234',\n",
       "       'L-methionine salvage from methylthioadenosine GO0019509',\n",
       "       'cAMP response element binding GO0035497',\n",
       "       'dendrite morphogenesis GO0048813',\n",
       "       'negative regulation of peptidyl-serine dephosphorylation GO1902309',\n",
       "       'regulation of epithelial to mesenchymal transition GO0010717',\n",
       "       'small ribosomal subunit GO0015935',\n",
       "       'mitotic spindle assembly GO0090307',\n",
       "       'visual perception GO0007601', '9+0 non-motile cilium GO0097731',\n",
       "       'ATP metabolic process GO0046034', 'ATP transport GO0015867',\n",
       "       'EP4 subtype prostaglandin E2 receptor binding GO0031867',\n",
       "       'Golgi-associated vesicle membrane GO0030660',\n",
       "       'RNA decapping GO0110154', 'abscission GO0009838',\n",
       "       'acetyl-CoA carboxylase activity GO0003989',\n",
       "       'acetyl-CoA metabolic process GO0006084',\n",
       "       'adenosine 5-hexahydrogen pentaphosphate catabolic process GO1901911',\n",
       "       'anterior commissure morphogenesis GO0021960',\n",
       "       'bis5-adenosyl-hexaphosphatase activity GO0034431',\n",
       "       'bis5-adenosyl-pentaphosphatase activity GO0034432',\n",
       "       'chemorepulsion involved in embryonic olfactory bulb interneuron precursor migration GO0021834',\n",
       "       'club cell differentiation GO0060486',\n",
       "       'diadenosine hexaphosphate catabolic process GO1901909',\n",
       "       'diadenosine pentaphosphate catabolic process GO1901907',\n",
       "       'diadenosine polyphosphate catabolic process GO0015961',\n",
       "       'diphosphoinositol-polyphosphate diphosphatase activity GO0008486',\n",
       "       'diphosphoinositol polyphosphate catabolic process GO0071544',\n",
       "       'diphosphoinositol polyphosphate metabolic process GO0071543',\n",
       "       'endopolyphosphatase activity GO0000298',\n",
       "       'fatty-acyl-CoA biosynthetic process GO0046949',\n",
       "       'hindbrain development GO0030902',\n",
       "       'inositol diphosphate pentakisphosphate diphosphatase activity GO0052842',\n",
       "       'inositol diphosphate tetrakisphosphate diphosphatase activity GO0052840',\n",
       "       'interleukin-6-mediated signaling pathway GO0070102',\n",
       "       'lung ciliated cell differentiation GO0061141',\n",
       "       'mRNA 3-end processing GO0031124',\n",
       "       'mitochondrial transcription GO0006390',\n",
       "       'negative regulation of epithelial cell proliferation involved in lung morphogenesis GO2000795',\n",
       "       'negative regulation of mesenchymal cell proliferation involved in lung development GO2000791',\n",
       "       'principal sensory nucleus of trigeminal nerve development GO0021740',\n",
       "       'proteasome regulatory particle binding GO1904855',\n",
       "       'regulation of cellular response to stress GO0080135',\n",
       "       'regulation of proteasome assembly GO0090364',\n",
       "       'response to cold GO0009409',\n",
       "       'type II pneumocyte differentiation GO0060510',\n",
       "       'type I pneumocyte differentiation GO0060509', 'ACACA', 'APH1A',\n",
       "       'FEM1A', 'MAP2K3', 'MRPS16', 'NFIB', 'NUDT3', 'SPART', 'TMED4',\n",
       "       'TNRC18P2', 'UBLCP1', 'ZER1', 'ZNF251', 'ZNF764',\n",
       "       'positive regulation of cellular metabolic process',\n",
       "       'Central carbon metabolism in cancer',\n",
       "       'transport across blood-brain barrier GO0150104',\n",
       "       'acylglycerol lipase activity GO0047372',\n",
       "       'apoptotic cell clearance GO0043277',\n",
       "       'keratan sulfate biosynthetic process GO0018146',\n",
       "       'lens fiber cell morphogenesis GO0070309',\n",
       "       'mitotic sister chromatid cohesion, centromeric GO0071962',\n",
       "       'negative regulation of fat cell differentiation GO0045599',\n",
       "       'positive regulation of receptor signaling pathway via STAT GO1904894',\n",
       "       'Glycosaminoglycan biosynthesis - keratan sulfate',\n",
       "       'Cell Cycle Checkpoints',\n",
       "       'determination of leftright symmetry GO0007368',\n",
       "       'enzyme regulator activity GO0030234',\n",
       "       'toll-like receptor 3 signaling pathway GO0034138',\n",
       "       'Phototransduction', 'PHLDA3', 'RABGAP1L',\n",
       "       'blood microparticle GO0072562', 'T cell migration GO0072678',\n",
       "       'axon extension involved in axon guidance GO0048846',\n",
       "       'methylosome GO0034709',\n",
       "       'positive regulation of calcium ion transport GO0051928',\n",
       "       'positive regulation of interleukin-10 production GO0032733',\n",
       "       'ubiquitin binding GO0043130', 'Lysosome',\n",
       "       'cell migration GO0016477',\n",
       "       'inward rectifier potassium channel activity GO0005242',\n",
       "       '1-phosphatidylinositol-4-phosphate 3-kinase activity GO0035005',\n",
       "       'Wnt signalosome GO1990909',\n",
       "       'attachment of mitotic spindle microtubules to kinetochore GO0051315',\n",
       "       'cyclin B1-CDK1 complex GO0097125',\n",
       "       'phosphatidylinositol 3-kinase activity GO0035004',\n",
       "       'positive regulation of B cell receptor signaling pathway GO0050861',\n",
       "       'positive regulation of mitochondrial ATP synthesis coupled electron transport GO1905448',\n",
       "       'FYTTD1', 'TOR2A',\n",
       "       'GO0000122 negative regulation of transcription from rna polymerase ii promoter',\n",
       "       'GO0000226 microtubule cytoskeleton organization',\n",
       "       'GO0003690 double stranded dna binding',\n",
       "       'GO0005576 extracellular region',\n",
       "       'GO0005815 microtubule organizing center',\n",
       "       'GO0006915 apoptotic process', 'GO0007067 mitosis',\n",
       "       'GO0007173 epidermal growth factor receptor signaling pathway',\n",
       "       'GO0007569 cell aging', 'GO0008022 protein c terminus binding',\n",
       "       'GO0008286 insulin receptor signaling pathway',\n",
       "       'GO0008543 fibroblast growth factor receptor signaling pathway',\n",
       "       'GO0010467 gene expression',\n",
       "       'GO0048011 nerve growth factor receptor signaling pathway',\n",
       "       'P00031 Inflammation mediated by chemokine and cytokine signaling pathway',\n",
       "       'B cell lymphoma of malt type', 'Intellectual disability',\n",
       "       'FoxO signaling pathway', 'Pathways in cancer',\n",
       "       'Proteoglycans in cancer', 'p53 signaling pathway',\n",
       "       'Signal Transduction',\n",
       "       'Chromosomal and microsatellite instability in colorectal cancer',\n",
       "       'DNA Damage Response only ATM dependent',\n",
       "       'Integrin-mediated Cell Adhesion', 'Retinoblastoma Gene in Cancer',\n",
       "       'GGA3', 'LTBP4', 'NCKAP5L', 'PTGDS', 'SAP18', 'SEMA4C', 'SYN2',\n",
       "       'ZNF385A', 'cholesterol biosynthetic process GO0006695',\n",
       "       'ephrin receptor signaling pathway GO0048013',\n",
       "       'glucose metabolic process GO0006006',\n",
       "       'positive regulation of autophagy GO0010508',\n",
       "       'positive regulation of interleukin-6 production GO0032755',\n",
       "       'histone deacetylase binding GO0042826',\n",
       "       'CARD domain binding GO0050700',\n",
       "       'N-acetylglucosamine 6-O-sulfotransferase activity GO0001517',\n",
       "       'TIM22 mitochondrial import inner membrane insertion complex GO0042721',\n",
       "       'cyclin A2-CDK2 complex GO0097124',\n",
       "       'intracellular copper ion homeostasis GO0006878',\n",
       "       'negative regulation of interleukin-1 beta production GO0032691',\n",
       "       'negative regulation of phagocytosis GO0050765',\n",
       "       'photoreceptor disc membrane GO0097381',\n",
       "       'regulation of MAP kinase activity GO0043405',\n",
       "       'respiratory chain complex IV assembly GO0008535',\n",
       "       'ribosomal large subunit binding GO0043023',\n",
       "       'sulfur compound metabolic process GO0006790', 'JARID2', 'KIF1C',\n",
       "       'Cervical squamous cell carcinoma',\n",
       "       'High-grade serous ovarian carcinoma', 'Lung small cell carcinoma',\n",
       "       'Ovarian cancer', 'Ovarian carcinoma', 'Ovarian neoplasms',\n",
       "       'Small cell lung cancer', 'CD3 expressed', 'CD56 expressed',\n",
       "       '4-hydroxynonenal', 'Apoptosis Modulation and Signaling', 'ATPIF1',\n",
       "       'NADPH regeneration GO0006740', 'NADP metabolic process GO0006739',\n",
       "       'glucose-6-phosphate dehydrogenase activity GO0004345',\n",
       "       'negative regulation of protein glutathionylation GO0010734',\n",
       "       'pentose-phosphate shunt, oxidative branch GO0009051',\n",
       "       'pentose-phosphate shunt GO0006098',\n",
       "       'pentose biosynthetic process GO0019322',\n",
       "       'positive regulation of aspartic-type endopeptidase activity involved in amyloid precursor protein catabolic process GO1902961',\n",
       "       'regulation of cell adhesion mediated by integrin GO0033628',\n",
       "       'response to interferon-beta GO0035456',\n",
       "       'ribose phosphate biosynthetic process GO0046390', 'G6PD',\n",
       "       'transcription regulatory region DNA binding',\n",
       "       'Rho protein signal transduction GO0007266',\n",
       "       'cytoplasmic microtubule GO0005881',\n",
       "       'DNA-binding transcription factor activity GO0003700',\n",
       "       'phospholipid catabolic process GO0009395',\n",
       "       'cholesterol metabolic process GO0008203',\n",
       "       'metaphase plate congression GO0051310',\n",
       "       'retinoic acid receptor signaling pathway GO0048384',\n",
       "       'synaptic vesicle membrane GO0030672', 'Microvesicle',\n",
       "       'T cell proliferation GO0042098',\n",
       "       'lipopolysaccharide binding GO0001530',\n",
       "       'membrane insertase activity GO0032977',\n",
       "       'proteolysis involved in protein catabolic process GO0051603',\n",
       "       'protein-DNA complex GO0032993',\n",
       "       'regulation of gene expression GO0010468',\n",
       "       'double-stranded DNA binding GO0003690', 'Wnt signaling pathway',\n",
       "       'cell projection GO0042995', 'chloride channel activity GO0005254',\n",
       "       'negative regulation of cell size GO0045792',\n",
       "       'positive regulation of interleukin-12 production GO0032735',\n",
       "       'acute-phase response GO0006953', 'axon ensheathment GO0008366',\n",
       "       'intraciliary transport particle B GO0030992',\n",
       "       'negative regulation of intracellular estrogen receptor signaling pathway GO0033147',\n",
       "       'phosphatidylinositol-3-phosphate biosynthetic process GO0036092',\n",
       "       'GO0003677 dna binding', 'GO0005515 protein binding',\n",
       "       'GO0005634 nucleus', 'GO0005737 cytoplasm', 'GO0005829 cytosol',\n",
       "       'GO0005886 plasma membrane', 'Rheumatoid arthritis', 'Cell cycle',\n",
       "       'receptor ligand activity GO0048018', 'social behavior GO0035176',\n",
       "       'JAK pathway signal transduction adaptor activity GO0008269',\n",
       "       'NEDD8 ligase activity GO0061663',\n",
       "       'alanine transmembrane transporter activity GO0022858',\n",
       "       'cell cycle GO0007049',\n",
       "       'central nervous system vasculogenesis GO0022009',\n",
       "       'interleukin-22 receptor activity GO0042018',\n",
       "       'ligase activity GO0016874',\n",
       "       'multivesicular body membrane GO0032585',\n",
       "       'negative regulation of intrinsic apoptotic signaling pathway by p53 class mediator GO1902254',\n",
       "       'negative regulation of protein autophosphorylation GO0031953',\n",
       "       'regulation of glucose transmembrane transport GO0010827',\n",
       "       'taurine transmembrane transporter activity GO0005368',\n",
       "       'taurine transport GO0015734',\n",
       "       'transcription factor catabolic process GO0036369',\n",
       "       'type I interferon receptor activity GO0004905',\n",
       "       'Phenylalanine metabolism', 'Chromosome 12', 'ELMSAN1', 'IFNAR2',\n",
       "       'MDM2', 'SLC36A1', 'ZNF460', 'Breast neoplasms',\n",
       "       'DNA IR-Double Strand Breaks DSBs and cellular response via ATM',\n",
       "       'C-type lectin receptor signaling pathway',\n",
       "       'Osteoclast differentiation',\n",
       "       'negative regulation of vascular associated smooth muscle cell proliferation GO1904706',\n",
       "       'embryonic cranial skeleton morphogenesis GO0048701',\n",
       "       'Kidney renal papillary cell carcinoma',\n",
       "       'Papillary renal cell carcinoma',\n",
       "       'lysine-acetylated histone binding GO0070577',\n",
       "       'positive regulation of transforming growth factor beta receptor signaling pathway GO0030511',\n",
       "       'protein-folding chaperone binding GO0051087',\n",
       "       'molecular condensate scaffold activity GO0140693',\n",
       "       'mRNA splice site selection GO0006376',\n",
       "       'phospholipase activity GO0004620', 'tRNA binding GO0000049',\n",
       "       'transcription coregulator binding GO0001221',\n",
       "       'double-stranded RNA binding GO0003725',\n",
       "       'transport vesicle GO0030133', 'histone binding GO0042393',\n",
       "       'Breast cancer', 'Coronavirus disease - COVID-19',\n",
       "       '1-phosphatidylinositol-3-kinase activity GO0016303',\n",
       "       '5-nucleotidase activity GO0008253', 'BLOC-1 complex GO0031083',\n",
       "       'L-alanine transmembrane transporter activity GO0015180',\n",
       "       'L-alanine transport GO0015808',\n",
       "       'L-proline transmembrane transporter activity GO0015193',\n",
       "       'amino acidproton symporter activity GO0005280',\n",
       "       'anterograde synaptic vesicle transport GO0048490',\n",
       "       'branched-chain amino acid catabolic process GO0009083',\n",
       "       'cysteine-type endopeptidase activator activity involved in apoptotic process GO0008656',\n",
       "       'extrinsic component of endoplasmic reticulum membrane GO0042406',\n",
       "       'glutamate secretion GO0014047',\n",
       "       'glycine transmembrane transporter activity GO0015187',\n",
       "       'glycine transport GO0015816', 'lateral element GO0000800',\n",
       "       'mitochondrion-endoplasmic reticulum membrane tethering GO1990456',\n",
       "       'multi-pass translocon complex GO0160064',\n",
       "       'multi-pass transmembrane protein insertion into ER membrane GO0160063',\n",
       "       'negative regulation of beta-galactosidase activity GO1903770',\n",
       "       'negative regulation of exonuclease activity GO1905778',\n",
       "       'negative regulation of glucocorticoid receptor signaling pathway GO2000323',\n",
       "       'negative regulation of telomere capping GO1904354',\n",
       "       'negative regulation of telomere maintenance GO0032205',\n",
       "       'negative regulation of telomere maintenance via recombination GO0032208',\n",
       "       'negative regulation of telomere maintenance via semi-conservative replication GO0032214',\n",
       "       'negative regulation of telomere single strand break repair GO1903824',\n",
       "       'negative regulation of telomeric D-loop disassembly GO1905839',\n",
       "       'nuclear ubiquitin ligase complex GO0000152',\n",
       "       'positive regulation of chromosome separation GO1905820',\n",
       "       'positive regulation of melanin biosynthetic process GO0048023',\n",
       "       'proline transmembrane transport GO0035524',\n",
       "       'prolineproton symporter activity GO0005297',\n",
       "       'protection from non-homologous end joining at telomere GO0031848',\n",
       "       'regulation of double-strand break repair via homologous recombination GO0010569',\n",
       "       'regulation of telomere maintenance via telomerase GO0032210',\n",
       "       'shelterin complex GO0070187', 'steroid binding GO0005496',\n",
       "       'telomeric loop formation GO0031627',\n",
       "       'vacuolar membrane GO0005774', 'vesicle membrane GO0012506',\n",
       "       'DIRAS1', 'GIPC1', 'TERF2', 'TMCO1', 'TMEM170A',\n",
       "       'Rac protein signal transduction GO0016601',\n",
       "       'actin polymerization or depolymerization GO0008154',\n",
       "       'humoral immune response GO0006959',\n",
       "       'mitochondrial ribosome GO0005761', 'GO0005739 mitochondrion',\n",
       "       'Myopathy', 'Synovial sarcoma', 'colon cancer upregulated',\n",
       "       'renal cancer deregulated', 'apical part of cell GO0045177',\n",
       "       'Familial ovarian cancer', 'Familiar ovarian carcinoma',\n",
       "       'Her2-receptor positive breast cancer',\n",
       "       'Progesterone-receptor negative breast cancer',\n",
       "       'Hypertrophic cardiomyopathy',\n",
       "       'Cytokine-cytokine receptor interaction',\n",
       "       'neutrophil chemotaxis GO0030593', 'Thyroid hormone synthesis',\n",
       "       'lipid droplet GO0005811',\n",
       "       'nuclear retinoid X receptor binding GO0046965', 'Renin secretion',\n",
       "       '2-methylacyl-CoA dehydrogenase activity GO0003853',\n",
       "       'AP-1 adaptor complex binding GO0035650',\n",
       "       'AP-3 adaptor complex binding GO0035651',\n",
       "       'BAT3 complex binding GO1904288',\n",
       "       'BLOC-2 complex binding GO0036461',\n",
       "       'C-X-C chemokine binding GO0019958',\n",
       "       'DNA geometric change GO0032392',\n",
       "       'DNA topological change GO0006265',\n",
       "       'GPI-anchor transamidase complex GO0042765',\n",
       "       'N-acetylglucosaminyl-proteoglycan 4-beta-glucuronosyltransferase activity GO0050509',\n",
       "       'RAGE receptor binding GO0050786',\n",
       "       'T-helper 1 cell activation GO0035711',\n",
       "       'T-helper 1 cell differentiation GO0045063',\n",
       "       'VDJ recombination GO0033151',\n",
       "       'acyl-CoA dehydrogenase activity GO0003995',\n",
       "       'alphav-beta3 integrin-HMGB1 complex GO0035868',\n",
       "       'angiostatin binding GO0043532',\n",
       "       'cellular response to misfolded protein GO0071218',\n",
       "       'epithelial cilium movement involved in determination of leftright asymmetry GO0060287',\n",
       "       'error-free translesion synthesis GO0070987',\n",
       "       'glucuronosyl-N-acetylglucosaminyl-proteoglycan 4-alpha-N-acetylglucosaminyltransferase activity GO0050508',\n",
       "       'heparan sulfate N-acetylglucosaminyltransferase activity GO0042328',\n",
       "       'histone pre-mRNA 3end processing complex GO0071204',\n",
       "       'inorganic cation transmembrane transport GO0098662',\n",
       "       'isoleucine catabolic process GO0006550',\n",
       "       'kinesin I complex GO0016938',\n",
       "       'large ribosomal subunit rRNA binding GO0070180',\n",
       "       'metal ion transport GO0030001',\n",
       "       'mitochondrial proton-transporting ATP synthase, catalytic core GO0005754',\n",
       "       'mitochondrial proton-transporting ATP synthase complex GO0005753',\n",
       "       'myelin assembly GO0032288',\n",
       "       'myeloid dendritic cell activation GO0001773',\n",
       "       'negative regulation of apoptotic cell clearance GO2000426',\n",
       "       'neuroblast differentiation GO0014016',\n",
       "       'neutrophil clearance GO0097350',\n",
       "       'peptidyl-proline hydroxylation to 3-hydroxy-L-proline GO0018400',\n",
       "       'phospholipid efflux GO0033700',\n",
       "       'positive regulation of centriole elongation GO1903724',\n",
       "       'positive regulation of centrosome duplication GO0010825',\n",
       "       'positive regulation of cilium assembly GO0045724',\n",
       "       'positive regulation of dendritic cell differentiation GO2001200',\n",
       "       'positive regulation of establishment of protein localization GO1904951',\n",
       "       'positive regulation of microglial cell activation GO1903980',\n",
       "       'positive regulation of mismatch repair GO0032425',\n",
       "       'positive regulation of nucleic acid-templated transcription GO1903508',\n",
       "       'positive regulation of sodium ion transmembrane transporter activity GO2000651',\n",
       "       'positive regulation of toll-like receptor 9 signaling pathway GO0034165',\n",
       "       'protein insertion into mitochondrial membrane GO0051204',\n",
       "       'protein localization to cytosolic proteasome complex involved in ERAD pathway GO1904379',\n",
       "       'proton-transporting ATP synthase activity, rotational mechanism GO0046933',\n",
       "       'proton-transporting ATP synthase complex, catalytic core F1 GO0045261',\n",
       "       'proton-transporting ATP synthase complex GO0045259',\n",
       "       'proton motive force-driven ATP synthesis GO0015986',\n",
       "       'pyramidal neuron migration to cerebral cortex GO0021852',\n",
       "       'regulation of B cell proliferation GO0030888',\n",
       "       'regulation of SREBP signaling pathway GO2000638',\n",
       "       'regulation of T cell mediated immune response to tumor cell GO0002840',\n",
       "       'regulation of monoatomic ion transmembrane transport GO0034765',\n",
       "       'regulation of neuroblast proliferation GO1902692',\n",
       "       'regulation of protein sumoylation GO0033233',\n",
       "       'regulation of restriction endodeoxyribonuclease activity GO0032072',\n",
       "       'regulation of tolerance induction GO0002643',\n",
       "       'serine C-palmitoyltransferase activity GO0004758',\n",
       "       'serine C-palmitoyltransferase complex GO0017059',\n",
       "       'short-branched-chain-acyl-CoA dehydrogenase activity GO0016937',\n",
       "       'vacuolar proton-transporting V-type ATPase complex assembly GO0070072',\n",
       "       'ACADSB', 'AMFR', 'ARHGAP18', 'ATP1B4', 'ATP5A1', 'ATXN3',\n",
       "       'CCDC149', 'CDC5L', 'CDCP1', 'CEP120', 'COX18', 'CRTAP', 'DCAF8',\n",
       "       'DNAJC24', 'EGFLAM', 'EXT1', 'FAM19A3', 'HMGB1', 'ING1', 'KLC2',\n",
       "       'KLHDC10', 'LONRF2', 'PIGS', 'PMP22', 'PNMA8A', 'POLDIP2',\n",
       "       'PRR5-ARHGAP8', 'RAB32', 'RABGEF1', 'RPL12', 'SIGLEC9', 'TM6SF2',\n",
       "       'VMA21', 'WNK2', 'ZMIZ1', 'ZNF543', 'RP11-217B1.2', 'RP4-561L24.3',\n",
       "       'cyclosporin A binding GO0016018',\n",
       "       'glycosaminoglycan biosynthetic process GO0006024',\n",
       "       'organic anion transmembrane transporter activity GO0008514',\n",
       "       'positive regulation of ATP-dependent activity GO0032781',\n",
       "       'positive regulation of activated T cell proliferation GO0042104',\n",
       "       'positive regulation of mitochondrial translation GO0070131',\n",
       "       'positive regulation of mitotic cell cycle GO0045931',\n",
       "       'positive regulation of receptor-mediated endocytosis GO0048260',\n",
       "       'regulation of protein ubiquitination GO0031396',\n",
       "       'P-body assembly GO0033962',\n",
       "       'Roundabout signaling pathway GO0035385',\n",
       "       'endopeptidase activator activity GO0061133',\n",
       "       'endothelial tube lumen extension GO0097498',\n",
       "       'insulin-like growth factor receptor signaling pathway GO0048009',\n",
       "       'membrane protein intracellular domain proteolysis GO0031293',\n",
       "       'polyA binding GO0008143',\n",
       "       'regulation of synaptic transmission, glutamatergic GO0051966',\n",
       "       'plasma membrane repair GO0001778',\n",
       "       'positive regulation of cellular extravasation GO0002693',\n",
       "       'receptor clustering GO0043113',\n",
       "       'type I transforming growth factor beta receptor binding GO0034713',\n",
       "       'mRNA 3-UTR AU-rich region binding GO0035925',\n",
       "       'DNA-binding transcription activator activity, RNA polymerase II-specific GO0001228',\n",
       "       'cell surface GO0009986', 'endocytic vesicle lumen GO0071682',\n",
       "       'establishment of cell polarity GO0030010',\n",
       "       'positive regulation of innate immune response GO0045089',\n",
       "       'positive regulation of lipase activity GO0060193',\n",
       "       'stem cell differentiation GO0048863',\n",
       "       'stress granule assembly GO0034063',\n",
       "       'Other types of O-glycan biosynthesis',\n",
       "       'Transcriptional misregulation in cancer',\n",
       "       'membrane fission GO0090148',\n",
       "       'Notch receptor processing GO0007220',\n",
       "       'RNA polymerase II, core complex GO0005665',\n",
       "       'RNA polymerase II activity GO0001055',\n",
       "       'amyloid precursor protein catabolic process GO0042987',\n",
       "       'barbed-end actin filament capping GO0051016',\n",
       "       'calcium-ion regulated exocytosis GO0017156',\n",
       "       'cellular response to estrogen stimulus GO0071391',\n",
       "       'cytokine binding GO0019955',\n",
       "       'establishment of Golgi localization GO0051683',\n",
       "       'establishment of epithelial cell polarity GO0090162',\n",
       "       'guanyl-nucleotide exchange factor complex GO0032045',\n",
       "       'hippocampus development GO0021766', 'inclusion body GO0016234',\n",
       "       'late endosome to vacuole transport via multivesicular body sorting pathway GO0032511',\n",
       "       'maintenance of blood-brain barrier GO0035633',\n",
       "       'microtubule plus-end GO0035371',\n",
       "       'mitral valve morphogenesis GO0003183',\n",
       "       'negative regulation of ATP-dependent activity GO0032780',\n",
       "       'negative regulation of exocytosis GO0045920',\n",
       "       'neuron remodeling GO0016322',\n",
       "       'oligosaccharide biosynthetic process GO0009312',\n",
       "       'phosphatidylinositol phosphate binding GO1901981',\n",
       "       'positive regulation of lipid storage GO0010884',\n",
       "       'protein localization to microtubule GO0035372',\n",
       "       'receptor signaling pathway via JAK-STAT GO0007259', 'YIPF4',\n",
       "       'negative regulation of neuron projection development GO0010977',\n",
       "       'BBSome GO0034464',\n",
       "       'NLS-dependent protein nuclear import complex GO0042564',\n",
       "       'intermediate filament GO0005882',\n",
       "       'intracellular oxygen homeostasis GO0032364',\n",
       "       'negative regulation of T cell activation GO0050868',\n",
       "       'positive regulation of viral budding via host ESCRT complex GO1903774',\n",
       "       'regulation of CD8-positive, alpha-beta T cell proliferation GO2000564'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# find the overlap of the overlap of supepath name and subcategory\n",
    "miEAA['Subcategory'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Signal Transduction', 'Metabolism of Proteins',\n",
       "       'Infectious Disease', 'Innate Immune System',\n",
       "       'Gene Expression (Transcription)',\n",
       "       'Regulation of Activated PAK-2p34 By Proteasome Mediated Degradation',\n",
       "       'Cellular Responses to Stimuli', 'Cell Cycle, Mitotic',\n",
       "       'RNA Polymerase I Promoter Opening', 'Cellular Senescence',\n",
       "       'Signaling By Rho GTPases', 'Prolactin Signaling',\n",
       "       'Vesicle-mediated Transport',\n",
       "       'Processing of Capped Intron-Containing Pre-mRNA',\n",
       "       'Deubiquitination', 'Gene Silencing By RNA',\n",
       "       'Sertoli-Sertoli Cell Junction Dynamics', 'ERK Signaling',\n",
       "       'Apoptotic Pathways in Synovial Fibroblasts',\n",
       "       'MAPK Family Signaling Cascades',\n",
       "       'Cytokine Signaling in Immune System',\n",
       "       'Nervous System Development', 'DNA Damage',\n",
       "       'Chromatin Regulation / Acetylation', 'ESR-mediated Signaling',\n",
       "       'Transport of The SLBP Independent Mature MRNA',\n",
       "       'Signaling By Receptor Tyrosine Kinases', 'Cytoskeletal Signaling',\n",
       "       'SARS-CoV-2 Infection', 'Pleural Mesothelioma',\n",
       "       'Mitotic G1 Phase and G1/S Transition', 'Cell Cycle Checkpoints',\n",
       "       'HCMV Infection',\n",
       "       'Class I MHC Mediated Antigen Processing and Presentation',\n",
       "       'Assembly of The Pre-replicative Complex',\n",
       "       'Homology Directed Repair', 'Endometrial Cancer',\n",
       "       'Pre-NOTCH Expression and Processing', 'Metabolism',\n",
       "       'Chromatin Organization', 'CKAP4 Signaling Pathway Map',\n",
       "       'Signaling Events Mediated By HDAC Class I', 'MAPK-Erk Pathway',\n",
       "       'Toll-like Receptor Signaling Pathway', 'MAP Kinase Signaling',\n",
       "       'Angiopoietin-like Protein 8 Regulatory Pathway',\n",
       "       'Packaging Of Telomere Ends', 'GPCR Pathway',\n",
       "       'Influenza Infection', 'Autophagy Pathway',\n",
       "       'Regulation of TP53 Activity', 'TGF-Beta Pathway', 'Cell Cycle',\n",
       "       'Development VEGF Signaling Via VEGFR2 - Generic Cascades',\n",
       "       'Signaling By WNT',\n",
       "       'Validated Targets of C-MYC Transcriptional Activation',\n",
       "       'Hepatocyte Growth Factor Receptor Signaling',\n",
       "       'E2F Transcription Factor Network', 'PIP3 Activates AKT Signaling',\n",
       "       'NFAT and Cardiac Hypertrophy', 'Translational Control',\n",
       "       'IL-1 Family Signaling Pathways', 'P70S6K Signaling',\n",
       "       'Senescence and Autophagy in Cancer', 'Neuroscience',\n",
       "       'PDGFR-beta Signaling Pathway', 'Peptide Chain Elongation',\n",
       "       'Cell Cycle Regulation of G1/S Transition (part 1)',\n",
       "       'Apoptosis and Autophagy',\n",
       "       'ADORA2B Mediated Anti-inflammatory Cytokines Production',\n",
       "       'Regulation of Actin Dynamics For Phagocytic Cup Formation',\n",
       "       'Activation of The Pre-replicative Complex',\n",
       "       'Cytoskeleton Remodeling Regulation of Actin Cytoskeleton By Rho GTPases',\n",
       "       'MAPK Signaling: Mitogens',\n",
       "       'Development Ligand-independent Activation of ESR1 and ESR2',\n",
       "       'Programmed Cell Death', 'MAPK Signaling Pathway',\n",
       "       'Development IGF-1 Receptor Signaling', 'Akt Signaling',\n",
       "       'P73 Transcription Factor Network',\n",
       "       'Rab Regulation of Trafficking',\n",
       "       'Cellular Response to Heat Stress', 'CNTF Signaling',\n",
       "       'PKMTs Methylate Histone Lysines', 'VEGFA-VEGFR2 Signaling',\n",
       "       'Chromosome Maintenance', 'Integrated Breast Cancer Pathway',\n",
       "       'DNA Double Strand Break Response', 'Semaphorin Interactions',\n",
       "       'Small Cell Lung Cancer', 'TNFR1 Pathway',\n",
       "       'Oncogene Induced Senescence',\n",
       "       'DNA Damage Response (only ATM Dependent)',\n",
       "       'Development Endothelin-1/EDNRA Transactivation of EGFR',\n",
       "       'EPH-Ephrin Signaling', 'Wnt / Hedgehog / Notch',\n",
       "       'Gastrin Signaling Pathway', 'Integrin-mediated Cell Adhesion',\n",
       "       'Retinoblastoma Gene in Cancer',\n",
       "       'Smith-Magenis and Potocki-Lupski Syndrome Copy Number Variation',\n",
       "       'Metabolic Reprogramming in Colon Cancer',\n",
       "       'Signaling By ALK in Cancer',\n",
       "       'MiRNA Regulation of DNA Damage Response',\n",
       "       'Chks in Checkpoint Regulation', 'TCR Signaling (Qiagen)',\n",
       "       'MAPK Signaling: Oxidative Stress', 'Focal Adhesion',\n",
       "       'FMLP Pathway', 'Hepatitis C and Hepatocellular Carcinoma',\n",
       "       'Meiosis', 'Regulation of Expression of SLITs and ROBOs',\n",
       "       'DNA IR-damage and Cellular Response Via ATR', 'G0 and Early G1',\n",
       "       'AP-1 Transcription Factor Network',\n",
       "       'Corticotropin-releasing Hormone Signaling Pathway',\n",
       "       'Nuclear Events (kinase and Transcription Factor Activation)',\n",
       "       'DREAM Repression and Dynorphin Expression',\n",
       "       'Defective Binding of RB1 Mutants to E2F1,(E2F2, E2F3)',\n",
       "       'TGF-beta Signaling Pathways', 'Toll Comparative Pathway',\n",
       "       'Beta-Adrenergic Signaling', 'HIV Life Cycle',\n",
       "       'PI3K-Akt Signaling Pathway', 'Glucose / Energy Metabolism',\n",
       "       'Development Slit-Robo Signaling',\n",
       "       'Ovarian Tumor Domain Proteases', 'Clathrin-mediated Endocytosis',\n",
       "       'GDNF Signaling', 'MET Promotes Cell Motility', 'PAK Pathway',\n",
       "       'Cell Junction Organization', 'Breast Cancer Pathway',\n",
       "       'Nanog in Mammalian ESC Pluripotency', 'FLT3 Signaling in Disease',\n",
       "       'Nuclear Events Mediated By NFE2L2', 'Signaling By NTRKs',\n",
       "       'CCL18 Signaling Pathway',\n",
       "       'Immune Response_IL-6 Signaling Pathway', 'NGF Pathway',\n",
       "       'WtCFTR and Delta508-CFTR Traffic / Generic Schema (norm and CF)',\n",
       "       'Regulation of Actin Cytoskeleton', 'Acute Viral Myocarditis',\n",
       "       'Oncogenic MAPK Signaling',\n",
       "       '7q11.23 Copy Number Variation Syndrome',\n",
       "       'DYRK1A Involvement Regarding Cell Proliferation in Brain Development',\n",
       "       'P75 NTR Receptor-mediated Signalling',\n",
       "       'Transcription Androgen Receptor Nuclear Signaling',\n",
       "       'Guidance Cues and Growth Cone Motility', 'TNF Signaling',\n",
       "       'Signaling By ERBB2', 'RRNA Processing in The Nucleus and Cytosol',\n",
       "       'Death Receptor Signaling Pathway', 'Regulation of Telomerase',\n",
       "       'Transport to The Golgi and Subsequent Modification',\n",
       "       'Androgen Receptor Network in Prostate Cancer',\n",
       "       'Head and Neck Squamous Cell Carcinoma',\n",
       "       'TP53 Regulates Metabolic Genes',\n",
       "       'Apoptosis-related Network Due to Altered Notch3 in Ovarian Cancer',\n",
       "       'EPHA-mediated Growth Cone Collapse', 'RAB Geranylgeranylation',\n",
       "       'Bladder Cancer', 'Beta-2 Adrenergic-dependent CFTR Expression',\n",
       "       'Jak-Stat Signaling Pathway',\n",
       "       'NcRNAs Involved in Wnt Signaling in Hepatocellular Carcinoma',\n",
       "       'Integrin Pathway', 'Nuclear Receptors Meta-pathway',\n",
       "       'Loss of Proteins Required For Interphase Microtubule Organization From The Centrosome',\n",
       "       'Development Dopamine D2 Receptor Transactivation of EGFR',\n",
       "       'C-MYB Transcription Factor Network',\n",
       "       'Signaling Events Mediated By Hepatocyte Growth Factor Receptor (c-Met)',\n",
       "       'Development FGFR Signaling Pathway',\n",
       "       'Signaling By TGFB Family Members',\n",
       "       'Ebola Virus Infection in Host', 'Colorectal Cancer Metastasis',\n",
       "       'Cell Cycle Role of 14-3-3 Proteins in Cell Cycle Regulation',\n",
       "       'Aryl Hydrocarbon Receptor Pathway', 'RhoGDI Pathway',\n",
       "       'MRNA Splicing - Minor Pathway',\n",
       "       'Cyclins and Cell Cycle Regulation',\n",
       "       'Integrin-linked Kinase Signaling',\n",
       "       'Microtubule Cytoskeleton Regulation',\n",
       "       'P38 MAPK Signaling Pathway (WikiPathways)', 'Spinal Cord Injury',\n",
       "       'Stabilization and Expansion of The E-cadherin Adherens Junction',\n",
       "       'Signaling By EGFR', 'Intrinsic Pathway For Apoptosis',\n",
       "       '5q35 Copy Number Variation',\n",
       "       'Translation Insulin Regulation of Translation',\n",
       "       'CCR5 Pathway in Macrophages',\n",
       "       'Regulation of CDH11 Expression and Function',\n",
       "       'RHOBTB GTPase Cycle',\n",
       "       'Apoptosis and Survival BAD Phosphorylation',\n",
       "       'Regulation of RUNX1 Expression and Activity',\n",
       "       'IL2 Signaling Events Mediated By STAT5',\n",
       "       'HIF-1-alpha Transcription Factor Network',\n",
       "       'Glycolysis (REACTOME)', 'IL-9 Signaling Pathways',\n",
       "       'Direct P53 Effectors',\n",
       "       'Interleukin-4 and Interleukin-13 Signaling',\n",
       "       'Apoptosis and Survival FAS Signaling Cascades',\n",
       "       'Sudden Infant Death Syndrome (SIDS) Susceptibility Pathways',\n",
       "       'Activation of AMPK Downstream of NMDARs',\n",
       "       'MyD88 Dependent Cascade Initiated on Endosome',\n",
       "       'Development EGFR Signaling Pathway',\n",
       "       'TP53 Regulates Transcription of Cell Cycle Genes',\n",
       "       'Immune Response IL-2 Activation and Signaling Pathway',\n",
       "       'Immune Response_Role of Integrins in NK Cells Cytotoxicity',\n",
       "       'Integrins in Angiogenesis', 'CXCR4-mediated Signaling Events',\n",
       "       'GADD45 Pathway', 'Response to Elevated Platelet Cytosolic Ca2+',\n",
       "       'Immune Response IL-23 Signaling Pathway',\n",
       "       'FOXM1 Transcription Factor Network', 'CD28 Co-stimulation',\n",
       "       'EML4 and NUDC in Mitotic Spindle Formation',\n",
       "       'Separation of Sister Chromatids',\n",
       "       'Development Angiotensin Activation of ERK',\n",
       "       'NF-KappaB Family Pathway', 'NF-kappaB Signaling',\n",
       "       'Epithelial to Mesenchymal Transition in Colorectal Cancer',\n",
       "       'APC-Cdc20 Mediated Degradation of Nek2A',\n",
       "       'Mesodermal Commitment Pathway',\n",
       "       'Activated TAK1 Mediates P38 MAPK Activation',\n",
       "       'Regulation of TP53 Expression and Degradation',\n",
       "       'Thromboxane Signalling Through TP Receptor', 'Signaling By PTK6',\n",
       "       'Immune Response_IFN Gamma Signaling Pathway',\n",
       "       \"Alzheimer's Disease and MiRNA Effects\",\n",
       "       'P38 Signaling Mediated By MAPKAP Kinases', 'P53 Pathway',\n",
       "       'TGF-beta Receptor Signaling',\n",
       "       'Cooperation of Prefoldin and TriC/CCT in Actin and Tubulin Folding',\n",
       "       'P53 Signaling', 'Uptake and Actions of Bacterial Toxins',\n",
       "       'Metabolic Reprogramming in Pancreatic Cancer',\n",
       "       'Actin Nucleation By ARP-WASP Complex',\n",
       "       'MIF Mediated Glucocorticoid Regulation',\n",
       "       'G12-G13 in Cellular Signaling', 'VEGF Signaling Pathway',\n",
       "       'Transcriptional Activity of SMAD2/SMAD3:SMAD4 Heterotrimer',\n",
       "       'ATM Pathway', 'G-protein Signaling RAC1 in Cellular Process',\n",
       "       'EphB-EphrinB Signaling', 'Netrin-1 Signaling',\n",
       "       'DNA Damage Induced 14-3-3Sigma Signaling',\n",
       "       'Regulation of Nuclear SMAD2/3 Signaling',\n",
       "       'Golgi-to-ER Retrograde Transport',\n",
       "       'NR1H2 and NR1H3-mediated Signaling',\n",
       "       'Transport of Mature Transcript to Cytoplasm',\n",
       "       'RAF-independent MAPK1/3 Activation', 'RAC2 GTPase Cycle',\n",
       "       'Transport of Inorganic Cations/anions and Amino Acids/oligopeptides',\n",
       "       'Apoptotic Cleavage of Cellular Proteins',\n",
       "       'TCR Signaling (REACTOME)', 'Syndecan-4-mediated Signaling Events',\n",
       "       'Acute Myeloid Leukemia',\n",
       "       'Microglia Pathogen Phagocytosis Pathway',\n",
       "       'Nucleotide-binding Domain, Leucine Rich Repeat Containing Receptor (NLR) Signaling Pathways',\n",
       "       'Adipogenesis', 'DNA Damage_ATM/ATR Regulation of G1/S Checkpoint',\n",
       "       'Interleukin-11 Signaling Pathway',\n",
       "       'Overlap Between Signal Transduction Pathways Contributing to LMNA Laminopathies',\n",
       "       'Antiviral Mechanism By IFN-stimulated Genes',\n",
       "       'NRP1-triggered Signaling Pathways in Pancreatic Cancer',\n",
       "       'Sphingolipid Metabolism', 'Glycolysis (BioCyc)',\n",
       "       'CCR3 Pathway in Eosinophils', 'Interleukin-1 Family Signaling'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc['SuperPath Name'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Category</th>\n",
       "      <th>Subcategory</th>\n",
       "      <th>Enrichment</th>\n",
       "      <th>P-value</th>\n",
       "      <th>P-adjusted</th>\n",
       "      <th>Q-value</th>\n",
       "      <th>Observed</th>\n",
       "      <th>miRNAs/precursors</th>\n",
       "      <th>Score</th>\n",
       "      <th>SuperPath Name</th>\n",
       "      <th># SuperPath Total Genes</th>\n",
       "      <th># SuperPath Matched Genes</th>\n",
       "      <th>Matched Genes (Symbols)</th>\n",
       "      <th>Evidence URL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WikiPathways (miRPathDB)</td>\n",
       "      <td>Cell Cycle</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.006669</td>\n",
       "      <td>0.006669</td>\n",
       "      <td>0.266763</td>\n",
       "      <td>5</td>\n",
       "      <td>hsa-miR-6802-5p; hsa-miR-6729-5p; hsa-miR-6800...</td>\n",
       "      <td>24.44</td>\n",
       "      <td>Cell Cycle</td>\n",
       "      <td>137</td>\n",
       "      <td>47</td>\n",
       "      <td>WEE1, CCNA2, CCNB1, CCND1, CCND2, MDM2, MYC, C...</td>\n",
       "      <td>https://pathcards.genecards.org/card/cell_cycle</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WikiPathways (miRPathDB)</td>\n",
       "      <td>Mesodermal Commitment Pathway</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.015152</td>\n",
       "      <td>0.015152</td>\n",
       "      <td>0.418301</td>\n",
       "      <td>2</td>\n",
       "      <td>hsa-miR-6802-5p; hsa-miR-6800-5p</td>\n",
       "      <td>11.81</td>\n",
       "      <td>Mesodermal Commitment Pathway</td>\n",
       "      <td>146</td>\n",
       "      <td>38</td>\n",
       "      <td>AXIN1, EXT1, CCND1, KLF4, DIP2A, ELK4, CSRP2, ...</td>\n",
       "      <td>https://pathcards.genecards.org/card/mesoderma...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Reactome (miRPathDB)</td>\n",
       "      <td>Cell Cycle Checkpoints</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.026782</td>\n",
       "      <td>0.026782</td>\n",
       "      <td>0.458347</td>\n",
       "      <td>5</td>\n",
       "      <td>hsa-miR-6802-5p; hsa-miR-6729-5p; hsa-miR-663a...</td>\n",
       "      <td>31.67</td>\n",
       "      <td>Cell Cycle Checkpoints</td>\n",
       "      <td>292</td>\n",
       "      <td>86</td>\n",
       "      <td>RANBP2, TOP3A, BRCA1, NUP43, WEE1, CCNA2, CCNB...</td>\n",
       "      <td>https://pathcards.genecards.org/card/cell_cycl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Reactome (miRPathDB)</td>\n",
       "      <td>Signal Transduction</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.028283</td>\n",
       "      <td>0.028283</td>\n",
       "      <td>0.458347</td>\n",
       "      <td>2</td>\n",
       "      <td>hsa-miR-6802-5p; hsa-miR-663a</td>\n",
       "      <td>166.10</td>\n",
       "      <td>Signal Transduction</td>\n",
       "      <td>2597</td>\n",
       "      <td>559</td>\n",
       "      <td>MUC20, ARL2, MAP2K2, DGKH, ARL4C, PTK6, NHS, C...</td>\n",
       "      <td>https://pathcards.genecards.org/card/signal_tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>WikiPathways (miRPathDB)</td>\n",
       "      <td>Integrin-mediated Cell Adhesion</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.028283</td>\n",
       "      <td>0.028283</td>\n",
       "      <td>0.418301</td>\n",
       "      <td>2</td>\n",
       "      <td>hsa-miR-6802-5p; hsa-miR-663a</td>\n",
       "      <td>18.52</td>\n",
       "      <td>Integrin-mediated Cell Adhesion</td>\n",
       "      <td>140</td>\n",
       "      <td>43</td>\n",
       "      <td>MAP2K2, MAP2K3, CAPN6, CAPNS1, MAP2K7, MAP3K2,...</td>\n",
       "      <td>https://pathcards.genecards.org/card/integrin-...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Category                      Subcategory Enrichment  \\\n",
       "0  WikiPathways (miRPathDB)                       Cell Cycle   enriched   \n",
       "1  WikiPathways (miRPathDB)    Mesodermal Commitment Pathway   enriched   \n",
       "2      Reactome (miRPathDB)           Cell Cycle Checkpoints   enriched   \n",
       "3      Reactome (miRPathDB)              Signal Transduction   enriched   \n",
       "4  WikiPathways (miRPathDB)  Integrin-mediated Cell Adhesion   enriched   \n",
       "\n",
       "    P-value  P-adjusted   Q-value  Observed  \\\n",
       "0  0.006669    0.006669  0.266763         5   \n",
       "1  0.015152    0.015152  0.418301         2   \n",
       "2  0.026782    0.026782  0.458347         5   \n",
       "3  0.028283    0.028283  0.458347         2   \n",
       "4  0.028283    0.028283  0.418301         2   \n",
       "\n",
       "                                   miRNAs/precursors   Score  \\\n",
       "0  hsa-miR-6802-5p; hsa-miR-6729-5p; hsa-miR-6800...   24.44   \n",
       "1                   hsa-miR-6802-5p; hsa-miR-6800-5p   11.81   \n",
       "2  hsa-miR-6802-5p; hsa-miR-6729-5p; hsa-miR-663a...   31.67   \n",
       "3                      hsa-miR-6802-5p; hsa-miR-663a  166.10   \n",
       "4                      hsa-miR-6802-5p; hsa-miR-663a   18.52   \n",
       "\n",
       "                    SuperPath Name  # SuperPath Total Genes  \\\n",
       "0                       Cell Cycle                      137   \n",
       "1    Mesodermal Commitment Pathway                      146   \n",
       "2           Cell Cycle Checkpoints                      292   \n",
       "3              Signal Transduction                     2597   \n",
       "4  Integrin-mediated Cell Adhesion                      140   \n",
       "\n",
       "   # SuperPath Matched Genes  \\\n",
       "0                         47   \n",
       "1                         38   \n",
       "2                         86   \n",
       "3                        559   \n",
       "4                         43   \n",
       "\n",
       "                             Matched Genes (Symbols)  \\\n",
       "0  WEE1, CCNA2, CCNB1, CCND1, CCND2, MDM2, MYC, C...   \n",
       "1  AXIN1, EXT1, CCND1, KLF4, DIP2A, ELK4, CSRP2, ...   \n",
       "2  RANBP2, TOP3A, BRCA1, NUP43, WEE1, CCNA2, CCNB...   \n",
       "3  MUC20, ARL2, MAP2K2, DGKH, ARL4C, PTK6, NHS, C...   \n",
       "4  MAP2K2, MAP2K3, CAPN6, CAPNS1, MAP2K7, MAP3K2,...   \n",
       "\n",
       "                                        Evidence URL  \n",
       "0    https://pathcards.genecards.org/card/cell_cycle  \n",
       "1  https://pathcards.genecards.org/card/mesoderma...  \n",
       "2  https://pathcards.genecards.org/card/cell_cycl...  \n",
       "3  https://pathcards.genecards.org/card/signal_tr...  \n",
       "4  https://pathcards.genecards.org/card/integrin-...  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlap = miEAA.merge(gc, left_on='Subcategory', right_on='SuperPath Name')\n",
    "overlap.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "myenv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
