{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Category', 'Subcategory', 'Enrichment', 'P-value', 'P-adjusted',\n",
      "       'Q-value', 'Observed', 'miRNAs/precursors'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "miEAA = pd.read_csv('./s2_s3_miEAA.csv')\n",
    "print(miEAA.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Annotations derived over miRTarBase (Gene Ontology)'\n",
      " 'Target genes (miRTarBase)' 'Pathways (KEGG)' 'Diseases (MNDR)'\n",
      " 'Cell-type specific (Cellular microRNAome)'\n",
      " 'GO Molecular function (miRPathDB)' 'Interactions (NPInter)']\n"
     ]
    }
   ],
   "source": [
    "# Print unique categories\n",
    "print(miEAA['Category'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Score', 'SuperPath Name', '# SuperPath Total Genes',\n",
       "       '# SuperPath Matched Genes', 'Matched Genes (Symbols)', 'Evidence URL'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc = pd.read_csv('./s2_s3-genecards.csv')\n",
    "gc.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Category</th>\n",
       "      <th>Subcategory</th>\n",
       "      <th>Enrichment</th>\n",
       "      <th>P-value</th>\n",
       "      <th>P-adjusted</th>\n",
       "      <th>Q-value</th>\n",
       "      <th>Observed</th>\n",
       "      <th>miRNAs/precursors</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>negative regulation of extrinsic apoptotic sig...</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.000158</td>\n",
       "      <td>0.000158</td>\n",
       "      <td>0.237114</td>\n",
       "      <td>11</td>\n",
       "      <td>hsa-miR-4722-3p; hsa-miR-6829-3p; hsa-miR-6769...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>Hsp70 protein binding GO0030544</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.000214</td>\n",
       "      <td>0.000214</td>\n",
       "      <td>0.237114</td>\n",
       "      <td>5</td>\n",
       "      <td>hsa-miR-6829-3p; hsa-miR-6769b-5p; hsa-miR-428...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>cellular response to retinoic acid GO0071300</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.000223</td>\n",
       "      <td>0.000223</td>\n",
       "      <td>0.237114</td>\n",
       "      <td>6</td>\n",
       "      <td>hsa-miR-4722-3p; hsa-miR-6829-3p; hsa-miR-4282...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Annotations derived over miRTarBase (Gene Onto...</td>\n",
       "      <td>calcium-activated potassium channel activity G...</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>0.237114</td>\n",
       "      <td>3</td>\n",
       "      <td>hsa-miR-3130-3p; hsa-miR-4722-3p; hsa-miR-6829-3p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Target genes (miRTarBase)</td>\n",
       "      <td>FAM241A</td>\n",
       "      <td>enriched</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>0.276668</td>\n",
       "      <td>3</td>\n",
       "      <td>hsa-miR-3130-3p; hsa-miR-4722-3p; hsa-miR-6829-3p</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Category  \\\n",
       "0  Annotations derived over miRTarBase (Gene Onto...   \n",
       "1  Annotations derived over miRTarBase (Gene Onto...   \n",
       "2  Annotations derived over miRTarBase (Gene Onto...   \n",
       "3  Annotations derived over miRTarBase (Gene Onto...   \n",
       "4                          Target genes (miRTarBase)   \n",
       "\n",
       "                                         Subcategory Enrichment   P-value  \\\n",
       "0  negative regulation of extrinsic apoptotic sig...   enriched  0.000158   \n",
       "1                    Hsp70 protein binding GO0030544   enriched  0.000214   \n",
       "2       cellular response to retinoic acid GO0071300   enriched  0.000223   \n",
       "3  calcium-activated potassium channel activity G...   enriched  0.000305   \n",
       "4                                            FAM241A   enriched  0.000305   \n",
       "\n",
       "   P-adjusted   Q-value  Observed  \\\n",
       "0    0.000158  0.237114        11   \n",
       "1    0.000214  0.237114         5   \n",
       "2    0.000223  0.237114         6   \n",
       "3    0.000305  0.237114         3   \n",
       "4    0.000305  0.276668         3   \n",
       "\n",
       "                                   miRNAs/precursors  \n",
       "0  hsa-miR-4722-3p; hsa-miR-6829-3p; hsa-miR-6769...  \n",
       "1  hsa-miR-6829-3p; hsa-miR-6769b-5p; hsa-miR-428...  \n",
       "2  hsa-miR-4722-3p; hsa-miR-6829-3p; hsa-miR-4282...  \n",
       "3  hsa-miR-3130-3p; hsa-miR-4722-3p; hsa-miR-6829-3p  \n",
       "4  hsa-miR-3130-3p; hsa-miR-4722-3p; hsa-miR-6829-3p  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miEAA.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Score</th>\n",
       "      <th>SuperPath Name</th>\n",
       "      <th># SuperPath Total Genes</th>\n",
       "      <th># SuperPath Matched Genes</th>\n",
       "      <th>Matched Genes (Symbols)</th>\n",
       "      <th>Evidence URL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Signal Transduction</td>\n",
       "      <td>2597</td>\n",
       "      <td>530</td>\n",
       "      <td>MAP2K1, PPP1CC, DGKH, ARL4C, PTK6, NHS, RALGAP...</td>\n",
       "      <td>https://pathcards.genecards.org/card/signal_tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Metabolism of Proteins</td>\n",
       "      <td>2077</td>\n",
       "      <td>427</td>\n",
       "      <td>CANX, EIF2S1, RPS19, SAFB, GAN, RPL12, EIF2S3,...</td>\n",
       "      <td>https://pathcards.genecards.org/card/metabolis...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Infectious Disease</td>\n",
       "      <td>1818</td>\n",
       "      <td>399</td>\n",
       "      <td>CANX, RPS19, MAP2K1, MAP2K3, PPP1CC, RPL12, SL...</td>\n",
       "      <td>https://pathcards.genecards.org/card/infectiou...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Gene Expression (Transcription)</td>\n",
       "      <td>1543</td>\n",
       "      <td>374</td>\n",
       "      <td>MAP2K6, ZNF317, CARM1, CNOT2, PBRM1, NUDT21, T...</td>\n",
       "      <td>https://pathcards.genecards.org/card/gene_expr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>166.1</td>\n",
       "      <td>Regulation of Activated PAK-2p34 By Proteasome...</td>\n",
       "      <td>943</td>\n",
       "      <td>226</td>\n",
       "      <td>CANX, EIF2S1, EIF2S3, PTK6, MAP2K4, ABCA12, TN...</td>\n",
       "      <td>https://pathcards.genecards.org/card/regulatio...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Score                                     SuperPath Name  \\\n",
       "0  166.1                                Signal Transduction   \n",
       "1  166.1                             Metabolism of Proteins   \n",
       "2  166.1                                 Infectious Disease   \n",
       "3  166.1                    Gene Expression (Transcription)   \n",
       "4  166.1  Regulation of Activated PAK-2p34 By Proteasome...   \n",
       "\n",
       "   # SuperPath Total Genes  # SuperPath Matched Genes  \\\n",
       "0                     2597                        530   \n",
       "1                     2077                        427   \n",
       "2                     1818                        399   \n",
       "3                     1543                        374   \n",
       "4                      943                        226   \n",
       "\n",
       "                             Matched Genes (Symbols)  \\\n",
       "0  MAP2K1, PPP1CC, DGKH, ARL4C, PTK6, NHS, RALGAP...   \n",
       "1  CANX, EIF2S1, RPS19, SAFB, GAN, RPL12, EIF2S3,...   \n",
       "2  CANX, RPS19, MAP2K1, MAP2K3, PPP1CC, RPL12, SL...   \n",
       "3  MAP2K6, ZNF317, CARM1, CNOT2, PBRM1, NUDT21, T...   \n",
       "4  CANX, EIF2S1, EIF2S3, PTK6, MAP2K4, ABCA12, TN...   \n",
       "\n",
       "                                        Evidence URL  \n",
       "0  https://pathcards.genecards.org/card/signal_tr...  \n",
       "1  https://pathcards.genecards.org/card/metabolis...  \n",
       "2  https://pathcards.genecards.org/card/infectiou...  \n",
       "3  https://pathcards.genecards.org/card/gene_expr...  \n",
       "4  https://pathcards.genecards.org/card/regulatio...  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Annotations derived over miRTarBase (Gene Ontology)'\n",
      " 'Target genes (miRTarBase)' 'Pathways (KEGG)' 'Diseases (MNDR)'\n",
      " 'Cell-type specific (Cellular microRNAome)'\n",
      " 'GO Molecular function (miRPathDB)' 'Interactions (NPInter)']\n",
      "Category\n",
      "Annotations derived over miRTarBase (Gene Ontology)    782\n",
      "Target genes (miRTarBase)                              149\n",
      "Pathways (KEGG)                                         66\n",
      "Interactions (NPInter)                                  39\n",
      "Diseases (MNDR)                                          3\n",
      "Cell-type specific (Cellular microRNAome)                1\n",
      "GO Molecular function (miRPathDB)                        1\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# print the categories from Reactome\n",
    "print(miEAA['Category'].unique())\n",
    "print(miEAA['Category'].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Annotations derived over miRTarBase (Gene Ontology)\n",
      "['negative regulation of extrinsic apoptotic signaling pathway via death domain receptors GO1902042'\n",
      " 'Hsp70 protein binding GO0030544'\n",
      " 'cellular response to retinoic acid GO0071300'\n",
      " 'calcium-activated potassium channel activity GO0015269'\n",
      " 'RNA polymerase III complex GO0005666'\n",
      " 'DNA endonuclease activity GO0004520'\n",
      " 'selenocysteine incorporation GO0001514'\n",
      " 'tRNA transcription by RNA polymerase III GO0042797'\n",
      " 'Golgi cis cisterna GO0000137'\n",
      " 'negative regulation of extrinsic apoptotic signaling pathway in absence of ligand GO2001240'\n",
      " 'chondrocyte differentiation GO0002062'\n",
      " 'negative regulation of hematopoietic progenitor cell differentiation GO1901533'\n",
      " 'positive regulation of DNA-directed DNA polymerase activity GO1900264'\n",
      " 'replication fork GO0005657' 'ubiquitin ligase complex GO0000151'\n",
      " 'macroautophagy GO0016236' 'cell fate determination GO0001709'\n",
      " 'deadenylation-independent decapping of nuclear-transcribed mRNA GO0031087'\n",
      " 'estrogen response element binding GO0034056'\n",
      " 'negative regulation of cytokine production GO0001818'\n",
      " 'positive regulation of CREB transcription factor activity GO0032793'\n",
      " 'regulation of translation GO0006417'\n",
      " 'establishment or maintenance of epithelial cell apicalbasal polarity GO0045197'\n",
      " 'L-glutamate transmembrane transport GO0015813'\n",
      " 'U2-type prespliceosome assembly GO1903241' 'U2 snRNP GO0005686'\n",
      " 'catalytic complex GO1902494' 'chemoattractant activity GO0042056'\n",
      " 'negative regulation of execution phase of apoptosis GO1900118'\n",
      " 'early phagosome GO0032009'\n",
      " 'histone H3K27me2H3K27me3 demethylase activity GO0071558'\n",
      " 'neurotransmitter transmembrane transporter activity GO0005326'\n",
      " 'neurotransmitter transport GO0006836'\n",
      " 'regulation of interleukin-1 beta production GO0032651'\n",
      " 'regulation of platelet activation GO0010543'\n",
      " 'DNA-directed 5-3 RNA polymerase activity GO0003899'\n",
      " 'RNA polymerase III activity GO0001056'\n",
      " 'U12-type spliceosomal complex GO0005689'\n",
      " 'four-way junction DNA binding GO0000400'\n",
      " 'spermatid development GO0007286'\n",
      " 'COPII-coated vesicle cargo loading GO0090110'\n",
      " 'COPII vesicle coat GO0030127' 'myofibril assembly GO0030239'\n",
      " 'positive regulation of exocytosis GO0045921'\n",
      " 'striated muscle thin filament GO0005865'\n",
      " 'catalytic step 2 spliceosome GO0071013'\n",
      " '5S class rRNA transcription by RNA polymerase III GO0042791'\n",
      " 'DNA-templated DNA replication GO0006261'\n",
      " 'DNA metabolic process GO0006259'\n",
      " 'DNA replication preinitiation complex assembly GO0071163'\n",
      " 'L-cystine transport GO0015811'\n",
      " 'L-kynurenine transmembrane transport GO0140924'\n",
      " 'L-kynurenine transmembrane transporter activity GO0140926'\n",
      " 'RNA polymerase III general transcription initiation factor activity GO0000995'\n",
      " 'acetyl-CoA binding GO1905502' 'apoptotic cell clearance GO0043277'\n",
      " 'cystineglutamate antiporter activity GO0015327'\n",
      " 'deadenylation-dependent decapping of nuclear-transcribed mRNA GO0000290'\n",
      " 'inflammatory response to antigenic stimulus GO0002437'\n",
      " 'negative regulation of DNA-templated DNA replication GO2000104'\n",
      " 'negative regulation of ferroptosis GO0110076'\n",
      " 'plasma membrane organization GO0007009'\n",
      " 'positive regulation of chemokine production GO0032722'\n",
      " 'positive regulation of protein localization to kinetochore GO1905342'\n",
      " 'regulation of COPII vesicle coating GO0003400'\n",
      " 'sperm DNA condensation GO0035092'\n",
      " 'transcription factor TFIIIC complex GO0000127' 'Z disc GO0030018'\n",
      " 'centrosome cycle GO0007098'\n",
      " 'release of cytochrome c from mitochondria GO0001836'\n",
      " 'EP4 subtype prostaglandin E2 receptor binding GO0031867'\n",
      " 'myosin II binding GO0045159'\n",
      " 'negative regulation of glucokinase activity GO0033132'\n",
      " 'positive regulation of interleukin-1 production GO0032732'\n",
      " 'positive regulation of phospholipase C activity GO0010863'\n",
      " 'prostaglandin metabolic process GO0006693' 'transcytosis GO0045056'\n",
      " 'regulation of immune response GO0050776'\n",
      " 'negative regulation of cell differentiation GO0045596'\n",
      " 'phosphatidylinositol-4,5-bisphosphate binding GO0005546'\n",
      " 'positive regulation of stem cell population maintenance GO1902459'\n",
      " 'spindle pole GO0000922'\n",
      " 'positive regulation of fat cell differentiation GO0045600'\n",
      " 'response to unfolded protein GO0006986'\n",
      " 'sprouting angiogenesis GO0002040' 'ATPase activator activity GO0001671'\n",
      " 'antibacterial humoral response GO0019731'\n",
      " 'innate immune response in mucosa GO0002227'\n",
      " 'plasma membrane to endosome transport GO0048227'\n",
      " 'positive regulation of vascular endothelial growth factor signaling pathway GO1900748'\n",
      " 'RNA polymerase II activity GO0001055'\n",
      " 'cell growth involved in cardiac muscle cell development GO0061049'\n",
      " 'establishment of epithelial cell polarity GO0090162'\n",
      " 'positive regulation of fibroblast growth factor receptor signaling pathway GO0045743'\n",
      " 'syndecan binding GO0045545' 'vesicle targeting GO0006903'\n",
      " 'GBAF complex GO0140288' 'gamma-tubulin binding GO0043015'\n",
      " 'glucose metabolic process GO0006006'\n",
      " 'phosphatidylinositol-4-phosphate binding GO0070273'\n",
      " '4 iron, 4 sulfur cluster binding GO0051539'\n",
      " 'GTP metabolic process GO0046039' 'TORC2 signaling GO0038203'\n",
      " 'cellular response to glucocorticoid stimulus GO0071385'\n",
      " 'cellular response to platelet-derived growth factor stimulus GO0036120'\n",
      " 'lung development GO0030324' 'muscle structure development GO0061061'\n",
      " 'protein polyubiquitination GO0000209'\n",
      " 'enzyme activator activity GO0008047'\n",
      " 'regulation of G1S transition of mitotic cell cycle GO2000045'\n",
      " 'actin filament polymerization GO0030041'\n",
      " 'multivesicular body sorting pathway GO0071985'\n",
      " 'plasma membrane repair GO0001778'\n",
      " 'transcription by RNA polymerase III GO0006383'\n",
      " 'L-glutamine transmembrane transporter activity GO0015186'\n",
      " 'glutamine transport GO0006868' 'Hsp90 protein binding GO0051879'\n",
      " 'detection of calcium ion GO0005513'\n",
      " 'heparan sulfate proteoglycan binding GO0043395'\n",
      " 'stem cell population maintenance GO0019827'\n",
      " '3-dehydrosphinganine reductase activity GO0047560'\n",
      " '3-keto-sphinganine metabolic process GO0006666'\n",
      " '3 iron, 4 sulfur cluster binding GO0051538'\n",
      " '5-nucleotidase activity GO0008253' 'ANPR-A receptor complex GO1990620'\n",
      " 'Barr body GO0001740' 'C-X-C chemokine binding GO0019958'\n",
      " 'Ctf18 RFC-like complex GO0031390'\n",
      " 'DNA 5-adenosine monophosphate hydrolase activity GO0033699'\n",
      " 'DNA clamp loader activity GO0003689' 'DNA geometric change GO0032392'\n",
      " 'DNA replication factor C complex GO0005663'\n",
      " 'DNA topoisomerase type I single strand cut, ATP-independent activity GO0003917'\n",
      " 'DNA topological change GO0006265'\n",
      " 'GMP 5-nucleotidase activity GO0050484' 'GMP metabolic process GO0046037'\n",
      " 'GPI anchor binding GO0034235'\n",
      " 'G protein-coupled receptor kinase activity GO0004703'\n",
      " 'H4K20me3 modified histone binding GO1990889'\n",
      " 'IMP 5-nucleotidase activity GO0050483' 'IMP catabolic process GO0006204'\n",
      " 'JUN kinase kinase kinase activity GO0004706'\n",
      " 'L-cystine transmembrane transporter activity GO0015184'\n",
      " 'Mre11 complex GO0030870'\n",
      " 'NADPH dehydrogenase quinone activity GO0003955'\n",
      " 'RAGE receptor binding GO0050786'\n",
      " 'RecQ family helicase-topoisomerase III complex GO0031422'\n",
      " 'Ric1-Rgp1 guanyl-nucleotide exchange factor complex GO0034066'\n",
      " 'T-helper 1 cell activation GO0035711'\n",
      " 'T-helper 1 cell differentiation GO0045063' 'VDJ recombination GO0033151'\n",
      " 'aconitate hydratase activity GO0003994'\n",
      " 'adenosine metabolic process GO0046085'\n",
      " 'allantoin metabolic process GO0000255'\n",
      " 'alphav-beta3 integrin-HMGB1 complex GO0035868'\n",
      " 'alternative mRNA splicing, via spliceosome GO0000380'\n",
      " 'amino acid metabolic process GO0006520'\n",
      " 'aspartate-tRNAAsn ligase activity GO0050560'\n",
      " 'aspartate-tRNA ligase activity GO0004815'\n",
      " 'aspartyl-tRNA aminoacylation GO0006422'\n",
      " 'beta-adrenergic receptor kinase activity GO0047696'\n",
      " 'beta-glucuronidase activity GO0004566'\n",
      " 'branched-chain amino acid metabolic process GO0009081'\n",
      " 'cellular response to fibroblast growth factor stimulus GO0044344'\n",
      " 'cerebellar Purkinje cell differentiation GO0021702'\n",
      " 'chondrocyte proliferation GO0035988' 'chromosomal region GO0098687'\n",
      " 'chromosome separation GO0051304' 'cilium disassembly GO0061523'\n",
      " 'dGMP metabolic process GO0046054'\n",
      " 'deactivation of mitotic spindle assembly checkpoint GO1902426'\n",
      " 'deoxyribonuclease II activity GO0004531'\n",
      " 'desensitization of G protein-coupled receptor signaling pathway GO0002029'\n",
      " 'dihydronicotinamide riboside quinone reductase activity GO0001512'\n",
      " 'dipeptidase activity GO0016805'\n",
      " 'electron-transferring-flavoprotein dehydrogenase activity GO0004174'\n",
      " 'electron transport chain GO0022900'\n",
      " 'endoplasmic reticulum membrane organization GO0090158'\n",
      " 'ether lipid metabolic process GO0046485'\n",
      " 'exploration behavior GO0035640'\n",
      " 'eye photoreceptor cell differentiation GO0001754'\n",
      " 'farnesyltranstransferase activity GO0004311'\n",
      " 'fatty acid beta-oxidation using acyl-CoA dehydrogenase GO0033539'\n",
      " 'fructose-bisphosphate aldolase activity GO0004332'\n",
      " 'fructose 1,6-bisphosphate metabolic process GO0030388'\n",
      " 'fructose binding GO0070061' 'fructose metabolic process GO0006000'\n",
      " 'glyceryl-ether monooxygenase activity GO0050479'\n",
      " 'heparan sulfate proteoglycan catabolic process GO0030200'\n",
      " 'heparanase activity GO0030305'\n",
      " 'histone H4K20 trimethyltransferase activity GO0140943'\n",
      " 'immunoglobulin transcytosis in epithelial cells mediated by polymeric immunoglobulin receptor GO0002415'\n",
      " 'inner kinetochore GO0000939' 'iron-responsive element binding GO0030350'\n",
      " 'isoprenoid metabolic process GO0006720'\n",
      " 'kinetochore organization GO0051383' 'lyase activity GO0016829'\n",
      " 'male gamete generation GO0048232'\n",
      " 'meiotic DNA double-strand break formation GO0042138'\n",
      " 'melatonin binding GO1904408' 'membrane assembly GO0071709'\n",
      " 'membrane lipid metabolic process GO0006643'\n",
      " 'methylguanosine-cap decapping GO0110156'\n",
      " 'mismatched DNA binding GO0030983'\n",
      " 'mitochondrial asparaginyl-tRNA aminoacylation GO0070145'\n",
      " 'mitochondrial double-strand break repair via homologous recombination GO0097552'\n",
      " 'myeloid dendritic cell activation GO0001773'\n",
      " 'natriuretic peptide receptor activity GO0016941'\n",
      " 'negative regulation of CD4-positive, alpha-beta T cell differentiation GO0043371'\n",
      " 'negative regulation of apoptotic cell clearance GO2000426'\n",
      " 'negative regulation of defense response to virus by host GO0050689'\n",
      " 'negative regulation of intrinsic apoptotic signaling pathway in response to DNA damage by p53 class mediator GO1902166'\n",
      " 'negative regulation of organ growth GO0046621'\n",
      " 'negative regulation of protein autoubiquitination GO1905524'\n",
      " 'negative regulation of protein localization to kinetochore GO1905341'\n",
      " 'negative regulation of triglyceride catabolic process GO0010897'\n",
      " 'neuromuscular process GO0050905' 'neutrophil clearance GO0097350'\n",
      " 'nucleoside phosphotransferase activity GO0050146'\n",
      " 'oxalate transmembrane transporter activity GO0019531'\n",
      " 'oxidoreductase activity, acting on other nitrogenous compounds as donors GO0016661'\n",
      " 'peptide receptor activity GO0001653'\n",
      " 'peptidyl-proline 3-dioxygenase activity GO0031544'\n",
      " 'peptidyl-proline dioxygenase activity GO0031543'\n",
      " 'phosphoglycolate phosphatase activity GO0008967'\n",
      " 'phospholipid biosynthetic process GO0008654'\n",
      " 'phosphotransferase activity, for other substituted phosphate groups GO0016780'\n",
      " 'piRNA binding GO0034584' 'piRNA processing GO0034587'\n",
      " 'positive regulation of DNA-templated DNA replication GO2000105'\n",
      " 'positive regulation of cytoskeleton organization GO0051495'\n",
      " 'positive regulation of dendritic cell differentiation GO2001200'\n",
      " 'positive regulation of mismatch repair GO0032425'\n",
      " 'positive regulation of mitochondrial membrane permeability involved in apoptotic process GO1902110'\n",
      " 'positive regulation of nuclear cell cycle DNA replication GO0010571'\n",
      " 'positive regulation of phagocytosis, engulfment GO0060100'\n",
      " 'positive regulation of protein localization to cell cortex GO1904778'\n",
      " 'positive regulation of protein polyubiquitination GO1902916'\n",
      " 'positive regulation of renal sodium excretion GO0035815'\n",
      " 'positive regulation of sequestering of triglyceride GO0010890'\n",
      " 'positive regulation of telomerase RNA localization to Cajal body GO1904874'\n",
      " 'positive regulation of urine volume GO0035810'\n",
      " 'prenyltransferase activity GO0004659'\n",
      " 'propionyl-CoA carboxylase activity GO0004658'\n",
      " 'protein hydroxylation GO0018126'\n",
      " 'protein localization to postsynapse GO0062237'\n",
      " 'protein targeting to Golgi apparatus GO0140450'\n",
      " 'proteoglycan metabolic process GO0006029' 'quinone binding GO0048038'\n",
      " 'quinone catabolic process GO1901662'\n",
      " 'regulation of T cell mediated immune response to tumor cell GO0002840'\n",
      " 'regulation of hair follicle development GO0051797'\n",
      " 'regulation of mitotic recombination GO0000019'\n",
      " 'regulation of protein glycosylation GO0060049'\n",
      " 'regulation of protein metabolic process GO0051246'\n",
      " 'regulation of restriction endodeoxyribonuclease activity GO0032072'\n",
      " 'regulation of tolerance induction GO0002643'\n",
      " 'regulation of translational termination GO0006449'\n",
      " 'regulation of vascular permeability GO0043114'\n",
      " 'regulation of voltage-gated calcium channel activity GO1901385'\n",
      " 'resolution of recombination intermediates GO0071139'\n",
      " 'response to ironII ion GO0010040' 'response to sorbitol GO0072708'\n",
      " 'ribosomal protein import into nucleus GO0006610'\n",
      " 'short-chain fatty acid catabolic process GO0019626'\n",
      " 'single-strand break-containing DNA binding GO1990165'\n",
      " 'single strand break repair GO0000012'\n",
      " 'stereocilia ankle link complex GO0002142'\n",
      " 'sulfate transmembrane transport GO1902358'\n",
      " 'sulfate transmembrane transporter activity GO0015116'\n",
      " 'tRNA aminoacylation GO0043039'\n",
      " 'tRNA cytosine methyltransferase activity GO0016427'\n",
      " 'telomeric 3 overhang formation GO0031860' 'ubiquinone binding GO0048039'\n",
      " 'virus receptor activity GO0001618' 'RISC complex GO0016442'\n",
      " 'negative regulation of canonical Wnt signaling pathway GO0090090'\n",
      " 'dendritic spine GO0043197'\n",
      " 'regulation of protein localization GO0032880'\n",
      " 'G protein-coupled receptor binding GO0001664'\n",
      " 'animal organ morphogenesis GO0009887' 'centriolar satellite GO0034451'\n",
      " 'postsynapse GO0098794'\n",
      " 'positive regulation of focal adhesion assembly GO0051894'\n",
      " 'positive regulation of neuron apoptotic process GO0043525'\n",
      " 'U2-type precatalytic spliceosome GO0071005'\n",
      " 'negative regulation of inflammatory response GO0050728'\n",
      " 'positive regulation of inflammatory response GO0050729'\n",
      " 'MAP kinase kinase activity GO0004708' 'tRNA binding GO0000049'\n",
      " '5-flap endonuclease activity GO0017108'\n",
      " 'ATP-dependent chromatin remodeler activity GO0140658'\n",
      " 'DNA repair complex GO1990391' 'NADPH regeneration GO0006740'\n",
      " 'Schwann cell proliferation GO0014010'\n",
      " 'actin crosslink formation GO0051764'\n",
      " 'base-excision repair, gap-filling GO0006287'\n",
      " 'cellular response to steroid hormone stimulus GO0071383'\n",
      " 'commissural neuron axon guidance GO0071679'\n",
      " 'dosage compensation by inactivation of X chromosome GO0009048'\n",
      " 'positive regulation of axon extension involved in axon guidance GO0048842'\n",
      " 'positive regulation of phosphatidylinositol 3-kinase activity GO0043552'\n",
      " 'positive regulation of phosphorylation GO0042327'\n",
      " 'positive regulation of toll-like receptor 9 signaling pathway GO0034165'\n",
      " 'proton transmembrane transport GO1902600'\n",
      " 'regulation of double-strand break repair via nonhomologous end joining GO2001032'\n",
      " 'regulation of gliogenesis GO0014013'\n",
      " 'regulation of protein binding GO0043393'\n",
      " 'telomere maintenance via semi-conservative replication GO0032201'\n",
      " 'tricarboxylic acid cycle GO0006099'\n",
      " 'vascular endothelial growth factor receptor-2 signaling pathway GO0036324'\n",
      " 'brahma complex GO0035060'\n",
      " 'circadian regulation of translation GO0097167' 'nBAF complex GO0071565'\n",
      " 'npBAF complex GO0071564'\n",
      " 'positive regulation of chemokine C-X-C motif ligand 2 production GO2000343'\n",
      " 'regulation of protein secretion GO0050708'\n",
      " 'response to peptide hormone GO0043434' 'cell division GO0051301'\n",
      " 'cortical actin cytoskeleton GO0030864'\n",
      " 'positive regulation of neuron projection development GO0010976'\n",
      " 'positive regulation of protein serinethreonine kinase activity GO0071902'\n",
      " 'Golgi-associated vesicle membrane GO0030660' 'NuRD complex GO0016581'\n",
      " 'activation of NF-kappaB-inducing kinase activity GO0007250'\n",
      " 'extrinsic apoptotic signaling pathway in absence of ligand GO0097192'\n",
      " 'membrane organization GO0061024'\n",
      " 'negative regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway GO1902236'\n",
      " 'phosphatidylinositol-5-phosphate binding GO0010314'\n",
      " 'regulation of cell motility GO2000145'\n",
      " 'RNA polymerase I activity GO0001054'\n",
      " 'RNA polymerase I complex GO0005736'\n",
      " 'positive regulation of dendrite development GO1900006'\n",
      " 'regulation of cell differentiation GO0045595'\n",
      " 'mitotic G2 DNA damage checkpoint signaling GO0007095'\n",
      " 'phosphatidylinositol-3-phosphate binding GO0032266'\n",
      " 'positive regulation of protein autophosphorylation GO0031954'\n",
      " 'axonogenesis GO0007409'\n",
      " 'regulation of focal adhesion assembly GO0051893'\n",
      " 'cellular response to transforming growth factor beta stimulus GO0071560'\n",
      " 'phosphatidylinositol-3,4-bisphosphate binding GO0043325'\n",
      " 'phosphatidylinositol-3,5-bisphosphate binding GO0080025'\n",
      " 'positive regulation of epithelial cell migration GO0010634'\n",
      " 'type I transforming growth factor beta receptor binding GO0034713'\n",
      " 'cellular component GO0005575'\n",
      " 'phosphatidylserine acyl-chain remodeling GO0036150'\n",
      " 'face morphogenesis GO0060325' 'outflow tract morphogenesis GO0003151'\n",
      " 'defense response to Gram-positive bacterium GO0050830'\n",
      " 'microtubule nucleation GO0007020'\n",
      " 'negative regulation of vascular associated smooth muscle cell proliferation GO1904706'\n",
      " 'ribonucleoprotein complex GO1990904' 'spindle microtubule GO0005876'\n",
      " 'RNA splicing, via transesterification reactions GO0000375'\n",
      " 'GINS complex GO0000811' 'N-glycan processing GO0006491'\n",
      " 'NADP metabolic process GO0006739' 'adult heart development GO0007512'\n",
      " 'double-strand break repair via synthesis-dependent strand annealing GO0045003'\n",
      " 'extrinsic component of omegasome membrane GO0097629'\n",
      " 'extrinsic component of phagophore assembly site membrane GO0097632'\n",
      " 'forebrain morphogenesis GO0048853'\n",
      " 'forebrain radial glial cell differentiation GO0021861'\n",
      " 'glutamate secretion GO0014047'\n",
      " 'histone H3K36 demethylase activity GO0051864'\n",
      " 'hydrogen peroxide biosynthetic process GO0050665'\n",
      " 'mRNA N6-adenosine-methyltransferase activity GO0001734'\n",
      " 'negative regulation of axonogenesis GO0050771'\n",
      " 'negative regulation of synapse assembly GO0051964'\n",
      " 'neuron projection guidance GO0097485'\n",
      " 'neuronal cell body membrane GO0032809'\n",
      " 'positive regulation of DNA primase activity GO1903934'\n",
      " 'positive regulation of melanin biosynthetic process GO0048023'\n",
      " 'regulation of protein lipidation GO1903059'\n",
      " 'superoxide anion generation GO0042554'\n",
      " 'vitamin D response element binding GO0070644'\n",
      " 'miRNA processing GO0035196' 'muscle contraction GO0006936'\n",
      " 'positive regulation of mitotic nuclear division GO0045840'\n",
      " 'protein autoubiquitination GO0051865'\n",
      " 'ATP-dependent activity, acting on DNA GO0008094'\n",
      " 'lipid homeostasis GO0055088'\n",
      " 'positive regulation of protein kinase activity GO0045860'\n",
      " 'SCF ubiquitin ligase complex GO0019005'\n",
      " 'nucleocytoplasmic transport GO0006913'\n",
      " 'ESCRT III complex disassembly GO1904903'\n",
      " 'SREBP-SCAP-Insig complex GO0032937'\n",
      " 'SREBP-SCAP complex retention in endoplasmic reticulum GO0036316'\n",
      " 'SREBP signaling pathway GO0032933' 'amyloid-beta clearance GO0097242'\n",
      " 'amyloid precursor protein metabolic process GO0042982'\n",
      " 'carboxylesterase activity GO0106435' 'catabolic process GO0009056'\n",
      " 'establishment of spindle orientation GO0051294'\n",
      " 'fibroblast growth factor receptor activity GO0005007'\n",
      " 'hematopoietic stem cell proliferation GO0071425'\n",
      " 'intracellular oxygen homeostasis GO0032364'\n",
      " 'lipid droplet formation GO0140042'\n",
      " 'lipid droplet organization GO0034389'\n",
      " 'lipoprotein lipase activity GO0004465'\n",
      " 'mitochondrial translational termination GO0070126'\n",
      " 'negative regulation of ubiquitin-dependent protein catabolic process GO2000059'\n",
      " 'peptidyl-proline hydroxylation GO0019511'\n",
      " 'positive regulation of viral entry into host cell GO0046598'\n",
      " 'regulation of ATP-dependent activity GO0043462'\n",
      " 'regulation of establishment or maintenance of cell polarity GO0032878'\n",
      " 'regulation of mitochondrion organization GO0010821'\n",
      " 'regulation of multicellular organism growth GO0040014'\n",
      " 'sulfate transport GO0008272' 'synaptic vesicle maturation GO0016188'\n",
      " 'ubiquitin-like protein conjugating enzyme binding GO0044390'\n",
      " 'BH3 domain binding GO0051434' 'kinetochore microtubule GO0005828'\n",
      " 'lipopolysaccharide binding GO0001530'\n",
      " 'protein phosphatase inhibitor activity GO0004864'\n",
      " 'regulation of ubiquitin-protein transferase activity GO0051438'\n",
      " 'potassium channel regulator activity GO0015459'\n",
      " 'response to endoplasmic reticulum stress GO0034976'\n",
      " 'cysteine-type endopeptidase activity GO0004197'\n",
      " 'endoplasmic reticulum exit site GO0070971'\n",
      " 'heart morphogenesis GO0003007'\n",
      " 'mRNA splicing, via spliceosome GO0000398'\n",
      " 'positive regulation of cytokine production GO0001819'\n",
      " 'positive regulation of epidermal growth factor receptor signaling pathway GO0045742'\n",
      " 'CMG complex GO0071162'\n",
      " 'L-asparagine transmembrane transporter activity GO0015182'\n",
      " 'Roundabout binding GO0048495' 'U2-type spliceosomal complex GO0005684'\n",
      " 'response to bacterium GO0009617'\n",
      " 'very long-chain fatty acid metabolic process GO0000038'\n",
      " 'structural molecule activity GO0005198'\n",
      " 'GTPase activator activity GO0005096' 'SWISNF complex GO0016514'\n",
      " 'cell adhesion GO0007155' 'cullin family protein binding GO0097602'\n",
      " 'platelet aggregation GO0070527'\n",
      " 'regulation of G0 to G1 transition GO0070316'\n",
      " 'regulation of mitotic metaphaseanaphase transition GO0030071'\n",
      " 'regulation of nucleotide-excision repair GO2000819'\n",
      " 'response to ionizing radiation GO0010212'\n",
      " 'Bcl-2 family protein complex GO0097136' 'collagen binding GO0005518'\n",
      " 'endoplasmic reticulum mannose trimming GO1904380'\n",
      " 'cellular response to oxidative stress GO0034599'\n",
      " 'kinase binding GO0019900' 'mitotic cell cycle GO0000278'\n",
      " 'protein K48-linked ubiquitination GO0070936'\n",
      " 'ubiquitin-dependent ERAD pathway GO0030433'\n",
      " 'ER-associated misfolded protein catabolic process GO0071712'\n",
      " 'Fc-gamma receptor signaling pathway involved in phagocytosis GO0038096'\n",
      " 'I band GO0031674' 'S-adenosyl-L-methionine binding GO1904047'\n",
      " 'late endosome to vacuole transport via multivesicular body sorting pathway GO0032511'\n",
      " 'nuclear pore central transport channel GO0044613'\n",
      " 'phosphatidylcholine acyl-chain remodeling GO0036151'\n",
      " 'positive regulation of focal adhesion disassembly GO0120183'\n",
      " 'signaling receptor complex adaptor activity GO0030159'\n",
      " 'supercoiled DNA binding GO0097100' 'PAR polarity complex GO0120157'\n",
      " 'establishment of endothelial barrier GO0061028'\n",
      " 'negative regulation of NF-kappaB transcription factor activity GO0032088'\n",
      " 'calcium channel regulator activity GO0005246'\n",
      " 'negative regulation of insulin secretion GO0046676'\n",
      " 'negative regulation of intrinsic apoptotic signaling pathway GO2001243'\n",
      " 'polyA+ mRNA export from nucleus GO0016973'\n",
      " 'positive regulation of smooth muscle contraction GO0045987'\n",
      " 'positive regulation of vascular endothelial cell proliferation GO1905564'\n",
      " 'protein-disulfide reductase activity GO0015035'\n",
      " 'protein kinase C binding GO0005080'\n",
      " 'acyl-CoA hydrolase activity GO0047617'\n",
      " 'motor neuron axon guidance GO0008045'\n",
      " 'negative regulation of fat cell proliferation GO0070345'\n",
      " 'nucleolus GO0005730' 'paracrine signaling GO0038001'\n",
      " 'regulation of rhodopsin mediated signaling pathway GO0022400'\n",
      " 'ribosomal large subunit binding GO0043023'\n",
      " 'striated muscle contraction GO0006941' 'thioesterase binding GO0031996'\n",
      " 'DNA-binding transcription repressor activity GO0001217'\n",
      " 'Golgi apparatus GO0005794'\n",
      " 'L-serine transmembrane transporter activity GO0015194'\n",
      " 'L-serine transport GO0015825' 'RNA polymerase I core binding GO0001042'\n",
      " 'acetylcholine-mediated vasodilation involved in regulation of systemic arterial blood pressure GO0003069'\n",
      " 'acetylpyruvate hydrolase activity GO0018773'\n",
      " 'acrosome assembly GO0001675' 'adult locomotory behavior GO0008344'\n",
      " 'autophagosome membrane docking GO0016240'\n",
      " 'calcium ion transmembrane transporter activity GO0015085'\n",
      " 'cellular response to low-density lipoprotein particle stimulus GO0071404'\n",
      " 'cellular response to nutrient levels GO0031669'\n",
      " 'de novo actin filament nucleation GO0070060'\n",
      " 'embryo development GO0009790'\n",
      " 'fumarylpyruvate hydrolase activity GO0034545'\n",
      " 'gamma-aminobutyric acid signaling pathway GO0007214'\n",
      " 'intracellular cholesterol transport GO0032367'\n",
      " 'intracellular copper ion homeostasis GO0006878'\n",
      " 'negative regulation of cAMP-mediated signaling GO0043951'\n",
      " 'negative regulation of myotube differentiation GO0010832'\n",
      " 'negative regulation of stem cell differentiation GO2000737'\n",
      " 'nervous system process GO0050877'\n",
      " 'organic substance transport GO0071702'\n",
      " 'oxaloacetate decarboxylase activity GO0008948'\n",
      " 'positive regulation of myelination GO0031643'\n",
      " 'positive regulation of vascular associated smooth muscle cell differentiation involved in phenotypic switching GO1905932'\n",
      " 'regulation of phosphorylation GO0042325'\n",
      " 'regulation of postsynaptic membrane neurotransmitter receptor levels GO0099072'\n",
      " 'regulation of somitogenesis GO0014807'\n",
      " 'regulation of telomerase RNA localization to Cajal body GO1904872'\n",
      " 'response to superoxide GO0000303'\n",
      " 'response to type II interferon GO0034341'\n",
      " 'retinal cone cell development GO0046549'\n",
      " 'ribosomal subunit export from nucleus GO0000054'\n",
      " 'selenium binding GO0008430'\n",
      " 'serine transmembrane transporter activity GO0022889'\n",
      " 'small ribosomal subunit rRNA binding GO0070181'\n",
      " 'steroid binding GO0005496' 'steroid hormone binding GO1990239'\n",
      " 'superoxide dismutase activity GO0004784'\n",
      " 'tRNA catabolic process GO0016078' 'tRNA processing GO0008033'\n",
      " 'telomerase RNA binding GO0070034' 'thyroid hormone generation GO0006590'\n",
      " 'thyroxine 5-deiodinase activity GO0004800'\n",
      " 'thyroxine 5-deiodinase activity GO0033798' 'xenophagy GO0098792'\n",
      " 'actin filament GO0005884'\n",
      " 'establishment or maintenance of cell polarity GO0007163'\n",
      " 'positive regulation of lamellipodium assembly GO0010592'\n",
      " 'Wnt signaling pathway, planar cell polarity pathway GO0060071'\n",
      " 'apical junction complex GO0043296'\n",
      " 'intracellular protein transport GO0006886'\n",
      " 'membrane insertase activity GO0032977'\n",
      " 'flavin adenine dinucleotide binding GO0050660'\n",
      " 'Golgi organization GO0007030' 'cell-cell signaling GO0007267'\n",
      " 'protein-macromolecule adaptor activity GO0030674'\n",
      " 'transcription by RNA polymerase II GO0006366' 'ATPase binding GO0051117'\n",
      " 'cell leading edge GO0031252'\n",
      " 'cytokine-mediated signaling pathway GO0019221'\n",
      " 'negative regulation of protein phosphorylation GO0001933'\n",
      " 'phosphatidylserine binding GO0001786' 'biological process GO0008150'\n",
      " 'deubiquitinase activity GO0101005' 'phospholipid homeostasis GO0055091'\n",
      " 'positive regulation of trophoblast cell migration GO1901165'\n",
      " 'K63-linked deubiquitinase activity GO0061578'\n",
      " 'T cell receptor signaling pathway GO0050852'\n",
      " 'antiviral innate immune response GO0140374' 'filopodium GO0030175'\n",
      " 'heat shock protein binding GO0031072' 'mesoderm development GO0007498'\n",
      " 'nucleotide-excision repair GO0006289' 'cytokine activity GO0005125'\n",
      " 'sodium ion transmembrane transport GO0035725'\n",
      " 'negative regulation of double-strand break repair via homologous recombination GO2000042'\n",
      " 'positive regulation of blood vessel endothelial cell migration GO0043536'\n",
      " 'spliceosomal complex GO0005681' 'ubiquitin binding GO0043130'\n",
      " 'DNA-binding transcription repressor activity, RNA polymerase II-specific GO0001227'\n",
      " 'IMP metabolic process GO0046040' 'T cell activation GO0042110'\n",
      " 'acyltransferase activity GO0016746'\n",
      " 'aldo-keto reductase NADP activity GO0004033'\n",
      " 'apoptotic DNA fragmentation GO0006309' 'fatty acid binding GO0005504'\n",
      " 'hormone binding GO0042562' 'myoblast differentiation GO0045445'\n",
      " 'pantothenate kinase activity GO0004594'\n",
      " 'positive regulation of leukocyte chemotaxis GO0002690'\n",
      " 'positive regulation of monocyte chemotaxis GO0090026'\n",
      " 'positive regulation of ubiquitin protein ligase activity GO1904668'\n",
      " 'positive regulation of vascular associated smooth muscle cell apoptotic process GO1905461'\n",
      " 'regulation of DNA-templated DNA replication initiation GO0030174'\n",
      " 'sarcoplasmic reticulum GO0016529' 'vesicle organization GO0016050'\n",
      " 'blood coagulation GO0007596'\n",
      " 'hematopoietic stem cell homeostasis GO0061484'\n",
      " 'positive regulation of B cell proliferation GO0030890'\n",
      " 'positive regulation of receptor-mediated endocytosis GO0048260'\n",
      " 'triglyceride metabolic process GO0006641' 'nuclear matrix GO0016363'\n",
      " 'positive regulation of intracellular estrogen receptor signaling pathway GO0033148'\n",
      " 'endosome membrane GO0010008' 'endocytosis GO0006897'\n",
      " 'amino acid transmembrane transport GO0003333'\n",
      " 'DNA damage response GO0006974'\n",
      " 'cellular response to chemokine GO1990869'\n",
      " 'regulation of cell fate specification GO0042659'\n",
      " 'DNA damage response, signal transduction by p53 class mediator GO0030330'\n",
      " 'GID complex GO0034657' 'Golgi cisterna membrane GO0032580'\n",
      " 'RNA N6-methyladenosine methyltransferase complex GO0036396'\n",
      " 'asymmetric cell division GO0008356' 'bBAF complex GO0140092'\n",
      " 'blood vessel morphogenesis GO0048514'\n",
      " 'cGMP biosynthetic process GO0006182' 'dense body GO0097433'\n",
      " 'gamma-tubulin ring complex GO0000931'\n",
      " 'glial cell differentiation GO0010001'\n",
      " 'guanylate cyclase activity GO0004383'\n",
      " 'negative regulation of oxidative stress-induced intrinsic apoptotic signaling pathway GO1902176'\n",
      " 'nerve growth factor signaling pathway GO0038180'\n",
      " 'neural crest cell development GO0014032'\n",
      " 'nitric-oxide synthase binding GO0050998'\n",
      " 'nucleotide-binding oligomerization domain containing signaling pathway GO0070423'\n",
      " 'peptidyl-lysine methylation GO0018022'\n",
      " 'positive regulation by host of viral process GO0044794'\n",
      " 'positive regulation of centriole replication GO0046601'\n",
      " 'positive regulation of endothelial cell chemotaxis GO2001028'\n",
      " 'positive regulation of osteoblast proliferation GO0033690'\n",
      " 'positive regulation of vitamin D receptor signaling pathway GO0070564'\n",
      " 'protein methylation GO0006479'\n",
      " 'regulation of cell junction assembly GO1901888'\n",
      " 'regulation of hair cycle GO0042634'\n",
      " 'regulation of transepithelial transport GO0150111'\n",
      " 'retinoic acid-responsive element binding GO0044323'\n",
      " 'sterol transporter activity GO0015248'\n",
      " 'cellular response to glucose starvation GO0042149'\n",
      " 'protein transport GO0015031'\n",
      " 'intrinsic apoptotic signaling pathway in response to DNA damage GO0008630'\n",
      " 'protein phosphatase 2A binding GO0051721' 'histone binding GO0042393'\n",
      " 'negative regulation of fibroblast proliferation GO0048147'\n",
      " 'positive regulation of DNA binding GO0043388'\n",
      " 'regulation of synaptic plasticity GO0048167' 'ATP transport GO0015867'\n",
      " 'DNA N-glycosylase activity GO0019104'\n",
      " 'aflatoxin metabolic process GO0046222' 'basal cortex GO0045180'\n",
      " 'calcium-independent phospholipase A2 activity GO0047499'\n",
      " 'cellular response to staurosporine GO0072734'\n",
      " 'centrosomal corona GO0031592' 'channel activity GO0015267'\n",
      " 'clathrin vesicle coat GO0030125' 'dystroglycan binding GO0002162'\n",
      " 'exit from mitosis GO0010458'\n",
      " 'hematopoietic stem cell differentiation GO0060218'\n",
      " 'hemidesmosome GO0030056'\n",
      " 'histone arginine N-methyltransferase activity GO0008469'\n",
      " 'histone methylation GO0016571' 'kinetochore binding GO0043515'\n",
      " 'large ribosomal subunit rRNA binding GO0070180'\n",
      " 'microtubule organizing center organization GO0031023'\n",
      " 'negative regulation of cargo loading into COPII-coated vesicle GO1901303'\n",
      " 'negative regulation of mitochondrial membrane permeability involved in apoptotic process GO1902109'\n",
      " 'negative regulation of protein exit from endoplasmic reticulum GO0070862'\n",
      " 'negative regulation of vascular associated smooth muscle cell differentiation GO1905064'\n",
      " 'negative regulation of wound healing, spreading of epidermal cells GO1903690'\n",
      " 'oxidoreductase activity, acting on paired donors, with incorporation or reduction of molecular oxygen, reduced flavin or flavoprotein as one donor, and incorporation of one atom of oxygen GO0016712'\n",
      " 'phosphatidylcholine catabolic process GO0034638'\n",
      " 'positive regulation of epithelial cell proliferation involved in wound healing GO0060054'\n",
      " 'positive regulation of glomerular mesangial cell proliferation GO0072126'\n",
      " 'positive regulation of hemoglobin biosynthetic process GO0046985'\n",
      " 'positive regulation of oxidative stress-induced neuron intrinsic apoptotic signaling pathway GO1903378'\n",
      " 'regulation of intracellular estrogen receptor signaling pathway GO0033146'\n",
      " 'sulfuric ester hydrolase activity GO0008484'\n",
      " 'superoxide-generating NADPH oxidase activator activity GO0016176'\n",
      " 'vascular associated smooth muscle cell development GO0097084'\n",
      " 'venous blood vessel morphogenesis GO0048845'\n",
      " 'voltage-gated chloride channel activity GO0005247'\n",
      " 'cardiac vascular smooth muscle cell differentiation GO0060947'\n",
      " 'dendritic shaft GO0043198'\n",
      " 'double-stranded methylated DNA binding GO0010385'\n",
      " 'ion binding GO0043167' 'muscle alpha-actinin binding GO0051371'\n",
      " 'p38MAPK cascade GO0038066' 'perikaryon GO0043204'\n",
      " 'calcium ion transport GO0006816' 'lung epithelium development GO0060428'\n",
      " 'positive regulation of mRNA splicing, via spliceosome GO0048026'\n",
      " 'triglyceride lipase activity GO0004806'\n",
      " 'Fc-epsilon receptor signaling pathway GO0038095'\n",
      " 'condensed chromosome GO0000793' 'mRNA 3-UTR binding GO0003730'\n",
      " 'response to oxidative stress GO0006979' 'glucose homeostasis GO0042593'\n",
      " 'molecular function inhibitor activity GO0140678'\n",
      " 'voltage-gated potassium channel complex GO0008076'\n",
      " 'positive regulation of MAP kinase activity GO0043406'\n",
      " 'regulation of stem cell differentiation GO2000736'\n",
      " 'heparin binding GO0008201' 'miRNA binding GO0035198'\n",
      " 'positive regulation of double-strand break repair GO2000781'\n",
      " 'ruffle membrane GO0032587' 'skeletal muscle fiber development GO0048741'\n",
      " 'specific granule lumen GO0035580'\n",
      " 'cellular response to mechanical stimulus GO0071260'\n",
      " 'vesicle docking involved in exocytosis GO0006904'\n",
      " 'RISC-loading complex GO0070578' 'RISC complex assembly GO0070922'\n",
      " 'autophagosome maturation GO0097352'\n",
      " 'cortical actin cytoskeleton organization GO0030866'\n",
      " 'eye development GO0001654' 'fatty-acyl-CoA binding GO0000062'\n",
      " 'heterotrimeric G-protein complex GO0005834'\n",
      " 'lysosome localization GO0032418' 'microtubule cytoskeleton GO0015630'\n",
      " 'multivesicular body assembly GO0036258'\n",
      " 'negative regulation of epithelial to mesenchymal transition GO0010719'\n",
      " 'negative regulation of protein catabolic process GO0042177'\n",
      " 'negative regulation of protein localization to plasma membrane GO1903077'\n",
      " 'platelet activation GO0030168'\n",
      " 'positive regulation of interleukin-1 beta production GO0032731'\n",
      " 'positive regulation of tyrosine phosphorylation of STAT protein GO0042531'\n",
      " 'pre-miRNA processing GO0031054' 'protein glycosylation GO0006486'\n",
      " 'regulation of blood pressure GO0008217'\n",
      " 'response to toxic substance GO0009636' 'secretory granule GO0030141'\n",
      " 'L-aspartate transmembrane transporter activity GO0015183'\n",
      " 'MyD88-dependent toll-like receptor signaling pathway GO0002755'\n",
      " 'P granule GO0043186' 'artery morphogenesis GO0048844'\n",
      " 'barbed-end actin filament capping GO0051016'\n",
      " 'endoplasmic reticulum tubular network organization GO0071786'\n",
      " 'epidermal growth factor receptor binding GO0005154'\n",
      " 'protein localization to cytoplasmic stress granule GO1903608'\n",
      " 'regulation of DNA replication origin binding GO1902595'\n",
      " 'signal sequence binding GO0005048'\n",
      " '1-acylglycerol-3-phosphate O-acyltransferase activity GO0003841'\n",
      " 'activation of cysteine-type endopeptidase activity GO0097202'\n",
      " 'adenylate cyclase-modulating G protein-coupled receptor signaling pathway GO0007188'\n",
      " 'axon initial segment GO0043194'\n",
      " 'branching morphogenesis of an epithelial tube GO0048754'\n",
      " 'cation channel complex GO0034703'\n",
      " 'cellular response to unfolded protein GO0034620'\n",
      " 'cobalamin binding GO0031419'\n",
      " 'cytoplasmic sequestering of transcription factor GO0042994'\n",
      " 'detection of mechanical stimulus GO0050982'\n",
      " 'histone methyltransferase binding GO1990226'\n",
      " 'intrinsic apoptotic signaling pathway in response to endoplasmic reticulum stress GO0070059'\n",
      " 'metanephric collecting duct development GO0072205'\n",
      " 'monoatomic cation channel activity GO0005261'\n",
      " 'negative regulation of substrate adhesion-dependent cell spreading GO1900025'\n",
      " 'positive regulation of natural killer cell mediated cytotoxicity GO0045954'\n",
      " 'regulation of cellular localization GO0060341'\n",
      " 'regulation of centrosome duplication GO0010824'\n",
      " 'regulation of sodium ion transport GO0002028'\n",
      " 'telomere maintenance via telomerase GO0007004'\n",
      " 'ciliary membrane GO0060170' 'double-stranded RNA binding GO0003725'\n",
      " 'microtubule cytoskeleton organization GO0000226' 'vesicle GO0031982'\n",
      " 'ATP-dependent protein disaggregase activity GO0140545'\n",
      " 'Atg1ULK1 kinase complex GO1990316' 'DNA duplex unwinding GO0032508'\n",
      " 'G protein-coupled receptor signaling pathway, coupled to cyclic nucleotide second messenger GO0007187'\n",
      " 'Notch binding GO0005112' 'RNA polymerase II, core complex GO0005665'\n",
      " 'RPAP3R2TPprefoldin-like complex GO1990062'\n",
      " 'U2-type prespliceosome GO0071004' 'actinin binding GO0042805'\n",
      " 'aminoacyl-tRNA synthetase multienzyme complex GO0017101'\n",
      " 'apoptotic mitochondrial changes GO0008637' 'axon terminus GO0043679'\n",
      " 'blood vessel diameter maintenance GO0097746'\n",
      " 'cardiac left ventricle morphogenesis GO0003214'\n",
      " 'cell-cell contact zone GO0044291'\n",
      " 'cobalamin metabolic process GO0009235'\n",
      " 'collagen receptor activity GO0038064' 'complement activation GO0006956'\n",
      " 'corrinoid adenosyltransferase activity GO0008817'\n",
      " 'gliogenesis GO0042063'\n",
      " 'mannosyl-oligosaccharide 1,2-alpha-mannosidase activity GO0004571'\n",
      " 'mitochondrial fusion GO0008053'\n",
      " 'modulation of chemical synaptic transmission GO0050804'\n",
      " 'negative regulation of anoikis GO2000811' 'nuclear envelope GO0005635'\n",
      " 'oligosaccharide biosynthetic process GO0009312'\n",
      " 'phosphatase inhibitor activity GO0019212'\n",
      " 'positive chemotaxis GO0050918'\n",
      " 'protein disulfide isomerase activity GO0003756'\n",
      " 'protein localization to nuclear envelope GO0090435'\n",
      " 'protein methyltransferase activity GO0008276' 'protein tag GO0031386'\n",
      " 'regulation of mitotic spindle assembly GO1901673'\n",
      " 'suppression by virus of host apoptotic process GO0019050'\n",
      " 'tRNA aminoacylation for protein translation GO0006418'\n",
      " 'transcription by RNA polymerase I GO0006360'\n",
      " 'transferase activity, transferring alkyl or aryl other than methyl groups GO0016765']\n",
      "Target genes (miRTarBase)\n",
      "['FAM241A' 'KCNMB1' 'ASB16' 'RRP36' 'TOR4A' 'MRPL17' 'SLC7A11' 'TMOD3'\n",
      " 'FEM1A' 'MIDN' 'MEX3A' 'PDLIM5' 'POC1A' 'POLR2E' 'TMEM151A'\n",
      " '2022-03-07 000000' 'ABCF3' 'ABL2' 'ACO1' 'AGMO' 'AGTPBP1' 'ALDOA'\n",
      " 'AP1S3' 'APTX' 'BAIAP2L1' 'C3orf62' 'CABP4' 'CDT1' 'CENPI' 'CENPM'\n",
      " 'CNDP1' 'CNNM1' 'COMMD2' 'CTNS' 'DARS2' 'DCP1A' 'DENND5B' 'DNASE2'\n",
      " 'ERCC6L2' 'ETFDH' 'FKTN' 'GGPS1' 'GRK3' 'GTF3C6' 'HMGB1' 'HPSE' 'INO80'\n",
      " 'KDSR' 'KIAA0408' 'LETM2' 'LMAN2L' 'LRIF1' 'LRP10' 'LRRC58' 'MACC1'\n",
      " 'MAP3K9' 'MCM8' 'METTL2B' 'MRE11' 'MTRF1L' 'MYDGF' 'NPR1' 'NQO2' 'NT5C2'\n",
      " 'OGFOD1' 'PALM2' 'PCCB' 'PIAS2' 'PIWIL1' 'PLEKHM3' 'RAPGEF1' 'RFC2'\n",
      " 'RGP1' 'RGS17' 'SAR1A' 'SEC23IP' 'SLC25A45' 'SLC26A2' 'STPG1' 'TACO1'\n",
      " 'TLCD2' 'TOP3A' 'TTLL12' 'UGGT2' 'VEZT' 'VTA1' 'ZDHHC15' 'ZNF106'\n",
      " 'ZNF383' 'ZNF573' 'ZNF677' 'ZNF716' 'ZNF786' 'NF2' 'RNF149' 'FAM136A'\n",
      " 'GIPC1' 'MEF2D' 'METTL14' 'METTL21A' 'MRO' 'OSBPL10' 'SLIT1' 'SP110'\n",
      " 'ZBTB46' 'ZNF561' 'CES2' 'CRISPLD2' 'LLGL1' 'KLHL36' 'HSPE1-MOB4' 'MOB4'\n",
      " 'SYAP1' 'ZNF347' 'ZNF8' 'CALCOCO2' 'CCDC171' 'DIO2' 'EMC7' 'FAHD1'\n",
      " 'FAM126B' 'FAM71F2' 'GPR137C' 'GTDC1' 'JMY' 'MRNIP' 'MRPL51' 'MRPS16'\n",
      " 'NR3C1' 'PANK3' 'QKI' 'SOD2' 'SPTY2D1' 'TGOLN2' 'WDR26' 'ZNF623' 'OCIAD2'\n",
      " 'C9orf3' 'CLASP1' 'INSIG1' 'MCL1' 'MTRNR2L5' 'RAB5B' 'RNF157' 'RPL12'\n",
      " 'TMED8' 'VHLL' 'BCL2L1' 'MMAB']\n",
      "Pathways (KEGG)\n",
      "['Glycosaminoglycan degradation' 'RNA polymerase' 'GnRH secretion'\n",
      " 'Cytosolic DNA-sensing pathway' 'Fc gamma R-mediated phagocytosis'\n",
      " 'Cholinergic synapse' 'Transcriptional misregulation in cancer'\n",
      " 'Hedgehog signaling pathway' 'Mannose type O-glycan biosynthesis'\n",
      " 'Apelin signaling pathway' 'Central carbon metabolism in cancer'\n",
      " 'Alzheimer disease' 'Lipid and atherosclerosis' 'Wnt signaling pathway'\n",
      " 'Calcium signaling pathway' 'Phagosome' 'VEGF signaling pathway'\n",
      " 'Ribosome biogenesis in eukaryotes' 'Glutamatergic synapse'\n",
      " 'Propanoate metabolism' 'Prion disease' 'Yersinia infection'\n",
      " 'Adrenergic signaling in cardiomyocytes' 'Estrogen signaling pathway'\n",
      " 'Relaxin signaling pathway' 'Ferroptosis'\n",
      " 'Chemical carcinogenesis - reactive oxygen species'\n",
      " 'Vasopressin-regulated water reabsorption' 'Apoptosis'\n",
      " 'Chemical carcinogenesis - receptor activation'\n",
      " 'Cortisol synthesis and secretion' 'Legionellosis' 'Dopaminergic synapse'\n",
      " 'Glutathione metabolism' 'Thyroid hormone synthesis' 'Thyroid cancer'\n",
      " 'Basal cell carcinoma' 'Sphingolipid signaling pathway'\n",
      " 'Aldosterone synthesis and secretion' 'Amoebiasis' 'Renin secretion'\n",
      " 'Alcoholic liver disease' 'Tuberculosis' 'Apoptosis - multiple species'\n",
      " 'NF-kappa B signaling pathway' 'mRNA surveillance pathway'\n",
      " 'Fc epsilon RI signaling pathway' 'GnRH signaling pathway'\n",
      " 'Long-term potentiation' 'NOD-like receptor signaling pathway'\n",
      " 'Ovarian steroidogenesis' 'Herpes simplex virus 1 infection'\n",
      " 'Non-alcoholic fatty liver disease' 'Taurine and hypotaurine metabolism'\n",
      " 'Human immunodeficiency virus 1 infection'\n",
      " 'Cysteine and methionine metabolism' 'Homologous recombination'\n",
      " 'Dilated cardiomyopathy' 'Lysosome' 'Choline metabolism in cancer'\n",
      " 'Gap junction' 'Diabetic cardiomyopathy'\n",
      " 'Fluid shear stress and atherosclerosis'\n",
      " 'Growth hormone synthesis, secretion and action'\n",
      " 'Neurotrophin signaling pathway' 'beta-Alanine metabolism']\n",
      "Diseases (MNDR)\n",
      "['Hepatocellular carcinoma' 'Cardiovascular disease' 'Pancreatic cancer']\n",
      "Cell-type specific (Cellular microRNAome)\n",
      "['Beta cell like derived specific']\n",
      "GO Molecular function (miRPathDB)\n",
      "['cell adhesion molecule binding']\n",
      "Interactions (NPInter)\n",
      "['ARMCX4' 'C7orf41' 'CALU' 'CAND1' 'CASQ2' 'CHN2' 'CSRNP3' 'DEPDC1B' 'DLD'\n",
      " 'DNER' 'EFR3A' 'FBXW11' 'GFM1' 'GNL3L' 'HEY1' 'HOXA9' 'ILF3' 'KLF6'\n",
      " 'LEPROT' 'LONRF1' 'MFHAS1' 'MOBP' 'MRC1' 'POLR1D' 'PQLC1' 'R3HDM2'\n",
      " 'RAP1B' 'RASGEF1A' 'RBMS1' 'SPTBN1' 'TLE3' 'TOMM70A' 'UBAP2L' 'VLDLR'\n",
      " 'ZNF644' 'ZSWIM6' 'PCF11' 'GIGYF1' 'ZDHHC18']\n"
     ]
    }
   ],
   "source": [
    "# find the overlap of the overlap of supepath name and subcategory\n",
    "for i in miEAA['Category'].unique():\n",
    "    print(i)\n",
    "    print(miEAA[miEAA['Category'] == i]['Subcategory'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Signal Transduction', 'Metabolism of Proteins',\n",
       "       'Infectious Disease', 'Gene Expression (Transcription)',\n",
       "       'Regulation of Activated PAK-2p34 By Proteasome Mediated Degradation',\n",
       "       'Cellular Responses to Stimuli', 'Cell Cycle, Mitotic',\n",
       "       'Signaling By Rho GTPases', 'Metabolism',\n",
       "       'Processing of Capped Intron-Containing Pre-mRNA', 'DNA Damage',\n",
       "       'Regulation of TP53 Activity', 'RNA Polymerase I Promoter Opening',\n",
       "       'Vesicle-mediated Transport', 'Cell Cycle Checkpoints',\n",
       "       'Pleural Mesothelioma',\n",
       "       'Transport of The SLBP Independent Mature MRNA',\n",
       "       'Innate Immune System', 'Cellular Senescence',\n",
       "       'Cytoskeletal Signaling', 'Translational Control',\n",
       "       'Prolactin Signaling', 'MAPK Family Signaling Cascades',\n",
       "       'Endometrial Cancer', 'Homology Directed Repair',\n",
       "       'Mitotic G1 Phase and G1/S Transition',\n",
       "       'PIP3 Activates AKT Signaling',\n",
       "       'EML4 and NUDC in Mitotic Spindle Formation',\n",
       "       'Chromatin Regulation / Acetylation',\n",
       "       'DNA Damage_ATM/ATR Regulation of G1/S Checkpoint',\n",
       "       'Apoptotic Pathways in Synovial Fibroblasts',\n",
       "       'PI3K-Akt Signaling Pathway',\n",
       "       'Assembly of The Pre-replicative Complex',\n",
       "       'PDGFR-beta Signaling Pathway',\n",
       "       'COPI-independent Golgi-to-ER Retrograde Traffic',\n",
       "       'Class I MHC Mediated Antigen Processing and Presentation',\n",
       "       'Separation of Sister Chromatids',\n",
       "       'Angiopoietin-like Protein 8 Regulatory Pathway',\n",
       "       'Defective Binding of RB1 Mutants to E2F1,(E2F2, E2F3)',\n",
       "       'Chks in Checkpoint Regulation',\n",
       "       'Validated Targets of C-MYC Transcriptional Activation',\n",
       "       'Glucose / Energy Metabolism',\n",
       "       'Sertoli-Sertoli Cell Junction Dynamics',\n",
       "       'Cell Cycle Regulation of G1/S Transition (part 1)',\n",
       "       'Signaling By TGFB Family Members',\n",
       "       'Development IGF-1 Receptor Signaling',\n",
       "       'Nervous System Development',\n",
       "       'MiRNA Regulation of DNA Damage Response',\n",
       "       'Toll-like Receptor Signaling Pathway', 'Focal Adhesion',\n",
       "       'RAC1 GTPase Cycle', 'Influenza Infection',\n",
       "       'ESR-mediated Signaling', 'MAPK Signaling: Mitogens',\n",
       "       'MAP Kinase Signaling', 'Gene Silencing By RNA',\n",
       "       'VEGFA-VEGFR2 Signaling', 'GPCR Pathway',\n",
       "       'Development Endothelin-1/EDNRA Transactivation of EGFR',\n",
       "       'GADD45 Pathway', 'TP53 Regulates Metabolic Genes',\n",
       "       'Activation of The Pre-replicative Complex', 'Cell Cycle',\n",
       "       'Autophagy Pathway',\n",
       "       'ADORA2B Mediated Anti-inflammatory Cytokines Production',\n",
       "       'Regulation of Expression of SLITs and ROBOs',\n",
       "       'SARS-CoV-2 Infection', 'MAPK-Erk Pathway',\n",
       "       'Development_TGF-beta Receptor Signaling', 'MAP Kinase Activation',\n",
       "       'Regulation of Actin Dynamics For Phagocytic Cup Formation',\n",
       "       'DNA Damage Response (only ATM Dependent)',\n",
       "       'Uptake and Actions of Bacterial Toxins',\n",
       "       'Cytokine Signaling in Immune System', 'RHOD GTPase Cycle',\n",
       "       'Lissencephaly Gene (LIS1) in Neuronal Migration and Development',\n",
       "       'G-protein Signaling RAC1 in Cellular Process',\n",
       "       'Glycerophospholipid Biosynthesis', 'MTOR Signaling Pathway',\n",
       "       'Exercise-induced Circadian Regulation',\n",
       "       'FLT3 Signaling in Disease', 'ERK Signaling', 'TGF-Beta Pathway',\n",
       "       'Response to Elevated Platelet Cytosolic Ca2+',\n",
       "       'Integrated Breast Cancer Pathway',\n",
       "       'Translation Insulin Regulation of Translation',\n",
       "       'Regulation of Actin Cytoskeleton', 'Signaling By ALK in Cancer',\n",
       "       'Development Slit-Robo Signaling', 'TGF-beta Signaling Pathways',\n",
       "       'Small Cell Lung Cancer', 'Golgi-to-ER Retrograde Transport',\n",
       "       'Formation of HIV Elongation Complex in The Absence of HIV Tat',\n",
       "       'Microtubule Cytoskeleton Regulation', 'HIV Life Cycle',\n",
       "       'P75 NTR Receptor-mediated Signalling', 'MAPK Signaling Pathway',\n",
       "       'RRNA Processing in The Nucleus and Cytosol',\n",
       "       'Peptide Chain Elongation', 'Beta-Adrenergic Signaling',\n",
       "       'Clathrin-mediated Endocytosis', 'TNFR1 Pathway', 'Neuroscience',\n",
       "       'Semaphorin Interactions', 'Ubiquitination Cascade',\n",
       "       'TNF Signaling', 'DNA IR-damage and Cellular Response Via ATR',\n",
       "       'Retinoblastoma Gene in Cancer', 'Circadian Clock',\n",
       "       'Packaging Of Telomere Ends',\n",
       "       'Smith-Magenis and Potocki-Lupski Syndrome Copy Number Variation',\n",
       "       'EPH-Ephrin Signaling',\n",
       "       'Cytoskeleton Remodeling Regulation of Actin Cytoskeleton By Rho GTPases',\n",
       "       'RHOQ GTPase Cycle',\n",
       "       'Loss of Proteins Required For Interphase Microtubule Organization From The Centrosome',\n",
       "       'ATM Signaling in Development and Disease', 'Signaling By PTK6',\n",
       "       'Signaling Events Mediated By HDAC Class I',\n",
       "       'Signaling By Receptor Tyrosine Kinases', 'P53 Pathway',\n",
       "       'Activated TAK1 Mediates P38 MAPK Activation',\n",
       "       'Signaling Downstream of RAS Mutants',\n",
       "       'Oncogene Induced Senescence', 'VEGF Signaling Pathway',\n",
       "       'RAC2 GTPase Cycle', 'IL2 Signaling Events Mediated By STAT5',\n",
       "       'Signaling By WNT',\n",
       "       'Stabilization and Expansion of The E-cadherin Adherens Junction',\n",
       "       'Sumoylation By RanBP2 Regulates Transcriptional Repression',\n",
       "       'Transcription Ligand-dependent Activation of The ESR1/SP Pathway',\n",
       "       'ATM Pathway',\n",
       "       'DNA IR-double Strand Breaks and Cellular Response Via ATM',\n",
       "       'FOXO-mediated Transcription',\n",
       "       'MyD88 Dependent Cascade Initiated on Endosome',\n",
       "       'N-cadherin Signaling Events',\n",
       "       'Cell Cycle Role of 14-3-3 Proteins in Cell Cycle Regulation',\n",
       "       'Remodeling of Adherens Junctions', 'Breast Cancer Pathway',\n",
       "       'TP53 Regulates Transcription of Cell Cycle Genes',\n",
       "       'Signaling By BMP',\n",
       "       'Development Dopamine D2 Receptor Transactivation of EGFR',\n",
       "       'E-cadherin Signaling in The Nascent Adherens Junction',\n",
       "       'Cyclins and Cell Cycle Regulation',\n",
       "       'FCERI Mediated Ca+2 Mobilization',\n",
       "       'Head and Neck Squamous Cell Carcinoma', 'HIF1Alpha Pathway',\n",
       "       'Nuclear Events Mediated By NFE2L2', 'Nectin Adhesion Pathway',\n",
       "       'Signaling By FGFR2', 'Senescence and Autophagy in Cancer',\n",
       "       'Chromatin Organization', 'GDNF Signaling',\n",
       "       'Regulation of RUNX1 Expression and Activity',\n",
       "       'PLK1 Signaling Events',\n",
       "       'Cooperation of Prefoldin and TriC/CCT in Actin and Tubulin Folding',\n",
       "       'Transcriptional Activity of SMAD2/SMAD3:SMAD4 Heterotrimer',\n",
       "       'Organelle Biogenesis and Maintenance',\n",
       "       'Metabolic Reprogramming in Colon Cancer',\n",
       "       'Serotonin HTR1 Group and FOS Pathway',\n",
       "       'Nuclear Receptors Meta-pathway',\n",
       "       'DNA Double Strand Break Response', 'Deubiquitination',\n",
       "       'Coregulation of Androgen Receptor Activity'], dtype=object)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc['SuperPath Name'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Category</th>\n",
       "      <th>Subcategory</th>\n",
       "      <th>Enrichment</th>\n",
       "      <th>P-value</th>\n",
       "      <th>P-adjusted</th>\n",
       "      <th>Q-value</th>\n",
       "      <th>Observed</th>\n",
       "      <th>miRNAs/precursors</th>\n",
       "      <th>Score</th>\n",
       "      <th>SuperPath Name</th>\n",
       "      <th># SuperPath Total Genes</th>\n",
       "      <th># SuperPath Matched Genes</th>\n",
       "      <th>Matched Genes (Symbols)</th>\n",
       "      <th>Evidence URL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Category, Subcategory, Enrichment, P-value, P-adjusted, Q-value, Observed, miRNAs/precursors, Score, SuperPath Name, # SuperPath Total Genes, # SuperPath Matched Genes, Matched Genes (Symbols), Evidence URL]\n",
       "Index: []"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlap = miEAA.merge(gc, left_on='Subcategory', right_on='SuperPath Name')\n",
    "overlap.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "myenv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
